// shippo/sdk/carrieraccounts.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CarrierAccounts = void 0;
const carrierAccountsCreate_js_1 = require("../funcs/carrierAccountsCreate.js");
const carrierAccountsGet_js_1 = require("../funcs/carrierAccountsGet.js");
const carrierAccountsGetRegistrationStatus_js_1 = require("../funcs/carrierAccountsGetRegistrationStatus.js");
const carrierAccountsInitiateOauth2Signin_js_1 = require("../funcs/carrierAccountsInitiateOauth2Signin.js");
const carrierAccountsList_js_1 = require("../funcs/carrierAccountsList.js");
const carrierAccountsRegister_js_1 = require("../funcs/carrierAccountsRegister.js");
const carrierAccountsUpdate_js_1 = require("../funcs/carrierAccountsUpdate.js");
const sdks_js_1 = require("../lib/sdks.js");
const fp_js_1 = require("../types/fp.js");
class CarrierAccounts extends sdks_js_1.ClientSDK {
    /**
     * List all carrier accounts
     *
     * @remarks
     * Returns a list of all carrier accounts connected to your Shippo account. These carrier accounts include both Shippo carrier accounts and your own carrier accounts that you have connected to your Shippo account.
     *
     * Additionally, you can get information about the service levels associated with each carrier account by passing in the `?service_levels=true` query parameter. <br>
     * Using it appends the property `service_levels` to each carrier account. <br>
     * By default, if the query parameter is omitted, the `service_levels` property will not be included in the response.
     */
    async list(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, carrierAccountsList_js_1.carrierAccountsList)(this, request, options));
    }
    /**
     * Create a new carrier account
     *
     * @remarks
     * Creates a new carrier account or connects an existing carrier account to the Shippo account.
     */
    async create(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, carrierAccountsCreate_js_1.carrierAccountsCreate)(this, request, options));
    }
    /**
     * Retrieve a carrier account
     *
     * @remarks
     * Returns an existing carrier account using an object ID.
     */
    async get(carrierAccountId, options) {
        return (0, fp_js_1.unwrapAsync)((0, carrierAccountsGet_js_1.carrierAccountsGet)(this, carrierAccountId, options));
    }
    /**
     * Update a carrier account
     *
     * @remarks
     * Updates an existing carrier account object. The account_id and carrier can't be updated. This is because they form the unique identifier together.
     */
    async update(carrierAccountId, carrierAccountBase, options) {
        return (0, fp_js_1.unwrapAsync)((0, carrierAccountsUpdate_js_1.carrierAccountsUpdate)(this, carrierAccountId, carrierAccountBase, options));
    }
    /**
     * Connect an existing carrier account using OAuth 2.0
     *
     * @remarks
     * Used by client applications to setup or reconnect an existing carrier account with carriers that support OAuth 2.0
     */
    async initiateOauth2Signin(carrierAccountObjectId, redirectUri, state, options) {
        return (0, fp_js_1.unwrapAsync)((0, carrierAccountsInitiateOauth2Signin_js_1.carrierAccountsInitiateOauth2Signin)(this, carrierAccountObjectId, redirectUri, state, options));
    }
    /**
     * Add a Shippo carrier account
     *
     * @remarks
     * Adds a Shippo carrier account
     */
    async register(request, options) {
        return (0, fp_js_1.unwrapAsync)((0, carrierAccountsRegister_js_1.carrierAccountsRegister)(this, request, options));
    }
    /**
     * Get Carrier Registration status
     *
     * @remarks
     * Returns the registration status for the given account for the given carrier
     */
    async getRegistrationStatus(carrier, options) {
        return (0, fp_js_1.unwrapAsync)((0, carrierAccountsGetRegistrationStatus_js_1.carrierAccountsGetRegistrationStatus)(this, carrier, options));
    }
}
exports.CarrierAccounts = CarrierAccounts;
