// shippo/sdk/sdk.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Shippo = void 0;
const sdks_js_1 = require("../lib/sdks.js");
const addresses_js_1 = require("./addresses.js");
const batches_js_1 = require("./batches.js");
const carrieraccounts_js_1 = require("./carrieraccounts.js");
const carrierparceltemplates_js_1 = require("./carrierparceltemplates.js");
const customsdeclarations_js_1 = require("./customsdeclarations.js");
const customsitems_js_1 = require("./customsitems.js");
const manifests_js_1 = require("./manifests.js");
const orders_js_1 = require("./orders.js");
const parcels_js_1 = require("./parcels.js");
const pickups_js_1 = require("./pickups.js");
const rates_js_1 = require("./rates.js");
const ratesatcheckout_js_1 = require("./ratesatcheckout.js");
const refunds_js_1 = require("./refunds.js");
const servicegroups_js_1 = require("./servicegroups.js");
const shipments_js_1 = require("./shipments.js");
const shippoaccounts_js_1 = require("./shippoaccounts.js");
const trackingstatus_js_1 = require("./trackingstatus.js");
const transactions_js_1 = require("./transactions.js");
const userparceltemplates_js_1 = require("./userparceltemplates.js");
const webhooks_js_1 = require("./webhooks.js");
class Shippo extends sdks_js_1.ClientSDK {
    get addresses() {
        return (this._addresses ?? (this._addresses = new addresses_js_1.Addresses(this._options)));
    }
    get batches() {
        return (this._batches ?? (this._batches = new batches_js_1.Batches(this._options)));
    }
    get carrierAccounts() {
        return (this._carrierAccounts ?? (this._carrierAccounts = new carrieraccounts_js_1.CarrierAccounts(this._options)));
    }
    get customsDeclarations() {
        return (this._customsDeclarations ?? (this._customsDeclarations = new customsdeclarations_js_1.CustomsDeclarations(this._options)));
    }
    get customsItems() {
        return (this._customsItems ?? (this._customsItems = new customsitems_js_1.CustomsItems(this._options)));
    }
    get ratesAtCheckout() {
        return (this._ratesAtCheckout ?? (this._ratesAtCheckout = new ratesatcheckout_js_1.RatesAtCheckout(this._options)));
    }
    get manifests() {
        return (this._manifests ?? (this._manifests = new manifests_js_1.Manifests(this._options)));
    }
    get orders() {
        return (this._orders ?? (this._orders = new orders_js_1.Orders(this._options)));
    }
    get carrierParcelTemplates() {
        return (this._carrierParcelTemplates ?? (this._carrierParcelTemplates = new carrierparceltemplates_js_1.CarrierParcelTemplates(this._options)));
    }
    get parcels() {
        return (this._parcels ?? (this._parcels = new parcels_js_1.Parcels(this._options)));
    }
    get pickups() {
        return (this._pickups ?? (this._pickups = new pickups_js_1.Pickups(this._options)));
    }
    get rates() {
        return (this._rates ?? (this._rates = new rates_js_1.Rates(this._options)));
    }
    get refunds() {
        return (this._refunds ?? (this._refunds = new refunds_js_1.Refunds(this._options)));
    }
    get serviceGroups() {
        return (this._serviceGroups ?? (this._serviceGroups = new servicegroups_js_1.ServiceGroups(this._options)));
    }
    get shipments() {
        return (this._shipments ?? (this._shipments = new shipments_js_1.Shipments(this._options)));
    }
    get trackingStatus() {
        return (this._trackingStatus ?? (this._trackingStatus = new trackingstatus_js_1.TrackingStatus(this._options)));
    }
    get transactions() {
        return (this._transactions ?? (this._transactions = new transactions_js_1.Transactions(this._options)));
    }
    get userParcelTemplates() {
        return (this._userParcelTemplates ?? (this._userParcelTemplates = new userparceltemplates_js_1.UserParcelTemplates(this._options)));
    }
    get shippoAccounts() {
        return (this._shippoAccounts ?? (this._shippoAccounts = new shippoaccounts_js_1.ShippoAccounts(this._options)));
    }
    get webhooks() {
        return (this._webhooks ?? (this._webhooks = new webhooks_js_1.Webhooks(this._options)));
    }
}
exports.Shippo = Shippo;
