// shippo/models/operations/addshipmentstobatch.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.addShipmentsToBatchRequestFromJSON = exports.addShipmentsToBatchRequestToJSON = exports.AddShipmentsToBatchRequest$ = exports.AddShipmentsToBatchRequest$outboundSchema = exports.AddShipmentsToBatchRequest$inboundSchema = exports.addShipmentsToBatchGlobalsFromJSON = exports.addShipmentsToBatchGlobalsToJSON = exports.AddShipmentsToBatchGlobals$ = exports.AddShipmentsToBatchGlobals$outboundSchema = exports.AddShipmentsToBatchGlobals$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.AddShipmentsToBatchGlobals$inboundSchema = z.object({
    "SHIPPO-API-VERSION": z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "SHIPPO-API-VERSION": "shippoApiVersion",
    });
});
/** @internal */
exports.AddShipmentsToBatchGlobals$outboundSchema = z.object({
    shippoApiVersion: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        shippoApiVersion: "SHIPPO-API-VERSION",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AddShipmentsToBatchGlobals$;
(function (AddShipmentsToBatchGlobals$) {
    /** @deprecated use `AddShipmentsToBatchGlobals$inboundSchema` instead. */
    AddShipmentsToBatchGlobals$.inboundSchema = exports.AddShipmentsToBatchGlobals$inboundSchema;
    /** @deprecated use `AddShipmentsToBatchGlobals$outboundSchema` instead. */
    AddShipmentsToBatchGlobals$.outboundSchema = exports.AddShipmentsToBatchGlobals$outboundSchema;
})(AddShipmentsToBatchGlobals$ || (exports.AddShipmentsToBatchGlobals$ = AddShipmentsToBatchGlobals$ = {}));
function addShipmentsToBatchGlobalsToJSON(addShipmentsToBatchGlobals) {
    return JSON.stringify(exports.AddShipmentsToBatchGlobals$outboundSchema.parse(addShipmentsToBatchGlobals));
}
exports.addShipmentsToBatchGlobalsToJSON = addShipmentsToBatchGlobalsToJSON;
function addShipmentsToBatchGlobalsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.AddShipmentsToBatchGlobals$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'AddShipmentsToBatchGlobals' from JSON`);
}
exports.addShipmentsToBatchGlobalsFromJSON = addShipmentsToBatchGlobalsFromJSON;
/** @internal */
exports.AddShipmentsToBatchRequest$inboundSchema = z.object({
    BatchId: z.string(),
    RequestBody: z.array(components.BatchShipmentCreateRequest$inboundSchema),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "BatchId": "batchId",
        "RequestBody": "requestBody",
    });
});
/** @internal */
exports.AddShipmentsToBatchRequest$outboundSchema = z.object({
    batchId: z.string(),
    requestBody: z.array(components.BatchShipmentCreateRequest$outboundSchema),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        batchId: "BatchId",
        requestBody: "RequestBody",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AddShipmentsToBatchRequest$;
(function (AddShipmentsToBatchRequest$) {
    /** @deprecated use `AddShipmentsToBatchRequest$inboundSchema` instead. */
    AddShipmentsToBatchRequest$.inboundSchema = exports.AddShipmentsToBatchRequest$inboundSchema;
    /** @deprecated use `AddShipmentsToBatchRequest$outboundSchema` instead. */
    AddShipmentsToBatchRequest$.outboundSchema = exports.AddShipmentsToBatchRequest$outboundSchema;
})(AddShipmentsToBatchRequest$ || (exports.AddShipmentsToBatchRequest$ = AddShipmentsToBatchRequest$ = {}));
function addShipmentsToBatchRequestToJSON(addShipmentsToBatchRequest) {
    return JSON.stringify(exports.AddShipmentsToBatchRequest$outboundSchema.parse(addShipmentsToBatchRequest));
}
exports.addShipmentsToBatchRequestToJSON = addShipmentsToBatchRequestToJSON;
function addShipmentsToBatchRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.AddShipmentsToBatchRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'AddShipmentsToBatchRequest' from JSON`);
}
exports.addShipmentsToBatchRequestFromJSON = addShipmentsToBatchRequestFromJSON;
