// shippo/models/operations/createtransaction.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.createTransactionRequestBodyFromJSON = exports.createTransactionRequestBodyToJSON = exports.CreateTransactionRequestBody$ = exports.CreateTransactionRequestBody$outboundSchema = exports.CreateTransactionRequestBody$inboundSchema = exports.createTransactionGlobalsFromJSON = exports.createTransactionGlobalsToJSON = exports.CreateTransactionGlobals$ = exports.CreateTransactionGlobals$outboundSchema = exports.CreateTransactionGlobals$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.CreateTransactionGlobals$inboundSchema = z.object({
    "SHIPPO-API-VERSION": z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "SHIPPO-API-VERSION": "shippoApiVersion",
    });
});
/** @internal */
exports.CreateTransactionGlobals$outboundSchema = z.object({
    shippoApiVersion: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        shippoApiVersion: "SHIPPO-API-VERSION",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CreateTransactionGlobals$;
(function (CreateTransactionGlobals$) {
    /** @deprecated use `CreateTransactionGlobals$inboundSchema` instead. */
    CreateTransactionGlobals$.inboundSchema = exports.CreateTransactionGlobals$inboundSchema;
    /** @deprecated use `CreateTransactionGlobals$outboundSchema` instead. */
    CreateTransactionGlobals$.outboundSchema = exports.CreateTransactionGlobals$outboundSchema;
})(CreateTransactionGlobals$ || (exports.CreateTransactionGlobals$ = CreateTransactionGlobals$ = {}));
function createTransactionGlobalsToJSON(createTransactionGlobals) {
    return JSON.stringify(exports.CreateTransactionGlobals$outboundSchema.parse(createTransactionGlobals));
}
exports.createTransactionGlobalsToJSON = createTransactionGlobalsToJSON;
function createTransactionGlobalsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CreateTransactionGlobals$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateTransactionGlobals' from JSON`);
}
exports.createTransactionGlobalsFromJSON = createTransactionGlobalsFromJSON;
/** @internal */
exports.CreateTransactionRequestBody$inboundSchema = z.union([
    components.TransactionCreateRequest$inboundSchema,
    components.InstantTransactionCreateRequest$inboundSchema,
]);
/** @internal */
exports.CreateTransactionRequestBody$outboundSchema = z.union([
    components.TransactionCreateRequest$outboundSchema,
    components.InstantTransactionCreateRequest$outboundSchema,
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CreateTransactionRequestBody$;
(function (CreateTransactionRequestBody$) {
    /** @deprecated use `CreateTransactionRequestBody$inboundSchema` instead. */
    CreateTransactionRequestBody$.inboundSchema = exports.CreateTransactionRequestBody$inboundSchema;
    /** @deprecated use `CreateTransactionRequestBody$outboundSchema` instead. */
    CreateTransactionRequestBody$.outboundSchema = exports.CreateTransactionRequestBody$outboundSchema;
})(CreateTransactionRequestBody$ || (exports.CreateTransactionRequestBody$ = CreateTransactionRequestBody$ = {}));
function createTransactionRequestBodyToJSON(createTransactionRequestBody) {
    return JSON.stringify(exports.CreateTransactionRequestBody$outboundSchema.parse(createTransactionRequestBody));
}
exports.createTransactionRequestBodyToJSON = createTransactionRequestBodyToJSON;
function createTransactionRequestBodyFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CreateTransactionRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateTransactionRequestBody' from JSON`);
}
exports.createTransactionRequestBodyFromJSON = createTransactionRequestBodyFromJSON;
