// shippo/models/operations/createrefund.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.createRefundGlobalsFromJSON = exports.createRefundGlobalsToJSON = exports.CreateRefundGlobals$ = exports.CreateRefundGlobals$outboundSchema = exports.CreateRefundGlobals$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.CreateRefundGlobals$inboundSchema = z.object({
    "SHIPPO-API-VERSION": z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "SHIPPO-API-VERSION": "shippoApiVersion",
    });
});
/** @internal */
exports.CreateRefundGlobals$outboundSchema = z.object({
    shippoApiVersion: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        shippoApiVersion: "SHIPPO-API-VERSION",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CreateRefundGlobals$;
(function (CreateRefundGlobals$) {
    /** @deprecated use `CreateRefundGlobals$inboundSchema` instead. */
    CreateRefundGlobals$.inboundSchema = exports.CreateRefundGlobals$inboundSchema;
    /** @deprecated use `CreateRefundGlobals$outboundSchema` instead. */
    CreateRefundGlobals$.outboundSchema = exports.CreateRefundGlobals$outboundSchema;
})(CreateRefundGlobals$ || (exports.CreateRefundGlobals$ = CreateRefundGlobals$ = {}));
function createRefundGlobalsToJSON(createRefundGlobals) {
    return JSON.stringify(exports.CreateRefundGlobals$outboundSchema.parse(createRefundGlobals));
}
exports.createRefundGlobalsToJSON = createRefundGlobalsToJSON;
function createRefundGlobalsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CreateRefundGlobals$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateRefundGlobals' from JSON`);
}
exports.createRefundGlobalsFromJSON = createRefundGlobalsFromJSON;
