// shippo/models/operations/createparcel.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.createParcelRequestBodyFromJSON = exports.createParcelRequestBodyToJSON = exports.CreateParcelRequestBody$ = exports.CreateParcelRequestBody$outboundSchema = exports.CreateParcelRequestBody$inboundSchema = exports.createParcelGlobalsFromJSON = exports.createParcelGlobalsToJSON = exports.CreateParcelGlobals$ = exports.CreateParcelGlobals$outboundSchema = exports.CreateParcelGlobals$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.CreateParcelGlobals$inboundSchema = z.object({
    "SHIPPO-API-VERSION": z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "SHIPPO-API-VERSION": "shippoApiVersion",
    });
});
/** @internal */
exports.CreateParcelGlobals$outboundSchema = z.object({
    shippoApiVersion: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        shippoApiVersion: "SHIPPO-API-VERSION",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CreateParcelGlobals$;
(function (CreateParcelGlobals$) {
    /** @deprecated use `CreateParcelGlobals$inboundSchema` instead. */
    CreateParcelGlobals$.inboundSchema = exports.CreateParcelGlobals$inboundSchema;
    /** @deprecated use `CreateParcelGlobals$outboundSchema` instead. */
    CreateParcelGlobals$.outboundSchema = exports.CreateParcelGlobals$outboundSchema;
})(CreateParcelGlobals$ || (exports.CreateParcelGlobals$ = CreateParcelGlobals$ = {}));
function createParcelGlobalsToJSON(createParcelGlobals) {
    return JSON.stringify(exports.CreateParcelGlobals$outboundSchema.parse(createParcelGlobals));
}
exports.createParcelGlobalsToJSON = createParcelGlobalsToJSON;
function createParcelGlobalsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CreateParcelGlobals$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateParcelGlobals' from JSON`);
}
exports.createParcelGlobalsFromJSON = createParcelGlobalsFromJSON;
/** @internal */
exports.CreateParcelRequestBody$inboundSchema = z.union([
    components.ParcelCreateFromTemplateRequest$inboundSchema,
    components.ParcelCreateRequest$inboundSchema,
]);
/** @internal */
exports.CreateParcelRequestBody$outboundSchema = z.union([
    components.ParcelCreateFromTemplateRequest$outboundSchema,
    components.ParcelCreateRequest$outboundSchema,
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CreateParcelRequestBody$;
(function (CreateParcelRequestBody$) {
    /** @deprecated use `CreateParcelRequestBody$inboundSchema` instead. */
    CreateParcelRequestBody$.inboundSchema = exports.CreateParcelRequestBody$inboundSchema;
    /** @deprecated use `CreateParcelRequestBody$outboundSchema` instead. */
    CreateParcelRequestBody$.outboundSchema = exports.CreateParcelRequestBody$outboundSchema;
})(CreateParcelRequestBody$ || (exports.CreateParcelRequestBody$ = CreateParcelRequestBody$ = {}));
function createParcelRequestBodyToJSON(createParcelRequestBody) {
    return JSON.stringify(exports.CreateParcelRequestBody$outboundSchema.parse(createParcelRequestBody));
}
exports.createParcelRequestBodyToJSON = createParcelRequestBodyToJSON;
function createParcelRequestBodyFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CreateParcelRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateParcelRequestBody' from JSON`);
}
exports.createParcelRequestBodyFromJSON = createParcelRequestBodyFromJSON;
