// shippo/models/operations/createservicegroup.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.createServiceGroupGlobalsFromJSON = exports.createServiceGroupGlobalsToJSON = exports.CreateServiceGroupGlobals$ = exports.CreateServiceGroupGlobals$outboundSchema = exports.CreateServiceGroupGlobals$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.CreateServiceGroupGlobals$inboundSchema = z.object({
    "SHIPPO-API-VERSION": z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "SHIPPO-API-VERSION": "shippoApiVersion",
    });
});
/** @internal */
exports.CreateServiceGroupGlobals$outboundSchema = z.object({
    shippoApiVersion: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        shippoApiVersion: "SHIPPO-API-VERSION",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CreateServiceGroupGlobals$;
(function (CreateServiceGroupGlobals$) {
    /** @deprecated use `CreateServiceGroupGlobals$inboundSchema` instead. */
    CreateServiceGroupGlobals$.inboundSchema = exports.CreateServiceGroupGlobals$inboundSchema;
    /** @deprecated use `CreateServiceGroupGlobals$outboundSchema` instead. */
    CreateServiceGroupGlobals$.outboundSchema = exports.CreateServiceGroupGlobals$outboundSchema;
})(CreateServiceGroupGlobals$ || (exports.CreateServiceGroupGlobals$ = CreateServiceGroupGlobals$ = {}));
function createServiceGroupGlobalsToJSON(createServiceGroupGlobals) {
    return JSON.stringify(exports.CreateServiceGroupGlobals$outboundSchema.parse(createServiceGroupGlobals));
}
exports.createServiceGroupGlobalsToJSON = createServiceGroupGlobalsToJSON;
function createServiceGroupGlobalsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CreateServiceGroupGlobals$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CreateServiceGroupGlobals' from JSON`);
}
exports.createServiceGroupGlobalsFromJSON = createServiceGroupGlobalsFromJSON;
