// shippo/models/operations/deletedefaultparceltemplate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.deleteDefaultParcelTemplateRequestFromJSON = exports.deleteDefaultParcelTemplateRequestToJSON = exports.DeleteDefaultParcelTemplateRequest$ = exports.DeleteDefaultParcelTemplateRequest$outboundSchema = exports.DeleteDefaultParcelTemplateRequest$inboundSchema = exports.deleteDefaultParcelTemplateGlobalsFromJSON = exports.deleteDefaultParcelTemplateGlobalsToJSON = exports.DeleteDefaultParcelTemplateGlobals$ = exports.DeleteDefaultParcelTemplateGlobals$outboundSchema = exports.DeleteDefaultParcelTemplateGlobals$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.DeleteDefaultParcelTemplateGlobals$inboundSchema = z.object({
    "SHIPPO-API-VERSION": z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "SHIPPO-API-VERSION": "shippoApiVersion",
    });
});
/** @internal */
exports.DeleteDefaultParcelTemplateGlobals$outboundSchema = z.object({
    shippoApiVersion: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        shippoApiVersion: "SHIPPO-API-VERSION",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DeleteDefaultParcelTemplateGlobals$;
(function (DeleteDefaultParcelTemplateGlobals$) {
    /** @deprecated use `DeleteDefaultParcelTemplateGlobals$inboundSchema` instead. */
    DeleteDefaultParcelTemplateGlobals$.inboundSchema = exports.DeleteDefaultParcelTemplateGlobals$inboundSchema;
    /** @deprecated use `DeleteDefaultParcelTemplateGlobals$outboundSchema` instead. */
    DeleteDefaultParcelTemplateGlobals$.outboundSchema = exports.DeleteDefaultParcelTemplateGlobals$outboundSchema;
})(DeleteDefaultParcelTemplateGlobals$ || (exports.DeleteDefaultParcelTemplateGlobals$ = DeleteDefaultParcelTemplateGlobals$ = {}));
function deleteDefaultParcelTemplateGlobalsToJSON(deleteDefaultParcelTemplateGlobals) {
    return JSON.stringify(exports.DeleteDefaultParcelTemplateGlobals$outboundSchema.parse(deleteDefaultParcelTemplateGlobals));
}
exports.deleteDefaultParcelTemplateGlobalsToJSON = deleteDefaultParcelTemplateGlobalsToJSON;
function deleteDefaultParcelTemplateGlobalsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.DeleteDefaultParcelTemplateGlobals$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DeleteDefaultParcelTemplateGlobals' from JSON`);
}
exports.deleteDefaultParcelTemplateGlobalsFromJSON = deleteDefaultParcelTemplateGlobalsFromJSON;
/** @internal */
exports.DeleteDefaultParcelTemplateRequest$inboundSchema = z.object({});
/** @internal */
exports.DeleteDefaultParcelTemplateRequest$outboundSchema = z.object({});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DeleteDefaultParcelTemplateRequest$;
(function (DeleteDefaultParcelTemplateRequest$) {
    /** @deprecated use `DeleteDefaultParcelTemplateRequest$inboundSchema` instead. */
    DeleteDefaultParcelTemplateRequest$.inboundSchema = exports.DeleteDefaultParcelTemplateRequest$inboundSchema;
    /** @deprecated use `DeleteDefaultParcelTemplateRequest$outboundSchema` instead. */
    DeleteDefaultParcelTemplateRequest$.outboundSchema = exports.DeleteDefaultParcelTemplateRequest$outboundSchema;
})(DeleteDefaultParcelTemplateRequest$ || (exports.DeleteDefaultParcelTemplateRequest$ = DeleteDefaultParcelTemplateRequest$ = {}));
function deleteDefaultParcelTemplateRequestToJSON(deleteDefaultParcelTemplateRequest) {
    return JSON.stringify(exports.DeleteDefaultParcelTemplateRequest$outboundSchema.parse(deleteDefaultParcelTemplateRequest));
}
exports.deleteDefaultParcelTemplateRequestToJSON = deleteDefaultParcelTemplateRequestToJSON;
function deleteDefaultParcelTemplateRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.DeleteDefaultParcelTemplateRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DeleteDefaultParcelTemplateRequest' from JSON`);
}
exports.deleteDefaultParcelTemplateRequestFromJSON = deleteDefaultParcelTemplateRequestFromJSON;
