// shippo/models/operations/deleteservicegroup.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.deleteServiceGroupRequestFromJSON = exports.deleteServiceGroupRequestToJSON = exports.DeleteServiceGroupRequest$ = exports.DeleteServiceGroupRequest$outboundSchema = exports.DeleteServiceGroupRequest$inboundSchema = exports.deleteServiceGroupGlobalsFromJSON = exports.deleteServiceGroupGlobalsToJSON = exports.DeleteServiceGroupGlobals$ = exports.DeleteServiceGroupGlobals$outboundSchema = exports.DeleteServiceGroupGlobals$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.DeleteServiceGroupGlobals$inboundSchema = z.object({
    "SHIPPO-API-VERSION": z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "SHIPPO-API-VERSION": "shippoApiVersion",
    });
});
/** @internal */
exports.DeleteServiceGroupGlobals$outboundSchema = z.object({
    shippoApiVersion: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        shippoApiVersion: "SHIPPO-API-VERSION",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DeleteServiceGroupGlobals$;
(function (DeleteServiceGroupGlobals$) {
    /** @deprecated use `DeleteServiceGroupGlobals$inboundSchema` instead. */
    DeleteServiceGroupGlobals$.inboundSchema = exports.DeleteServiceGroupGlobals$inboundSchema;
    /** @deprecated use `DeleteServiceGroupGlobals$outboundSchema` instead. */
    DeleteServiceGroupGlobals$.outboundSchema = exports.DeleteServiceGroupGlobals$outboundSchema;
})(DeleteServiceGroupGlobals$ || (exports.DeleteServiceGroupGlobals$ = DeleteServiceGroupGlobals$ = {}));
function deleteServiceGroupGlobalsToJSON(deleteServiceGroupGlobals) {
    return JSON.stringify(exports.DeleteServiceGroupGlobals$outboundSchema.parse(deleteServiceGroupGlobals));
}
exports.deleteServiceGroupGlobalsToJSON = deleteServiceGroupGlobalsToJSON;
function deleteServiceGroupGlobalsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.DeleteServiceGroupGlobals$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DeleteServiceGroupGlobals' from JSON`);
}
exports.deleteServiceGroupGlobalsFromJSON = deleteServiceGroupGlobalsFromJSON;
/** @internal */
exports.DeleteServiceGroupRequest$inboundSchema = z.object({
    ServiceGroupId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "ServiceGroupId": "serviceGroupId",
    });
});
/** @internal */
exports.DeleteServiceGroupRequest$outboundSchema = z.object({
    serviceGroupId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        serviceGroupId: "ServiceGroupId",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DeleteServiceGroupRequest$;
(function (DeleteServiceGroupRequest$) {
    /** @deprecated use `DeleteServiceGroupRequest$inboundSchema` instead. */
    DeleteServiceGroupRequest$.inboundSchema = exports.DeleteServiceGroupRequest$inboundSchema;
    /** @deprecated use `DeleteServiceGroupRequest$outboundSchema` instead. */
    DeleteServiceGroupRequest$.outboundSchema = exports.DeleteServiceGroupRequest$outboundSchema;
})(DeleteServiceGroupRequest$ || (exports.DeleteServiceGroupRequest$ = DeleteServiceGroupRequest$ = {}));
function deleteServiceGroupRequestToJSON(deleteServiceGroupRequest) {
    return JSON.stringify(exports.DeleteServiceGroupRequest$outboundSchema.parse(deleteServiceGroupRequest));
}
exports.deleteServiceGroupRequestToJSON = deleteServiceGroupRequestToJSON;
function deleteServiceGroupRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.DeleteServiceGroupRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DeleteServiceGroupRequest' from JSON`);
}
exports.deleteServiceGroupRequestFromJSON = deleteServiceGroupRequestFromJSON;
