// shippo/models/operations/getaddress.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getAddressRequestFromJSON = exports.getAddressRequestToJSON = exports.GetAddressRequest$ = exports.GetAddressRequest$outboundSchema = exports.GetAddressRequest$inboundSchema = exports.getAddressGlobalsFromJSON = exports.getAddressGlobalsToJSON = exports.GetAddressGlobals$ = exports.GetAddressGlobals$outboundSchema = exports.GetAddressGlobals$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.GetAddressGlobals$inboundSchema = z.object({
    "SHIPPO-API-VERSION": z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "SHIPPO-API-VERSION": "shippoApiVersion",
    });
});
/** @internal */
exports.GetAddressGlobals$outboundSchema = z.object({
    shippoApiVersion: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        shippoApiVersion: "SHIPPO-API-VERSION",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetAddressGlobals$;
(function (GetAddressGlobals$) {
    /** @deprecated use `GetAddressGlobals$inboundSchema` instead. */
    GetAddressGlobals$.inboundSchema = exports.GetAddressGlobals$inboundSchema;
    /** @deprecated use `GetAddressGlobals$outboundSchema` instead. */
    GetAddressGlobals$.outboundSchema = exports.GetAddressGlobals$outboundSchema;
})(GetAddressGlobals$ || (exports.GetAddressGlobals$ = GetAddressGlobals$ = {}));
function getAddressGlobalsToJSON(getAddressGlobals) {
    return JSON.stringify(exports.GetAddressGlobals$outboundSchema.parse(getAddressGlobals));
}
exports.getAddressGlobalsToJSON = getAddressGlobalsToJSON;
function getAddressGlobalsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetAddressGlobals$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetAddressGlobals' from JSON`);
}
exports.getAddressGlobalsFromJSON = getAddressGlobalsFromJSON;
/** @internal */
exports.GetAddressRequest$inboundSchema = z.object({
    AddressId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "AddressId": "addressId",
    });
});
/** @internal */
exports.GetAddressRequest$outboundSchema = z.object({
    addressId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        addressId: "AddressId",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetAddressRequest$;
(function (GetAddressRequest$) {
    /** @deprecated use `GetAddressRequest$inboundSchema` instead. */
    GetAddressRequest$.inboundSchema = exports.GetAddressRequest$inboundSchema;
    /** @deprecated use `GetAddressRequest$outboundSchema` instead. */
    GetAddressRequest$.outboundSchema = exports.GetAddressRequest$outboundSchema;
})(GetAddressRequest$ || (exports.GetAddressRequest$ = GetAddressRequest$ = {}));
function getAddressRequestToJSON(getAddressRequest) {
    return JSON.stringify(exports.GetAddressRequest$outboundSchema.parse(getAddressRequest));
}
exports.getAddressRequestToJSON = getAddressRequestToJSON;
function getAddressRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetAddressRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetAddressRequest' from JSON`);
}
exports.getAddressRequestFromJSON = getAddressRequestFromJSON;
