// shippo/models/operations/getbatch.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getBatchRequestFromJSON = exports.getBatchRequestToJSON = exports.GetBatchRequest$ = exports.GetBatchRequest$outboundSchema = exports.GetBatchRequest$inboundSchema = exports.getBatchGlobalsFromJSON = exports.getBatchGlobalsToJSON = exports.GetBatchGlobals$ = exports.GetBatchGlobals$outboundSchema = exports.GetBatchGlobals$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.GetBatchGlobals$inboundSchema = z.object({
    "SHIPPO-API-VERSION": z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "SHIPPO-API-VERSION": "shippoApiVersion",
    });
});
/** @internal */
exports.GetBatchGlobals$outboundSchema = z.object({
    shippoApiVersion: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        shippoApiVersion: "SHIPPO-API-VERSION",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetBatchGlobals$;
(function (GetBatchGlobals$) {
    /** @deprecated use `GetBatchGlobals$inboundSchema` instead. */
    GetBatchGlobals$.inboundSchema = exports.GetBatchGlobals$inboundSchema;
    /** @deprecated use `GetBatchGlobals$outboundSchema` instead. */
    GetBatchGlobals$.outboundSchema = exports.GetBatchGlobals$outboundSchema;
})(GetBatchGlobals$ || (exports.GetBatchGlobals$ = GetBatchGlobals$ = {}));
function getBatchGlobalsToJSON(getBatchGlobals) {
    return JSON.stringify(exports.GetBatchGlobals$outboundSchema.parse(getBatchGlobals));
}
exports.getBatchGlobalsToJSON = getBatchGlobalsToJSON;
function getBatchGlobalsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetBatchGlobals$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetBatchGlobals' from JSON`);
}
exports.getBatchGlobalsFromJSON = getBatchGlobalsFromJSON;
/** @internal */
exports.GetBatchRequest$inboundSchema = z.object({
    BatchId: z.string(),
    page: z.number().int().default(1),
    results: z.number().int().default(5),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "BatchId": "batchId",
    });
});
/** @internal */
exports.GetBatchRequest$outboundSchema = z.object({
    batchId: z.string(),
    page: z.number().int().default(1),
    results: z.number().int().default(5),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        batchId: "BatchId",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetBatchRequest$;
(function (GetBatchRequest$) {
    /** @deprecated use `GetBatchRequest$inboundSchema` instead. */
    GetBatchRequest$.inboundSchema = exports.GetBatchRequest$inboundSchema;
    /** @deprecated use `GetBatchRequest$outboundSchema` instead. */
    GetBatchRequest$.outboundSchema = exports.GetBatchRequest$outboundSchema;
})(GetBatchRequest$ || (exports.GetBatchRequest$ = GetBatchRequest$ = {}));
function getBatchRequestToJSON(getBatchRequest) {
    return JSON.stringify(exports.GetBatchRequest$outboundSchema.parse(getBatchRequest));
}
exports.getBatchRequestToJSON = getBatchRequestToJSON;
function getBatchRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetBatchRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetBatchRequest' from JSON`);
}
exports.getBatchRequestFromJSON = getBatchRequestFromJSON;
