// shippo/models/operations/getcarrierregistrationstatus.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getCarrierRegistrationStatusRequestFromJSON = exports.getCarrierRegistrationStatusRequestToJSON = exports.GetCarrierRegistrationStatusRequest$ = exports.GetCarrierRegistrationStatusRequest$outboundSchema = exports.GetCarrierRegistrationStatusRequest$inboundSchema = exports.Carrier$ = exports.Carrier$outboundSchema = exports.Carrier$inboundSchema = exports.getCarrierRegistrationStatusGlobalsFromJSON = exports.getCarrierRegistrationStatusGlobalsToJSON = exports.GetCarrierRegistrationStatusGlobals$ = exports.GetCarrierRegistrationStatusGlobals$outboundSchema = exports.GetCarrierRegistrationStatusGlobals$inboundSchema = exports.Carrier = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/**
 * filter by specific carrier
 */
exports.Carrier = {
    Ups: "ups",
    Usps: "usps",
    CanadaPost: "canada_post",
};
/** @internal */
exports.GetCarrierRegistrationStatusGlobals$inboundSchema = z.object({
    "SHIPPO-API-VERSION": z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "SHIPPO-API-VERSION": "shippoApiVersion",
    });
});
/** @internal */
exports.GetCarrierRegistrationStatusGlobals$outboundSchema = z.object({
    shippoApiVersion: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        shippoApiVersion: "SHIPPO-API-VERSION",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetCarrierRegistrationStatusGlobals$;
(function (GetCarrierRegistrationStatusGlobals$) {
    /** @deprecated use `GetCarrierRegistrationStatusGlobals$inboundSchema` instead. */
    GetCarrierRegistrationStatusGlobals$.inboundSchema = exports.GetCarrierRegistrationStatusGlobals$inboundSchema;
    /** @deprecated use `GetCarrierRegistrationStatusGlobals$outboundSchema` instead. */
    GetCarrierRegistrationStatusGlobals$.outboundSchema = exports.GetCarrierRegistrationStatusGlobals$outboundSchema;
})(GetCarrierRegistrationStatusGlobals$ || (exports.GetCarrierRegistrationStatusGlobals$ = GetCarrierRegistrationStatusGlobals$ = {}));
function getCarrierRegistrationStatusGlobalsToJSON(getCarrierRegistrationStatusGlobals) {
    return JSON.stringify(exports.GetCarrierRegistrationStatusGlobals$outboundSchema.parse(getCarrierRegistrationStatusGlobals));
}
exports.getCarrierRegistrationStatusGlobalsToJSON = getCarrierRegistrationStatusGlobalsToJSON;
function getCarrierRegistrationStatusGlobalsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetCarrierRegistrationStatusGlobals$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetCarrierRegistrationStatusGlobals' from JSON`);
}
exports.getCarrierRegistrationStatusGlobalsFromJSON = getCarrierRegistrationStatusGlobalsFromJSON;
/** @internal */
exports.Carrier$inboundSchema = z
    .nativeEnum(exports.Carrier);
/** @internal */
exports.Carrier$outboundSchema = exports.Carrier$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Carrier$;
(function (Carrier$) {
    /** @deprecated use `Carrier$inboundSchema` instead. */
    Carrier$.inboundSchema = exports.Carrier$inboundSchema;
    /** @deprecated use `Carrier$outboundSchema` instead. */
    Carrier$.outboundSchema = exports.Carrier$outboundSchema;
})(Carrier$ || (exports.Carrier$ = Carrier$ = {}));
/** @internal */
exports.GetCarrierRegistrationStatusRequest$inboundSchema = z.object({
    carrier: exports.Carrier$inboundSchema,
});
/** @internal */
exports.GetCarrierRegistrationStatusRequest$outboundSchema = z.object({
    carrier: exports.Carrier$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetCarrierRegistrationStatusRequest$;
(function (GetCarrierRegistrationStatusRequest$) {
    /** @deprecated use `GetCarrierRegistrationStatusRequest$inboundSchema` instead. */
    GetCarrierRegistrationStatusRequest$.inboundSchema = exports.GetCarrierRegistrationStatusRequest$inboundSchema;
    /** @deprecated use `GetCarrierRegistrationStatusRequest$outboundSchema` instead. */
    GetCarrierRegistrationStatusRequest$.outboundSchema = exports.GetCarrierRegistrationStatusRequest$outboundSchema;
})(GetCarrierRegistrationStatusRequest$ || (exports.GetCarrierRegistrationStatusRequest$ = GetCarrierRegistrationStatusRequest$ = {}));
function getCarrierRegistrationStatusRequestToJSON(getCarrierRegistrationStatusRequest) {
    return JSON.stringify(exports.GetCarrierRegistrationStatusRequest$outboundSchema.parse(getCarrierRegistrationStatusRequest));
}
exports.getCarrierRegistrationStatusRequestToJSON = getCarrierRegistrationStatusRequestToJSON;
function getCarrierRegistrationStatusRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetCarrierRegistrationStatusRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetCarrierRegistrationStatusRequest' from JSON`);
}
exports.getCarrierRegistrationStatusRequestFromJSON = getCarrierRegistrationStatusRequestFromJSON;
