// shippo/models/operations/getcarrieraccount.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getCarrierAccountRequestFromJSON = exports.getCarrierAccountRequestToJSON = exports.GetCarrierAccountRequest$ = exports.GetCarrierAccountRequest$outboundSchema = exports.GetCarrierAccountRequest$inboundSchema = exports.getCarrierAccountGlobalsFromJSON = exports.getCarrierAccountGlobalsToJSON = exports.GetCarrierAccountGlobals$ = exports.GetCarrierAccountGlobals$outboundSchema = exports.GetCarrierAccountGlobals$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.GetCarrierAccountGlobals$inboundSchema = z.object({
    "SHIPPO-API-VERSION": z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "SHIPPO-API-VERSION": "shippoApiVersion",
    });
});
/** @internal */
exports.GetCarrierAccountGlobals$outboundSchema = z.object({
    shippoApiVersion: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        shippoApiVersion: "SHIPPO-API-VERSION",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetCarrierAccountGlobals$;
(function (GetCarrierAccountGlobals$) {
    /** @deprecated use `GetCarrierAccountGlobals$inboundSchema` instead. */
    GetCarrierAccountGlobals$.inboundSchema = exports.GetCarrierAccountGlobals$inboundSchema;
    /** @deprecated use `GetCarrierAccountGlobals$outboundSchema` instead. */
    GetCarrierAccountGlobals$.outboundSchema = exports.GetCarrierAccountGlobals$outboundSchema;
})(GetCarrierAccountGlobals$ || (exports.GetCarrierAccountGlobals$ = GetCarrierAccountGlobals$ = {}));
function getCarrierAccountGlobalsToJSON(getCarrierAccountGlobals) {
    return JSON.stringify(exports.GetCarrierAccountGlobals$outboundSchema.parse(getCarrierAccountGlobals));
}
exports.getCarrierAccountGlobalsToJSON = getCarrierAccountGlobalsToJSON;
function getCarrierAccountGlobalsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetCarrierAccountGlobals$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetCarrierAccountGlobals' from JSON`);
}
exports.getCarrierAccountGlobalsFromJSON = getCarrierAccountGlobalsFromJSON;
/** @internal */
exports.GetCarrierAccountRequest$inboundSchema = z.object({
    CarrierAccountId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "CarrierAccountId": "carrierAccountId",
    });
});
/** @internal */
exports.GetCarrierAccountRequest$outboundSchema = z.object({
    carrierAccountId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        carrierAccountId: "CarrierAccountId",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetCarrierAccountRequest$;
(function (GetCarrierAccountRequest$) {
    /** @deprecated use `GetCarrierAccountRequest$inboundSchema` instead. */
    GetCarrierAccountRequest$.inboundSchema = exports.GetCarrierAccountRequest$inboundSchema;
    /** @deprecated use `GetCarrierAccountRequest$outboundSchema` instead. */
    GetCarrierAccountRequest$.outboundSchema = exports.GetCarrierAccountRequest$outboundSchema;
})(GetCarrierAccountRequest$ || (exports.GetCarrierAccountRequest$ = GetCarrierAccountRequest$ = {}));
function getCarrierAccountRequestToJSON(getCarrierAccountRequest) {
    return JSON.stringify(exports.GetCarrierAccountRequest$outboundSchema.parse(getCarrierAccountRequest));
}
exports.getCarrierAccountRequestToJSON = getCarrierAccountRequestToJSON;
function getCarrierAccountRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetCarrierAccountRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetCarrierAccountRequest' from JSON`);
}
exports.getCarrierAccountRequestFromJSON = getCarrierAccountRequestFromJSON;
