// shippo/models/operations/getcustomsitem.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getCustomsItemRequestFromJSON = exports.getCustomsItemRequestToJSON = exports.GetCustomsItemRequest$ = exports.GetCustomsItemRequest$outboundSchema = exports.GetCustomsItemRequest$inboundSchema = exports.getCustomsItemGlobalsFromJSON = exports.getCustomsItemGlobalsToJSON = exports.GetCustomsItemGlobals$ = exports.GetCustomsItemGlobals$outboundSchema = exports.GetCustomsItemGlobals$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.GetCustomsItemGlobals$inboundSchema = z.object({
    "SHIPPO-API-VERSION": z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "SHIPPO-API-VERSION": "shippoApiVersion",
    });
});
/** @internal */
exports.GetCustomsItemGlobals$outboundSchema = z.object({
    shippoApiVersion: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        shippoApiVersion: "SHIPPO-API-VERSION",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetCustomsItemGlobals$;
(function (GetCustomsItemGlobals$) {
    /** @deprecated use `GetCustomsItemGlobals$inboundSchema` instead. */
    GetCustomsItemGlobals$.inboundSchema = exports.GetCustomsItemGlobals$inboundSchema;
    /** @deprecated use `GetCustomsItemGlobals$outboundSchema` instead. */
    GetCustomsItemGlobals$.outboundSchema = exports.GetCustomsItemGlobals$outboundSchema;
})(GetCustomsItemGlobals$ || (exports.GetCustomsItemGlobals$ = GetCustomsItemGlobals$ = {}));
function getCustomsItemGlobalsToJSON(getCustomsItemGlobals) {
    return JSON.stringify(exports.GetCustomsItemGlobals$outboundSchema.parse(getCustomsItemGlobals));
}
exports.getCustomsItemGlobalsToJSON = getCustomsItemGlobalsToJSON;
function getCustomsItemGlobalsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetCustomsItemGlobals$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetCustomsItemGlobals' from JSON`);
}
exports.getCustomsItemGlobalsFromJSON = getCustomsItemGlobalsFromJSON;
/** @internal */
exports.GetCustomsItemRequest$inboundSchema = z.object({
    CustomsItemId: z.string(),
    page: z.number().int().default(1),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "CustomsItemId": "customsItemId",
    });
});
/** @internal */
exports.GetCustomsItemRequest$outboundSchema = z.object({
    customsItemId: z.string(),
    page: z.number().int().default(1),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        customsItemId: "CustomsItemId",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetCustomsItemRequest$;
(function (GetCustomsItemRequest$) {
    /** @deprecated use `GetCustomsItemRequest$inboundSchema` instead. */
    GetCustomsItemRequest$.inboundSchema = exports.GetCustomsItemRequest$inboundSchema;
    /** @deprecated use `GetCustomsItemRequest$outboundSchema` instead. */
    GetCustomsItemRequest$.outboundSchema = exports.GetCustomsItemRequest$outboundSchema;
})(GetCustomsItemRequest$ || (exports.GetCustomsItemRequest$ = GetCustomsItemRequest$ = {}));
function getCustomsItemRequestToJSON(getCustomsItemRequest) {
    return JSON.stringify(exports.GetCustomsItemRequest$outboundSchema.parse(getCustomsItemRequest));
}
exports.getCustomsItemRequestToJSON = getCustomsItemRequestToJSON;
function getCustomsItemRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetCustomsItemRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetCustomsItemRequest' from JSON`);
}
exports.getCustomsItemRequestFromJSON = getCustomsItemRequestFromJSON;
