// shippo/models/operations/getcarrierparceltemplate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getCarrierParcelTemplateRequestFromJSON = exports.getCarrierParcelTemplateRequestToJSON = exports.GetCarrierParcelTemplateRequest$ = exports.GetCarrierParcelTemplateRequest$outboundSchema = exports.GetCarrierParcelTemplateRequest$inboundSchema = exports.getCarrierParcelTemplateGlobalsFromJSON = exports.getCarrierParcelTemplateGlobalsToJSON = exports.GetCarrierParcelTemplateGlobals$ = exports.GetCarrierParcelTemplateGlobals$outboundSchema = exports.GetCarrierParcelTemplateGlobals$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.GetCarrierParcelTemplateGlobals$inboundSchema = z.object({
    "SHIPPO-API-VERSION": z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "SHIPPO-API-VERSION": "shippoApiVersion",
    });
});
/** @internal */
exports.GetCarrierParcelTemplateGlobals$outboundSchema = z.object({
    shippoApiVersion: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        shippoApiVersion: "SHIPPO-API-VERSION",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetCarrierParcelTemplateGlobals$;
(function (GetCarrierParcelTemplateGlobals$) {
    /** @deprecated use `GetCarrierParcelTemplateGlobals$inboundSchema` instead. */
    GetCarrierParcelTemplateGlobals$.inboundSchema = exports.GetCarrierParcelTemplateGlobals$inboundSchema;
    /** @deprecated use `GetCarrierParcelTemplateGlobals$outboundSchema` instead. */
    GetCarrierParcelTemplateGlobals$.outboundSchema = exports.GetCarrierParcelTemplateGlobals$outboundSchema;
})(GetCarrierParcelTemplateGlobals$ || (exports.GetCarrierParcelTemplateGlobals$ = GetCarrierParcelTemplateGlobals$ = {}));
function getCarrierParcelTemplateGlobalsToJSON(getCarrierParcelTemplateGlobals) {
    return JSON.stringify(exports.GetCarrierParcelTemplateGlobals$outboundSchema.parse(getCarrierParcelTemplateGlobals));
}
exports.getCarrierParcelTemplateGlobalsToJSON = getCarrierParcelTemplateGlobalsToJSON;
function getCarrierParcelTemplateGlobalsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetCarrierParcelTemplateGlobals$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetCarrierParcelTemplateGlobals' from JSON`);
}
exports.getCarrierParcelTemplateGlobalsFromJSON = getCarrierParcelTemplateGlobalsFromJSON;
/** @internal */
exports.GetCarrierParcelTemplateRequest$inboundSchema = z.object({
    CarrierParcelTemplateToken: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "CarrierParcelTemplateToken": "carrierParcelTemplateToken",
    });
});
/** @internal */
exports.GetCarrierParcelTemplateRequest$outboundSchema = z.object({
    carrierParcelTemplateToken: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        carrierParcelTemplateToken: "CarrierParcelTemplateToken",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetCarrierParcelTemplateRequest$;
(function (GetCarrierParcelTemplateRequest$) {
    /** @deprecated use `GetCarrierParcelTemplateRequest$inboundSchema` instead. */
    GetCarrierParcelTemplateRequest$.inboundSchema = exports.GetCarrierParcelTemplateRequest$inboundSchema;
    /** @deprecated use `GetCarrierParcelTemplateRequest$outboundSchema` instead. */
    GetCarrierParcelTemplateRequest$.outboundSchema = exports.GetCarrierParcelTemplateRequest$outboundSchema;
})(GetCarrierParcelTemplateRequest$ || (exports.GetCarrierParcelTemplateRequest$ = GetCarrierParcelTemplateRequest$ = {}));
function getCarrierParcelTemplateRequestToJSON(getCarrierParcelTemplateRequest) {
    return JSON.stringify(exports.GetCarrierParcelTemplateRequest$outboundSchema.parse(getCarrierParcelTemplateRequest));
}
exports.getCarrierParcelTemplateRequestToJSON = getCarrierParcelTemplateRequestToJSON;
function getCarrierParcelTemplateRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetCarrierParcelTemplateRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetCarrierParcelTemplateRequest' from JSON`);
}
exports.getCarrierParcelTemplateRequestFromJSON = getCarrierParcelTemplateRequestFromJSON;
