// shippo/models/operations/getcustomsdeclaration.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getCustomsDeclarationRequestFromJSON = exports.getCustomsDeclarationRequestToJSON = exports.GetCustomsDeclarationRequest$ = exports.GetCustomsDeclarationRequest$outboundSchema = exports.GetCustomsDeclarationRequest$inboundSchema = exports.getCustomsDeclarationGlobalsFromJSON = exports.getCustomsDeclarationGlobalsToJSON = exports.GetCustomsDeclarationGlobals$ = exports.GetCustomsDeclarationGlobals$outboundSchema = exports.GetCustomsDeclarationGlobals$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.GetCustomsDeclarationGlobals$inboundSchema = z.object({
    "SHIPPO-API-VERSION": z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "SHIPPO-API-VERSION": "shippoApiVersion",
    });
});
/** @internal */
exports.GetCustomsDeclarationGlobals$outboundSchema = z.object({
    shippoApiVersion: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        shippoApiVersion: "SHIPPO-API-VERSION",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetCustomsDeclarationGlobals$;
(function (GetCustomsDeclarationGlobals$) {
    /** @deprecated use `GetCustomsDeclarationGlobals$inboundSchema` instead. */
    GetCustomsDeclarationGlobals$.inboundSchema = exports.GetCustomsDeclarationGlobals$inboundSchema;
    /** @deprecated use `GetCustomsDeclarationGlobals$outboundSchema` instead. */
    GetCustomsDeclarationGlobals$.outboundSchema = exports.GetCustomsDeclarationGlobals$outboundSchema;
})(GetCustomsDeclarationGlobals$ || (exports.GetCustomsDeclarationGlobals$ = GetCustomsDeclarationGlobals$ = {}));
function getCustomsDeclarationGlobalsToJSON(getCustomsDeclarationGlobals) {
    return JSON.stringify(exports.GetCustomsDeclarationGlobals$outboundSchema.parse(getCustomsDeclarationGlobals));
}
exports.getCustomsDeclarationGlobalsToJSON = getCustomsDeclarationGlobalsToJSON;
function getCustomsDeclarationGlobalsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetCustomsDeclarationGlobals$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetCustomsDeclarationGlobals' from JSON`);
}
exports.getCustomsDeclarationGlobalsFromJSON = getCustomsDeclarationGlobalsFromJSON;
/** @internal */
exports.GetCustomsDeclarationRequest$inboundSchema = z.object({
    CustomsDeclarationId: z.string(),
    page: z.number().int().default(1),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "CustomsDeclarationId": "customsDeclarationId",
    });
});
/** @internal */
exports.GetCustomsDeclarationRequest$outboundSchema = z.object({
    customsDeclarationId: z.string(),
    page: z.number().int().default(1),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        customsDeclarationId: "CustomsDeclarationId",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetCustomsDeclarationRequest$;
(function (GetCustomsDeclarationRequest$) {
    /** @deprecated use `GetCustomsDeclarationRequest$inboundSchema` instead. */
    GetCustomsDeclarationRequest$.inboundSchema = exports.GetCustomsDeclarationRequest$inboundSchema;
    /** @deprecated use `GetCustomsDeclarationRequest$outboundSchema` instead. */
    GetCustomsDeclarationRequest$.outboundSchema = exports.GetCustomsDeclarationRequest$outboundSchema;
})(GetCustomsDeclarationRequest$ || (exports.GetCustomsDeclarationRequest$ = GetCustomsDeclarationRequest$ = {}));
function getCustomsDeclarationRequestToJSON(getCustomsDeclarationRequest) {
    return JSON.stringify(exports.GetCustomsDeclarationRequest$outboundSchema.parse(getCustomsDeclarationRequest));
}
exports.getCustomsDeclarationRequestToJSON = getCustomsDeclarationRequestToJSON;
function getCustomsDeclarationRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetCustomsDeclarationRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetCustomsDeclarationRequest' from JSON`);
}
exports.getCustomsDeclarationRequestFromJSON = getCustomsDeclarationRequestFromJSON;
