// shippo/models/operations/getdefaultparceltemplate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getDefaultParcelTemplateRequestFromJSON = exports.getDefaultParcelTemplateRequestToJSON = exports.GetDefaultParcelTemplateRequest$ = exports.GetDefaultParcelTemplateRequest$outboundSchema = exports.GetDefaultParcelTemplateRequest$inboundSchema = exports.getDefaultParcelTemplateGlobalsFromJSON = exports.getDefaultParcelTemplateGlobalsToJSON = exports.GetDefaultParcelTemplateGlobals$ = exports.GetDefaultParcelTemplateGlobals$outboundSchema = exports.GetDefaultParcelTemplateGlobals$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.GetDefaultParcelTemplateGlobals$inboundSchema = z.object({
    "SHIPPO-API-VERSION": z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "SHIPPO-API-VERSION": "shippoApiVersion",
    });
});
/** @internal */
exports.GetDefaultParcelTemplateGlobals$outboundSchema = z.object({
    shippoApiVersion: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        shippoApiVersion: "SHIPPO-API-VERSION",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetDefaultParcelTemplateGlobals$;
(function (GetDefaultParcelTemplateGlobals$) {
    /** @deprecated use `GetDefaultParcelTemplateGlobals$inboundSchema` instead. */
    GetDefaultParcelTemplateGlobals$.inboundSchema = exports.GetDefaultParcelTemplateGlobals$inboundSchema;
    /** @deprecated use `GetDefaultParcelTemplateGlobals$outboundSchema` instead. */
    GetDefaultParcelTemplateGlobals$.outboundSchema = exports.GetDefaultParcelTemplateGlobals$outboundSchema;
})(GetDefaultParcelTemplateGlobals$ || (exports.GetDefaultParcelTemplateGlobals$ = GetDefaultParcelTemplateGlobals$ = {}));
function getDefaultParcelTemplateGlobalsToJSON(getDefaultParcelTemplateGlobals) {
    return JSON.stringify(exports.GetDefaultParcelTemplateGlobals$outboundSchema.parse(getDefaultParcelTemplateGlobals));
}
exports.getDefaultParcelTemplateGlobalsToJSON = getDefaultParcelTemplateGlobalsToJSON;
function getDefaultParcelTemplateGlobalsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetDefaultParcelTemplateGlobals$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetDefaultParcelTemplateGlobals' from JSON`);
}
exports.getDefaultParcelTemplateGlobalsFromJSON = getDefaultParcelTemplateGlobalsFromJSON;
/** @internal */
exports.GetDefaultParcelTemplateRequest$inboundSchema = z.object({});
/** @internal */
exports.GetDefaultParcelTemplateRequest$outboundSchema = z.object({});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetDefaultParcelTemplateRequest$;
(function (GetDefaultParcelTemplateRequest$) {
    /** @deprecated use `GetDefaultParcelTemplateRequest$inboundSchema` instead. */
    GetDefaultParcelTemplateRequest$.inboundSchema = exports.GetDefaultParcelTemplateRequest$inboundSchema;
    /** @deprecated use `GetDefaultParcelTemplateRequest$outboundSchema` instead. */
    GetDefaultParcelTemplateRequest$.outboundSchema = exports.GetDefaultParcelTemplateRequest$outboundSchema;
})(GetDefaultParcelTemplateRequest$ || (exports.GetDefaultParcelTemplateRequest$ = GetDefaultParcelTemplateRequest$ = {}));
function getDefaultParcelTemplateRequestToJSON(getDefaultParcelTemplateRequest) {
    return JSON.stringify(exports.GetDefaultParcelTemplateRequest$outboundSchema.parse(getDefaultParcelTemplateRequest));
}
exports.getDefaultParcelTemplateRequestToJSON = getDefaultParcelTemplateRequestToJSON;
function getDefaultParcelTemplateRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetDefaultParcelTemplateRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetDefaultParcelTemplateRequest' from JSON`);
}
exports.getDefaultParcelTemplateRequestFromJSON = getDefaultParcelTemplateRequestFromJSON;
