// shippo/models/operations/getrate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getRateRequestFromJSON = exports.getRateRequestToJSON = exports.GetRateRequest$ = exports.GetRateRequest$outboundSchema = exports.GetRateRequest$inboundSchema = exports.getRateGlobalsFromJSON = exports.getRateGlobalsToJSON = exports.GetRateGlobals$ = exports.GetRateGlobals$outboundSchema = exports.GetRateGlobals$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.GetRateGlobals$inboundSchema = z.object({
    "SHIPPO-API-VERSION": z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "SHIPPO-API-VERSION": "shippoApiVersion",
    });
});
/** @internal */
exports.GetRateGlobals$outboundSchema = z.object({
    shippoApiVersion: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        shippoApiVersion: "SHIPPO-API-VERSION",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetRateGlobals$;
(function (GetRateGlobals$) {
    /** @deprecated use `GetRateGlobals$inboundSchema` instead. */
    GetRateGlobals$.inboundSchema = exports.GetRateGlobals$inboundSchema;
    /** @deprecated use `GetRateGlobals$outboundSchema` instead. */
    GetRateGlobals$.outboundSchema = exports.GetRateGlobals$outboundSchema;
})(GetRateGlobals$ || (exports.GetRateGlobals$ = GetRateGlobals$ = {}));
function getRateGlobalsToJSON(getRateGlobals) {
    return JSON.stringify(exports.GetRateGlobals$outboundSchema.parse(getRateGlobals));
}
exports.getRateGlobalsToJSON = getRateGlobalsToJSON;
function getRateGlobalsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetRateGlobals$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetRateGlobals' from JSON`);
}
exports.getRateGlobalsFromJSON = getRateGlobalsFromJSON;
/** @internal */
exports.GetRateRequest$inboundSchema = z.object({
    RateId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "RateId": "rateId",
    });
});
/** @internal */
exports.GetRateRequest$outboundSchema = z.object({
    rateId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        rateId: "RateId",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetRateRequest$;
(function (GetRateRequest$) {
    /** @deprecated use `GetRateRequest$inboundSchema` instead. */
    GetRateRequest$.inboundSchema = exports.GetRateRequest$inboundSchema;
    /** @deprecated use `GetRateRequest$outboundSchema` instead. */
    GetRateRequest$.outboundSchema = exports.GetRateRequest$outboundSchema;
})(GetRateRequest$ || (exports.GetRateRequest$ = GetRateRequest$ = {}));
function getRateRequestToJSON(getRateRequest) {
    return JSON.stringify(exports.GetRateRequest$outboundSchema.parse(getRateRequest));
}
exports.getRateRequestToJSON = getRateRequestToJSON;
function getRateRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetRateRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetRateRequest' from JSON`);
}
exports.getRateRequestFromJSON = getRateRequestFromJSON;
