// shippo/models/operations/getrefund.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getRefundRequestFromJSON = exports.getRefundRequestToJSON = exports.GetRefundRequest$ = exports.GetRefundRequest$outboundSchema = exports.GetRefundRequest$inboundSchema = exports.getRefundGlobalsFromJSON = exports.getRefundGlobalsToJSON = exports.GetRefundGlobals$ = exports.GetRefundGlobals$outboundSchema = exports.GetRefundGlobals$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.GetRefundGlobals$inboundSchema = z.object({
    "SHIPPO-API-VERSION": z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "SHIPPO-API-VERSION": "shippoApiVersion",
    });
});
/** @internal */
exports.GetRefundGlobals$outboundSchema = z.object({
    shippoApiVersion: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        shippoApiVersion: "SHIPPO-API-VERSION",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetRefundGlobals$;
(function (GetRefundGlobals$) {
    /** @deprecated use `GetRefundGlobals$inboundSchema` instead. */
    GetRefundGlobals$.inboundSchema = exports.GetRefundGlobals$inboundSchema;
    /** @deprecated use `GetRefundGlobals$outboundSchema` instead. */
    GetRefundGlobals$.outboundSchema = exports.GetRefundGlobals$outboundSchema;
})(GetRefundGlobals$ || (exports.GetRefundGlobals$ = GetRefundGlobals$ = {}));
function getRefundGlobalsToJSON(getRefundGlobals) {
    return JSON.stringify(exports.GetRefundGlobals$outboundSchema.parse(getRefundGlobals));
}
exports.getRefundGlobalsToJSON = getRefundGlobalsToJSON;
function getRefundGlobalsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetRefundGlobals$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetRefundGlobals' from JSON`);
}
exports.getRefundGlobalsFromJSON = getRefundGlobalsFromJSON;
/** @internal */
exports.GetRefundRequest$inboundSchema = z.object({
    RefundId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "RefundId": "refundId",
    });
});
/** @internal */
exports.GetRefundRequest$outboundSchema = z.object({
    refundId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        refundId: "RefundId",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetRefundRequest$;
(function (GetRefundRequest$) {
    /** @deprecated use `GetRefundRequest$inboundSchema` instead. */
    GetRefundRequest$.inboundSchema = exports.GetRefundRequest$inboundSchema;
    /** @deprecated use `GetRefundRequest$outboundSchema` instead. */
    GetRefundRequest$.outboundSchema = exports.GetRefundRequest$outboundSchema;
})(GetRefundRequest$ || (exports.GetRefundRequest$ = GetRefundRequest$ = {}));
function getRefundRequestToJSON(getRefundRequest) {
    return JSON.stringify(exports.GetRefundRequest$outboundSchema.parse(getRefundRequest));
}
exports.getRefundRequestToJSON = getRefundRequestToJSON;
function getRefundRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetRefundRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetRefundRequest' from JSON`);
}
exports.getRefundRequestFromJSON = getRefundRequestFromJSON;
