// shippo/models/operations/getshippoaccount.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getShippoAccountRequestFromJSON = exports.getShippoAccountRequestToJSON = exports.GetShippoAccountRequest$ = exports.GetShippoAccountRequest$outboundSchema = exports.GetShippoAccountRequest$inboundSchema = exports.getShippoAccountGlobalsFromJSON = exports.getShippoAccountGlobalsToJSON = exports.GetShippoAccountGlobals$ = exports.GetShippoAccountGlobals$outboundSchema = exports.GetShippoAccountGlobals$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.GetShippoAccountGlobals$inboundSchema = z.object({
    "SHIPPO-API-VERSION": z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "SHIPPO-API-VERSION": "shippoApiVersion",
    });
});
/** @internal */
exports.GetShippoAccountGlobals$outboundSchema = z.object({
    shippoApiVersion: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        shippoApiVersion: "SHIPPO-API-VERSION",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetShippoAccountGlobals$;
(function (GetShippoAccountGlobals$) {
    /** @deprecated use `GetShippoAccountGlobals$inboundSchema` instead. */
    GetShippoAccountGlobals$.inboundSchema = exports.GetShippoAccountGlobals$inboundSchema;
    /** @deprecated use `GetShippoAccountGlobals$outboundSchema` instead. */
    GetShippoAccountGlobals$.outboundSchema = exports.GetShippoAccountGlobals$outboundSchema;
})(GetShippoAccountGlobals$ || (exports.GetShippoAccountGlobals$ = GetShippoAccountGlobals$ = {}));
function getShippoAccountGlobalsToJSON(getShippoAccountGlobals) {
    return JSON.stringify(exports.GetShippoAccountGlobals$outboundSchema.parse(getShippoAccountGlobals));
}
exports.getShippoAccountGlobalsToJSON = getShippoAccountGlobalsToJSON;
function getShippoAccountGlobalsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetShippoAccountGlobals$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetShippoAccountGlobals' from JSON`);
}
exports.getShippoAccountGlobalsFromJSON = getShippoAccountGlobalsFromJSON;
/** @internal */
exports.GetShippoAccountRequest$inboundSchema = z.object({
    ShippoAccountId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "ShippoAccountId": "shippoAccountId",
    });
});
/** @internal */
exports.GetShippoAccountRequest$outboundSchema = z.object({
    shippoAccountId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        shippoAccountId: "ShippoAccountId",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetShippoAccountRequest$;
(function (GetShippoAccountRequest$) {
    /** @deprecated use `GetShippoAccountRequest$inboundSchema` instead. */
    GetShippoAccountRequest$.inboundSchema = exports.GetShippoAccountRequest$inboundSchema;
    /** @deprecated use `GetShippoAccountRequest$outboundSchema` instead. */
    GetShippoAccountRequest$.outboundSchema = exports.GetShippoAccountRequest$outboundSchema;
})(GetShippoAccountRequest$ || (exports.GetShippoAccountRequest$ = GetShippoAccountRequest$ = {}));
function getShippoAccountRequestToJSON(getShippoAccountRequest) {
    return JSON.stringify(exports.GetShippoAccountRequest$outboundSchema.parse(getShippoAccountRequest));
}
exports.getShippoAccountRequestToJSON = getShippoAccountRequestToJSON;
function getShippoAccountRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetShippoAccountRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetShippoAccountRequest' from JSON`);
}
exports.getShippoAccountRequestFromJSON = getShippoAccountRequestFromJSON;
