// shippo/models/operations/gettransaction.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getTransactionRequestFromJSON = exports.getTransactionRequestToJSON = exports.GetTransactionRequest$ = exports.GetTransactionRequest$outboundSchema = exports.GetTransactionRequest$inboundSchema = exports.getTransactionGlobalsFromJSON = exports.getTransactionGlobalsToJSON = exports.GetTransactionGlobals$ = exports.GetTransactionGlobals$outboundSchema = exports.GetTransactionGlobals$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.GetTransactionGlobals$inboundSchema = z.object({
    "SHIPPO-API-VERSION": z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "SHIPPO-API-VERSION": "shippoApiVersion",
    });
});
/** @internal */
exports.GetTransactionGlobals$outboundSchema = z.object({
    shippoApiVersion: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        shippoApiVersion: "SHIPPO-API-VERSION",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetTransactionGlobals$;
(function (GetTransactionGlobals$) {
    /** @deprecated use `GetTransactionGlobals$inboundSchema` instead. */
    GetTransactionGlobals$.inboundSchema = exports.GetTransactionGlobals$inboundSchema;
    /** @deprecated use `GetTransactionGlobals$outboundSchema` instead. */
    GetTransactionGlobals$.outboundSchema = exports.GetTransactionGlobals$outboundSchema;
})(GetTransactionGlobals$ || (exports.GetTransactionGlobals$ = GetTransactionGlobals$ = {}));
function getTransactionGlobalsToJSON(getTransactionGlobals) {
    return JSON.stringify(exports.GetTransactionGlobals$outboundSchema.parse(getTransactionGlobals));
}
exports.getTransactionGlobalsToJSON = getTransactionGlobalsToJSON;
function getTransactionGlobalsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetTransactionGlobals$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetTransactionGlobals' from JSON`);
}
exports.getTransactionGlobalsFromJSON = getTransactionGlobalsFromJSON;
/** @internal */
exports.GetTransactionRequest$inboundSchema = z.object({
    TransactionId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "TransactionId": "transactionId",
    });
});
/** @internal */
exports.GetTransactionRequest$outboundSchema = z.object({
    transactionId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        transactionId: "TransactionId",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetTransactionRequest$;
(function (GetTransactionRequest$) {
    /** @deprecated use `GetTransactionRequest$inboundSchema` instead. */
    GetTransactionRequest$.inboundSchema = exports.GetTransactionRequest$inboundSchema;
    /** @deprecated use `GetTransactionRequest$outboundSchema` instead. */
    GetTransactionRequest$.outboundSchema = exports.GetTransactionRequest$outboundSchema;
})(GetTransactionRequest$ || (exports.GetTransactionRequest$ = GetTransactionRequest$ = {}));
function getTransactionRequestToJSON(getTransactionRequest) {
    return JSON.stringify(exports.GetTransactionRequest$outboundSchema.parse(getTransactionRequest));
}
exports.getTransactionRequestToJSON = getTransactionRequestToJSON;
function getTransactionRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetTransactionRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetTransactionRequest' from JSON`);
}
exports.getTransactionRequestFromJSON = getTransactionRequestFromJSON;
