// shippo/models/operations/gettrack.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getTrackRequestFromJSON = exports.getTrackRequestToJSON = exports.GetTrackRequest$ = exports.GetTrackRequest$outboundSchema = exports.GetTrackRequest$inboundSchema = exports.getTrackGlobalsFromJSON = exports.getTrackGlobalsToJSON = exports.GetTrackGlobals$ = exports.GetTrackGlobals$outboundSchema = exports.GetTrackGlobals$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.GetTrackGlobals$inboundSchema = z.object({
    "SHIPPO-API-VERSION": z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "SHIPPO-API-VERSION": "shippoApiVersion",
    });
});
/** @internal */
exports.GetTrackGlobals$outboundSchema = z.object({
    shippoApiVersion: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        shippoApiVersion: "SHIPPO-API-VERSION",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetTrackGlobals$;
(function (GetTrackGlobals$) {
    /** @deprecated use `GetTrackGlobals$inboundSchema` instead. */
    GetTrackGlobals$.inboundSchema = exports.GetTrackGlobals$inboundSchema;
    /** @deprecated use `GetTrackGlobals$outboundSchema` instead. */
    GetTrackGlobals$.outboundSchema = exports.GetTrackGlobals$outboundSchema;
})(GetTrackGlobals$ || (exports.GetTrackGlobals$ = GetTrackGlobals$ = {}));
function getTrackGlobalsToJSON(getTrackGlobals) {
    return JSON.stringify(exports.GetTrackGlobals$outboundSchema.parse(getTrackGlobals));
}
exports.getTrackGlobalsToJSON = getTrackGlobalsToJSON;
function getTrackGlobalsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetTrackGlobals$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetTrackGlobals' from JSON`);
}
exports.getTrackGlobalsFromJSON = getTrackGlobalsFromJSON;
/** @internal */
exports.GetTrackRequest$inboundSchema = z.object({
    TrackingNumber: z.string(),
    Carrier: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "TrackingNumber": "trackingNumber",
        "Carrier": "carrier",
    });
});
/** @internal */
exports.GetTrackRequest$outboundSchema = z.object({
    trackingNumber: z.string(),
    carrier: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        trackingNumber: "TrackingNumber",
        carrier: "Carrier",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetTrackRequest$;
(function (GetTrackRequest$) {
    /** @deprecated use `GetTrackRequest$inboundSchema` instead. */
    GetTrackRequest$.inboundSchema = exports.GetTrackRequest$inboundSchema;
    /** @deprecated use `GetTrackRequest$outboundSchema` instead. */
    GetTrackRequest$.outboundSchema = exports.GetTrackRequest$outboundSchema;
})(GetTrackRequest$ || (exports.GetTrackRequest$ = GetTrackRequest$ = {}));
function getTrackRequestToJSON(getTrackRequest) {
    return JSON.stringify(exports.GetTrackRequest$outboundSchema.parse(getTrackRequest));
}
exports.getTrackRequestToJSON = getTrackRequestToJSON;
function getTrackRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.GetTrackRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetTrackRequest' from JSON`);
}
exports.getTrackRequestFromJSON = getTrackRequestFromJSON;
