// shippo/models/operations/listshippoaccounts.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.listShippoAccountsRequestFromJSON = exports.listShippoAccountsRequestToJSON = exports.ListShippoAccountsRequest$ = exports.ListShippoAccountsRequest$outboundSchema = exports.ListShippoAccountsRequest$inboundSchema = exports.listShippoAccountsGlobalsFromJSON = exports.listShippoAccountsGlobalsToJSON = exports.ListShippoAccountsGlobals$ = exports.ListShippoAccountsGlobals$outboundSchema = exports.ListShippoAccountsGlobals$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.ListShippoAccountsGlobals$inboundSchema = z.object({
    "SHIPPO-API-VERSION": z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "SHIPPO-API-VERSION": "shippoApiVersion",
    });
});
/** @internal */
exports.ListShippoAccountsGlobals$outboundSchema = z.object({
    shippoApiVersion: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        shippoApiVersion: "SHIPPO-API-VERSION",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ListShippoAccountsGlobals$;
(function (ListShippoAccountsGlobals$) {
    /** @deprecated use `ListShippoAccountsGlobals$inboundSchema` instead. */
    ListShippoAccountsGlobals$.inboundSchema = exports.ListShippoAccountsGlobals$inboundSchema;
    /** @deprecated use `ListShippoAccountsGlobals$outboundSchema` instead. */
    ListShippoAccountsGlobals$.outboundSchema = exports.ListShippoAccountsGlobals$outboundSchema;
})(ListShippoAccountsGlobals$ || (exports.ListShippoAccountsGlobals$ = ListShippoAccountsGlobals$ = {}));
function listShippoAccountsGlobalsToJSON(listShippoAccountsGlobals) {
    return JSON.stringify(exports.ListShippoAccountsGlobals$outboundSchema.parse(listShippoAccountsGlobals));
}
exports.listShippoAccountsGlobalsToJSON = listShippoAccountsGlobalsToJSON;
function listShippoAccountsGlobalsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ListShippoAccountsGlobals$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListShippoAccountsGlobals' from JSON`);
}
exports.listShippoAccountsGlobalsFromJSON = listShippoAccountsGlobalsFromJSON;
/** @internal */
exports.ListShippoAccountsRequest$inboundSchema = z.object({
    page: z.number().int().default(1),
    results: z.number().int().default(25),
});
/** @internal */
exports.ListShippoAccountsRequest$outboundSchema = z.object({
    page: z.number().int().default(1),
    results: z.number().int().default(25),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ListShippoAccountsRequest$;
(function (ListShippoAccountsRequest$) {
    /** @deprecated use `ListShippoAccountsRequest$inboundSchema` instead. */
    ListShippoAccountsRequest$.inboundSchema = exports.ListShippoAccountsRequest$inboundSchema;
    /** @deprecated use `ListShippoAccountsRequest$outboundSchema` instead. */
    ListShippoAccountsRequest$.outboundSchema = exports.ListShippoAccountsRequest$outboundSchema;
})(ListShippoAccountsRequest$ || (exports.ListShippoAccountsRequest$ = ListShippoAccountsRequest$ = {}));
function listShippoAccountsRequestToJSON(listShippoAccountsRequest) {
    return JSON.stringify(exports.ListShippoAccountsRequest$outboundSchema.parse(listShippoAccountsRequest));
}
exports.listShippoAccountsRequestToJSON = listShippoAccountsRequestToJSON;
function listShippoAccountsRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ListShippoAccountsRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListShippoAccountsRequest' from JSON`);
}
exports.listShippoAccountsRequestFromJSON = listShippoAccountsRequestFromJSON;
