// shippo/models/operations/listshipmentratesbycurrencycode.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.listShipmentRatesByCurrencyCodeRequestFromJSON = exports.listShipmentRatesByCurrencyCodeRequestToJSON = exports.ListShipmentRatesByCurrencyCodeRequest$ = exports.ListShipmentRatesByCurrencyCodeRequest$outboundSchema = exports.ListShipmentRatesByCurrencyCodeRequest$inboundSchema = exports.listShipmentRatesByCurrencyCodeGlobalsFromJSON = exports.listShipmentRatesByCurrencyCodeGlobalsToJSON = exports.ListShipmentRatesByCurrencyCodeGlobals$ = exports.ListShipmentRatesByCurrencyCodeGlobals$outboundSchema = exports.ListShipmentRatesByCurrencyCodeGlobals$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.ListShipmentRatesByCurrencyCodeGlobals$inboundSchema = z.object({
    "SHIPPO-API-VERSION": z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "SHIPPO-API-VERSION": "shippoApiVersion",
    });
});
/** @internal */
exports.ListShipmentRatesByCurrencyCodeGlobals$outboundSchema = z.object({
    shippoApiVersion: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        shippoApiVersion: "SHIPPO-API-VERSION",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ListShipmentRatesByCurrencyCodeGlobals$;
(function (ListShipmentRatesByCurrencyCodeGlobals$) {
    /** @deprecated use `ListShipmentRatesByCurrencyCodeGlobals$inboundSchema` instead. */
    ListShipmentRatesByCurrencyCodeGlobals$.inboundSchema = exports.ListShipmentRatesByCurrencyCodeGlobals$inboundSchema;
    /** @deprecated use `ListShipmentRatesByCurrencyCodeGlobals$outboundSchema` instead. */
    ListShipmentRatesByCurrencyCodeGlobals$.outboundSchema = exports.ListShipmentRatesByCurrencyCodeGlobals$outboundSchema;
})(ListShipmentRatesByCurrencyCodeGlobals$ || (exports.ListShipmentRatesByCurrencyCodeGlobals$ = ListShipmentRatesByCurrencyCodeGlobals$ = {}));
function listShipmentRatesByCurrencyCodeGlobalsToJSON(listShipmentRatesByCurrencyCodeGlobals) {
    return JSON.stringify(exports.ListShipmentRatesByCurrencyCodeGlobals$outboundSchema.parse(listShipmentRatesByCurrencyCodeGlobals));
}
exports.listShipmentRatesByCurrencyCodeGlobalsToJSON = listShipmentRatesByCurrencyCodeGlobalsToJSON;
function listShipmentRatesByCurrencyCodeGlobalsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ListShipmentRatesByCurrencyCodeGlobals$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListShipmentRatesByCurrencyCodeGlobals' from JSON`);
}
exports.listShipmentRatesByCurrencyCodeGlobalsFromJSON = listShipmentRatesByCurrencyCodeGlobalsFromJSON;
/** @internal */
exports.ListShipmentRatesByCurrencyCodeRequest$inboundSchema = z.object({
    ShipmentId: z.string(),
    CurrencyCode: z.string().default("USD"),
    page: z.number().int().default(1),
    results: z.number().int().default(25),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "ShipmentId": "shipmentId",
        "CurrencyCode": "currencyCode",
    });
});
/** @internal */
exports.ListShipmentRatesByCurrencyCodeRequest$outboundSchema = z.object({
    shipmentId: z.string(),
    currencyCode: z.string().default("USD"),
    page: z.number().int().default(1),
    results: z.number().int().default(25),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        shipmentId: "ShipmentId",
        currencyCode: "CurrencyCode",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ListShipmentRatesByCurrencyCodeRequest$;
(function (ListShipmentRatesByCurrencyCodeRequest$) {
    /** @deprecated use `ListShipmentRatesByCurrencyCodeRequest$inboundSchema` instead. */
    ListShipmentRatesByCurrencyCodeRequest$.inboundSchema = exports.ListShipmentRatesByCurrencyCodeRequest$inboundSchema;
    /** @deprecated use `ListShipmentRatesByCurrencyCodeRequest$outboundSchema` instead. */
    ListShipmentRatesByCurrencyCodeRequest$.outboundSchema = exports.ListShipmentRatesByCurrencyCodeRequest$outboundSchema;
})(ListShipmentRatesByCurrencyCodeRequest$ || (exports.ListShipmentRatesByCurrencyCodeRequest$ = ListShipmentRatesByCurrencyCodeRequest$ = {}));
function listShipmentRatesByCurrencyCodeRequestToJSON(listShipmentRatesByCurrencyCodeRequest) {
    return JSON.stringify(exports.ListShipmentRatesByCurrencyCodeRequest$outboundSchema.parse(listShipmentRatesByCurrencyCodeRequest));
}
exports.listShipmentRatesByCurrencyCodeRequestToJSON = listShipmentRatesByCurrencyCodeRequestToJSON;
function listShipmentRatesByCurrencyCodeRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ListShipmentRatesByCurrencyCodeRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListShipmentRatesByCurrencyCodeRequest' from JSON`);
}
exports.listShipmentRatesByCurrencyCodeRequestFromJSON = listShipmentRatesByCurrencyCodeRequestFromJSON;
