// shippo/models/operations/listmanifests.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.listManifestsRequestFromJSON = exports.listManifestsRequestToJSON = exports.ListManifestsRequest$ = exports.ListManifestsRequest$outboundSchema = exports.ListManifestsRequest$inboundSchema = exports.listManifestsGlobalsFromJSON = exports.listManifestsGlobalsToJSON = exports.ListManifestsGlobals$ = exports.ListManifestsGlobals$outboundSchema = exports.ListManifestsGlobals$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.ListManifestsGlobals$inboundSchema = z.object({
    "SHIPPO-API-VERSION": z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "SHIPPO-API-VERSION": "shippoApiVersion",
    });
});
/** @internal */
exports.ListManifestsGlobals$outboundSchema = z.object({
    shippoApiVersion: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        shippoApiVersion: "SHIPPO-API-VERSION",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ListManifestsGlobals$;
(function (ListManifestsGlobals$) {
    /** @deprecated use `ListManifestsGlobals$inboundSchema` instead. */
    ListManifestsGlobals$.inboundSchema = exports.ListManifestsGlobals$inboundSchema;
    /** @deprecated use `ListManifestsGlobals$outboundSchema` instead. */
    ListManifestsGlobals$.outboundSchema = exports.ListManifestsGlobals$outboundSchema;
})(ListManifestsGlobals$ || (exports.ListManifestsGlobals$ = ListManifestsGlobals$ = {}));
function listManifestsGlobalsToJSON(listManifestsGlobals) {
    return JSON.stringify(exports.ListManifestsGlobals$outboundSchema.parse(listManifestsGlobals));
}
exports.listManifestsGlobalsToJSON = listManifestsGlobalsToJSON;
function listManifestsGlobalsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ListManifestsGlobals$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListManifestsGlobals' from JSON`);
}
exports.listManifestsGlobalsFromJSON = listManifestsGlobalsFromJSON;
/** @internal */
exports.ListManifestsRequest$inboundSchema = z.object({
    page: z.number().int().default(1),
    results: z.number().int().default(5),
});
/** @internal */
exports.ListManifestsRequest$outboundSchema = z.object({
    page: z.number().int().default(1),
    results: z.number().int().default(5),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ListManifestsRequest$;
(function (ListManifestsRequest$) {
    /** @deprecated use `ListManifestsRequest$inboundSchema` instead. */
    ListManifestsRequest$.inboundSchema = exports.ListManifestsRequest$inboundSchema;
    /** @deprecated use `ListManifestsRequest$outboundSchema` instead. */
    ListManifestsRequest$.outboundSchema = exports.ListManifestsRequest$outboundSchema;
})(ListManifestsRequest$ || (exports.ListManifestsRequest$ = ListManifestsRequest$ = {}));
function listManifestsRequestToJSON(listManifestsRequest) {
    return JSON.stringify(exports.ListManifestsRequest$outboundSchema.parse(listManifestsRequest));
}
exports.listManifestsRequestToJSON = listManifestsRequestToJSON;
function listManifestsRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ListManifestsRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListManifestsRequest' from JSON`);
}
exports.listManifestsRequestFromJSON = listManifestsRequestFromJSON;
