// shippo/models/operations/listcarrieraccounts.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.listCarrierAccountsRequestFromJSON = exports.listCarrierAccountsRequestToJSON = exports.ListCarrierAccountsRequest$ = exports.ListCarrierAccountsRequest$outboundSchema = exports.ListCarrierAccountsRequest$inboundSchema = exports.listCarrierAccountsGlobalsFromJSON = exports.listCarrierAccountsGlobalsToJSON = exports.ListCarrierAccountsGlobals$ = exports.ListCarrierAccountsGlobals$outboundSchema = exports.ListCarrierAccountsGlobals$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.ListCarrierAccountsGlobals$inboundSchema = z.object({
    "SHIPPO-API-VERSION": z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "SHIPPO-API-VERSION": "shippoApiVersion",
    });
});
/** @internal */
exports.ListCarrierAccountsGlobals$outboundSchema = z.object({
    shippoApiVersion: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        shippoApiVersion: "SHIPPO-API-VERSION",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ListCarrierAccountsGlobals$;
(function (ListCarrierAccountsGlobals$) {
    /** @deprecated use `ListCarrierAccountsGlobals$inboundSchema` instead. */
    ListCarrierAccountsGlobals$.inboundSchema = exports.ListCarrierAccountsGlobals$inboundSchema;
    /** @deprecated use `ListCarrierAccountsGlobals$outboundSchema` instead. */
    ListCarrierAccountsGlobals$.outboundSchema = exports.ListCarrierAccountsGlobals$outboundSchema;
})(ListCarrierAccountsGlobals$ || (exports.ListCarrierAccountsGlobals$ = ListCarrierAccountsGlobals$ = {}));
function listCarrierAccountsGlobalsToJSON(listCarrierAccountsGlobals) {
    return JSON.stringify(exports.ListCarrierAccountsGlobals$outboundSchema.parse(listCarrierAccountsGlobals));
}
exports.listCarrierAccountsGlobalsToJSON = listCarrierAccountsGlobalsToJSON;
function listCarrierAccountsGlobalsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ListCarrierAccountsGlobals$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListCarrierAccountsGlobals' from JSON`);
}
exports.listCarrierAccountsGlobalsFromJSON = listCarrierAccountsGlobalsFromJSON;
/** @internal */
exports.ListCarrierAccountsRequest$inboundSchema = z.object({
    service_levels: z.boolean().optional(),
    carrier: components.CarriersEnum$inboundSchema.optional(),
    account_id: z.string().optional(),
    page: z.number().int().default(1),
    results: z.number().int().default(5),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "service_levels": "serviceLevels",
        "account_id": "accountId",
    });
});
/** @internal */
exports.ListCarrierAccountsRequest$outboundSchema = z.object({
    serviceLevels: z.boolean().optional(),
    carrier: components.CarriersEnum$outboundSchema.optional(),
    accountId: z.string().optional(),
    page: z.number().int().default(1),
    results: z.number().int().default(5),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        serviceLevels: "service_levels",
        accountId: "account_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ListCarrierAccountsRequest$;
(function (ListCarrierAccountsRequest$) {
    /** @deprecated use `ListCarrierAccountsRequest$inboundSchema` instead. */
    ListCarrierAccountsRequest$.inboundSchema = exports.ListCarrierAccountsRequest$inboundSchema;
    /** @deprecated use `ListCarrierAccountsRequest$outboundSchema` instead. */
    ListCarrierAccountsRequest$.outboundSchema = exports.ListCarrierAccountsRequest$outboundSchema;
})(ListCarrierAccountsRequest$ || (exports.ListCarrierAccountsRequest$ = ListCarrierAccountsRequest$ = {}));
function listCarrierAccountsRequestToJSON(listCarrierAccountsRequest) {
    return JSON.stringify(exports.ListCarrierAccountsRequest$outboundSchema.parse(listCarrierAccountsRequest));
}
exports.listCarrierAccountsRequestToJSON = listCarrierAccountsRequestToJSON;
function listCarrierAccountsRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ListCarrierAccountsRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListCarrierAccountsRequest' from JSON`);
}
exports.listCarrierAccountsRequestFromJSON = listCarrierAccountsRequestFromJSON;
