// shippo/models/operations/listshipments.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.listShipmentsRequestFromJSON = exports.listShipmentsRequestToJSON = exports.ListShipmentsRequest$ = exports.ListShipmentsRequest$outboundSchema = exports.ListShipmentsRequest$inboundSchema = exports.listShipmentsGlobalsFromJSON = exports.listShipmentsGlobalsToJSON = exports.ListShipmentsGlobals$ = exports.ListShipmentsGlobals$outboundSchema = exports.ListShipmentsGlobals$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.ListShipmentsGlobals$inboundSchema = z.object({
    "SHIPPO-API-VERSION": z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "SHIPPO-API-VERSION": "shippoApiVersion",
    });
});
/** @internal */
exports.ListShipmentsGlobals$outboundSchema = z.object({
    shippoApiVersion: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        shippoApiVersion: "SHIPPO-API-VERSION",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ListShipmentsGlobals$;
(function (ListShipmentsGlobals$) {
    /** @deprecated use `ListShipmentsGlobals$inboundSchema` instead. */
    ListShipmentsGlobals$.inboundSchema = exports.ListShipmentsGlobals$inboundSchema;
    /** @deprecated use `ListShipmentsGlobals$outboundSchema` instead. */
    ListShipmentsGlobals$.outboundSchema = exports.ListShipmentsGlobals$outboundSchema;
})(ListShipmentsGlobals$ || (exports.ListShipmentsGlobals$ = ListShipmentsGlobals$ = {}));
function listShipmentsGlobalsToJSON(listShipmentsGlobals) {
    return JSON.stringify(exports.ListShipmentsGlobals$outboundSchema.parse(listShipmentsGlobals));
}
exports.listShipmentsGlobalsToJSON = listShipmentsGlobalsToJSON;
function listShipmentsGlobalsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ListShipmentsGlobals$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListShipmentsGlobals' from JSON`);
}
exports.listShipmentsGlobalsFromJSON = listShipmentsGlobalsFromJSON;
/** @internal */
exports.ListShipmentsRequest$inboundSchema = z.object({
    page_token: z.string().optional(),
    page: z.number().int().default(1),
    results: z.number().int().default(25),
    object_created_gt: z.string().optional(),
    object_created_gte: z.string().optional(),
    object_created_lt: z.string().optional(),
    object_created_lte: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "page_token": "pageToken",
        "object_created_gt": "objectCreatedGt",
        "object_created_gte": "objectCreatedGte",
        "object_created_lt": "objectCreatedLt",
        "object_created_lte": "objectCreatedLte",
    });
});
/** @internal */
exports.ListShipmentsRequest$outboundSchema = z.object({
    pageToken: z.string().optional(),
    page: z.number().int().default(1),
    results: z.number().int().default(25),
    objectCreatedGt: z.string().optional(),
    objectCreatedGte: z.string().optional(),
    objectCreatedLt: z.string().optional(),
    objectCreatedLte: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        pageToken: "page_token",
        objectCreatedGt: "object_created_gt",
        objectCreatedGte: "object_created_gte",
        objectCreatedLt: "object_created_lt",
        objectCreatedLte: "object_created_lte",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ListShipmentsRequest$;
(function (ListShipmentsRequest$) {
    /** @deprecated use `ListShipmentsRequest$inboundSchema` instead. */
    ListShipmentsRequest$.inboundSchema = exports.ListShipmentsRequest$inboundSchema;
    /** @deprecated use `ListShipmentsRequest$outboundSchema` instead. */
    ListShipmentsRequest$.outboundSchema = exports.ListShipmentsRequest$outboundSchema;
})(ListShipmentsRequest$ || (exports.ListShipmentsRequest$ = ListShipmentsRequest$ = {}));
function listShipmentsRequestToJSON(listShipmentsRequest) {
    return JSON.stringify(exports.ListShipmentsRequest$outboundSchema.parse(listShipmentsRequest));
}
exports.listShipmentsRequestToJSON = listShipmentsRequestToJSON;
function listShipmentsRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ListShipmentsRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListShipmentsRequest' from JSON`);
}
exports.listShipmentsRequestFromJSON = listShipmentsRequestFromJSON;
