// shippo/models/operations/listservicegroups.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.listServiceGroupsRequestFromJSON = exports.listServiceGroupsRequestToJSON = exports.ListServiceGroupsRequest$ = exports.ListServiceGroupsRequest$outboundSchema = exports.ListServiceGroupsRequest$inboundSchema = exports.listServiceGroupsGlobalsFromJSON = exports.listServiceGroupsGlobalsToJSON = exports.ListServiceGroupsGlobals$ = exports.ListServiceGroupsGlobals$outboundSchema = exports.ListServiceGroupsGlobals$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.ListServiceGroupsGlobals$inboundSchema = z.object({
    "SHIPPO-API-VERSION": z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "SHIPPO-API-VERSION": "shippoApiVersion",
    });
});
/** @internal */
exports.ListServiceGroupsGlobals$outboundSchema = z.object({
    shippoApiVersion: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        shippoApiVersion: "SHIPPO-API-VERSION",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ListServiceGroupsGlobals$;
(function (ListServiceGroupsGlobals$) {
    /** @deprecated use `ListServiceGroupsGlobals$inboundSchema` instead. */
    ListServiceGroupsGlobals$.inboundSchema = exports.ListServiceGroupsGlobals$inboundSchema;
    /** @deprecated use `ListServiceGroupsGlobals$outboundSchema` instead. */
    ListServiceGroupsGlobals$.outboundSchema = exports.ListServiceGroupsGlobals$outboundSchema;
})(ListServiceGroupsGlobals$ || (exports.ListServiceGroupsGlobals$ = ListServiceGroupsGlobals$ = {}));
function listServiceGroupsGlobalsToJSON(listServiceGroupsGlobals) {
    return JSON.stringify(exports.ListServiceGroupsGlobals$outboundSchema.parse(listServiceGroupsGlobals));
}
exports.listServiceGroupsGlobalsToJSON = listServiceGroupsGlobalsToJSON;
function listServiceGroupsGlobalsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ListServiceGroupsGlobals$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListServiceGroupsGlobals' from JSON`);
}
exports.listServiceGroupsGlobalsFromJSON = listServiceGroupsGlobalsFromJSON;
/** @internal */
exports.ListServiceGroupsRequest$inboundSchema = z.object({});
/** @internal */
exports.ListServiceGroupsRequest$outboundSchema = z.object({});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ListServiceGroupsRequest$;
(function (ListServiceGroupsRequest$) {
    /** @deprecated use `ListServiceGroupsRequest$inboundSchema` instead. */
    ListServiceGroupsRequest$.inboundSchema = exports.ListServiceGroupsRequest$inboundSchema;
    /** @deprecated use `ListServiceGroupsRequest$outboundSchema` instead. */
    ListServiceGroupsRequest$.outboundSchema = exports.ListServiceGroupsRequest$outboundSchema;
})(ListServiceGroupsRequest$ || (exports.ListServiceGroupsRequest$ = ListServiceGroupsRequest$ = {}));
function listServiceGroupsRequestToJSON(listServiceGroupsRequest) {
    return JSON.stringify(exports.ListServiceGroupsRequest$outboundSchema.parse(listServiceGroupsRequest));
}
exports.listServiceGroupsRequestToJSON = listServiceGroupsRequestToJSON;
function listServiceGroupsRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ListServiceGroupsRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListServiceGroupsRequest' from JSON`);
}
exports.listServiceGroupsRequestFromJSON = listServiceGroupsRequestFromJSON;
