// shippo/models/operations/listuserparceltemplates.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.listUserParcelTemplatesRequestFromJSON = exports.listUserParcelTemplatesRequestToJSON = exports.ListUserParcelTemplatesRequest$ = exports.ListUserParcelTemplatesRequest$outboundSchema = exports.ListUserParcelTemplatesRequest$inboundSchema = exports.listUserParcelTemplatesGlobalsFromJSON = exports.listUserParcelTemplatesGlobalsToJSON = exports.ListUserParcelTemplatesGlobals$ = exports.ListUserParcelTemplatesGlobals$outboundSchema = exports.ListUserParcelTemplatesGlobals$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.ListUserParcelTemplatesGlobals$inboundSchema = z.object({
    "SHIPPO-API-VERSION": z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "SHIPPO-API-VERSION": "shippoApiVersion",
    });
});
/** @internal */
exports.ListUserParcelTemplatesGlobals$outboundSchema = z.object({
    shippoApiVersion: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        shippoApiVersion: "SHIPPO-API-VERSION",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ListUserParcelTemplatesGlobals$;
(function (ListUserParcelTemplatesGlobals$) {
    /** @deprecated use `ListUserParcelTemplatesGlobals$inboundSchema` instead. */
    ListUserParcelTemplatesGlobals$.inboundSchema = exports.ListUserParcelTemplatesGlobals$inboundSchema;
    /** @deprecated use `ListUserParcelTemplatesGlobals$outboundSchema` instead. */
    ListUserParcelTemplatesGlobals$.outboundSchema = exports.ListUserParcelTemplatesGlobals$outboundSchema;
})(ListUserParcelTemplatesGlobals$ || (exports.ListUserParcelTemplatesGlobals$ = ListUserParcelTemplatesGlobals$ = {}));
function listUserParcelTemplatesGlobalsToJSON(listUserParcelTemplatesGlobals) {
    return JSON.stringify(exports.ListUserParcelTemplatesGlobals$outboundSchema.parse(listUserParcelTemplatesGlobals));
}
exports.listUserParcelTemplatesGlobalsToJSON = listUserParcelTemplatesGlobalsToJSON;
function listUserParcelTemplatesGlobalsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ListUserParcelTemplatesGlobals$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListUserParcelTemplatesGlobals' from JSON`);
}
exports.listUserParcelTemplatesGlobalsFromJSON = listUserParcelTemplatesGlobalsFromJSON;
/** @internal */
exports.ListUserParcelTemplatesRequest$inboundSchema = z.object({});
/** @internal */
exports.ListUserParcelTemplatesRequest$outboundSchema = z.object({});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ListUserParcelTemplatesRequest$;
(function (ListUserParcelTemplatesRequest$) {
    /** @deprecated use `ListUserParcelTemplatesRequest$inboundSchema` instead. */
    ListUserParcelTemplatesRequest$.inboundSchema = exports.ListUserParcelTemplatesRequest$inboundSchema;
    /** @deprecated use `ListUserParcelTemplatesRequest$outboundSchema` instead. */
    ListUserParcelTemplatesRequest$.outboundSchema = exports.ListUserParcelTemplatesRequest$outboundSchema;
})(ListUserParcelTemplatesRequest$ || (exports.ListUserParcelTemplatesRequest$ = ListUserParcelTemplatesRequest$ = {}));
function listUserParcelTemplatesRequestToJSON(listUserParcelTemplatesRequest) {
    return JSON.stringify(exports.ListUserParcelTemplatesRequest$outboundSchema.parse(listUserParcelTemplatesRequest));
}
exports.listUserParcelTemplatesRequestToJSON = listUserParcelTemplatesRequestToJSON;
function listUserParcelTemplatesRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ListUserParcelTemplatesRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListUserParcelTemplatesRequest' from JSON`);
}
exports.listUserParcelTemplatesRequestFromJSON = listUserParcelTemplatesRequestFromJSON;
