// shippo/models/operations/listaddresses.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.listAddressesRequestFromJSON = exports.listAddressesRequestToJSON = exports.ListAddressesRequest$ = exports.ListAddressesRequest$outboundSchema = exports.ListAddressesRequest$inboundSchema = exports.listAddressesGlobalsFromJSON = exports.listAddressesGlobalsToJSON = exports.ListAddressesGlobals$ = exports.ListAddressesGlobals$outboundSchema = exports.ListAddressesGlobals$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.ListAddressesGlobals$inboundSchema = z.object({
    "SHIPPO-API-VERSION": z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "SHIPPO-API-VERSION": "shippoApiVersion",
    });
});
/** @internal */
exports.ListAddressesGlobals$outboundSchema = z.object({
    shippoApiVersion: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        shippoApiVersion: "SHIPPO-API-VERSION",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ListAddressesGlobals$;
(function (ListAddressesGlobals$) {
    /** @deprecated use `ListAddressesGlobals$inboundSchema` instead. */
    ListAddressesGlobals$.inboundSchema = exports.ListAddressesGlobals$inboundSchema;
    /** @deprecated use `ListAddressesGlobals$outboundSchema` instead. */
    ListAddressesGlobals$.outboundSchema = exports.ListAddressesGlobals$outboundSchema;
})(ListAddressesGlobals$ || (exports.ListAddressesGlobals$ = ListAddressesGlobals$ = {}));
function listAddressesGlobalsToJSON(listAddressesGlobals) {
    return JSON.stringify(exports.ListAddressesGlobals$outboundSchema.parse(listAddressesGlobals));
}
exports.listAddressesGlobalsToJSON = listAddressesGlobalsToJSON;
function listAddressesGlobalsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ListAddressesGlobals$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListAddressesGlobals' from JSON`);
}
exports.listAddressesGlobalsFromJSON = listAddressesGlobalsFromJSON;
/** @internal */
exports.ListAddressesRequest$inboundSchema = z.object({
    page: z.number().int().default(1),
    results: z.number().int().default(5),
});
/** @internal */
exports.ListAddressesRequest$outboundSchema = z.object({
    page: z.number().int().default(1),
    results: z.number().int().default(5),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ListAddressesRequest$;
(function (ListAddressesRequest$) {
    /** @deprecated use `ListAddressesRequest$inboundSchema` instead. */
    ListAddressesRequest$.inboundSchema = exports.ListAddressesRequest$inboundSchema;
    /** @deprecated use `ListAddressesRequest$outboundSchema` instead. */
    ListAddressesRequest$.outboundSchema = exports.ListAddressesRequest$outboundSchema;
})(ListAddressesRequest$ || (exports.ListAddressesRequest$ = ListAddressesRequest$ = {}));
function listAddressesRequestToJSON(listAddressesRequest) {
    return JSON.stringify(exports.ListAddressesRequest$outboundSchema.parse(listAddressesRequest));
}
exports.listAddressesRequestToJSON = listAddressesRequestToJSON;
function listAddressesRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ListAddressesRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListAddressesRequest' from JSON`);
}
exports.listAddressesRequestFromJSON = listAddressesRequestFromJSON;
