// shippo/models/operations/listparcels.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.listParcelsRequestFromJSON = exports.listParcelsRequestToJSON = exports.ListParcelsRequest$ = exports.ListParcelsRequest$outboundSchema = exports.ListParcelsRequest$inboundSchema = exports.listParcelsGlobalsFromJSON = exports.listParcelsGlobalsToJSON = exports.ListParcelsGlobals$ = exports.ListParcelsGlobals$outboundSchema = exports.ListParcelsGlobals$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.ListParcelsGlobals$inboundSchema = z.object({
    "SHIPPO-API-VERSION": z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "SHIPPO-API-VERSION": "shippoApiVersion",
    });
});
/** @internal */
exports.ListParcelsGlobals$outboundSchema = z.object({
    shippoApiVersion: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        shippoApiVersion: "SHIPPO-API-VERSION",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ListParcelsGlobals$;
(function (ListParcelsGlobals$) {
    /** @deprecated use `ListParcelsGlobals$inboundSchema` instead. */
    ListParcelsGlobals$.inboundSchema = exports.ListParcelsGlobals$inboundSchema;
    /** @deprecated use `ListParcelsGlobals$outboundSchema` instead. */
    ListParcelsGlobals$.outboundSchema = exports.ListParcelsGlobals$outboundSchema;
})(ListParcelsGlobals$ || (exports.ListParcelsGlobals$ = ListParcelsGlobals$ = {}));
function listParcelsGlobalsToJSON(listParcelsGlobals) {
    return JSON.stringify(exports.ListParcelsGlobals$outboundSchema.parse(listParcelsGlobals));
}
exports.listParcelsGlobalsToJSON = listParcelsGlobalsToJSON;
function listParcelsGlobalsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ListParcelsGlobals$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListParcelsGlobals' from JSON`);
}
exports.listParcelsGlobalsFromJSON = listParcelsGlobalsFromJSON;
/** @internal */
exports.ListParcelsRequest$inboundSchema = z.object({
    page: z.number().int().default(1),
    results: z.number().int().default(25),
});
/** @internal */
exports.ListParcelsRequest$outboundSchema = z.object({
    page: z.number().int().default(1),
    results: z.number().int().default(25),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ListParcelsRequest$;
(function (ListParcelsRequest$) {
    /** @deprecated use `ListParcelsRequest$inboundSchema` instead. */
    ListParcelsRequest$.inboundSchema = exports.ListParcelsRequest$inboundSchema;
    /** @deprecated use `ListParcelsRequest$outboundSchema` instead. */
    ListParcelsRequest$.outboundSchema = exports.ListParcelsRequest$outboundSchema;
})(ListParcelsRequest$ || (exports.ListParcelsRequest$ = ListParcelsRequest$ = {}));
function listParcelsRequestToJSON(listParcelsRequest) {
    return JSON.stringify(exports.ListParcelsRequest$outboundSchema.parse(listParcelsRequest));
}
exports.listParcelsRequestToJSON = listParcelsRequestToJSON;
function listParcelsRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ListParcelsRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListParcelsRequest' from JSON`);
}
exports.listParcelsRequestFromJSON = listParcelsRequestFromJSON;
