// shippo/models/operations/listcustomsitems.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.listCustomsItemsRequestFromJSON = exports.listCustomsItemsRequestToJSON = exports.ListCustomsItemsRequest$ = exports.ListCustomsItemsRequest$outboundSchema = exports.ListCustomsItemsRequest$inboundSchema = exports.listCustomsItemsGlobalsFromJSON = exports.listCustomsItemsGlobalsToJSON = exports.ListCustomsItemsGlobals$ = exports.ListCustomsItemsGlobals$outboundSchema = exports.ListCustomsItemsGlobals$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.ListCustomsItemsGlobals$inboundSchema = z.object({
    "SHIPPO-API-VERSION": z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "SHIPPO-API-VERSION": "shippoApiVersion",
    });
});
/** @internal */
exports.ListCustomsItemsGlobals$outboundSchema = z.object({
    shippoApiVersion: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        shippoApiVersion: "SHIPPO-API-VERSION",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ListCustomsItemsGlobals$;
(function (ListCustomsItemsGlobals$) {
    /** @deprecated use `ListCustomsItemsGlobals$inboundSchema` instead. */
    ListCustomsItemsGlobals$.inboundSchema = exports.ListCustomsItemsGlobals$inboundSchema;
    /** @deprecated use `ListCustomsItemsGlobals$outboundSchema` instead. */
    ListCustomsItemsGlobals$.outboundSchema = exports.ListCustomsItemsGlobals$outboundSchema;
})(ListCustomsItemsGlobals$ || (exports.ListCustomsItemsGlobals$ = ListCustomsItemsGlobals$ = {}));
function listCustomsItemsGlobalsToJSON(listCustomsItemsGlobals) {
    return JSON.stringify(exports.ListCustomsItemsGlobals$outboundSchema.parse(listCustomsItemsGlobals));
}
exports.listCustomsItemsGlobalsToJSON = listCustomsItemsGlobalsToJSON;
function listCustomsItemsGlobalsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ListCustomsItemsGlobals$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListCustomsItemsGlobals' from JSON`);
}
exports.listCustomsItemsGlobalsFromJSON = listCustomsItemsGlobalsFromJSON;
/** @internal */
exports.ListCustomsItemsRequest$inboundSchema = z.object({
    page: z.number().int().default(1),
    results: z.number().int().default(25),
});
/** @internal */
exports.ListCustomsItemsRequest$outboundSchema = z.object({
    page: z.number().int().default(1),
    results: z.number().int().default(25),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ListCustomsItemsRequest$;
(function (ListCustomsItemsRequest$) {
    /** @deprecated use `ListCustomsItemsRequest$inboundSchema` instead. */
    ListCustomsItemsRequest$.inboundSchema = exports.ListCustomsItemsRequest$inboundSchema;
    /** @deprecated use `ListCustomsItemsRequest$outboundSchema` instead. */
    ListCustomsItemsRequest$.outboundSchema = exports.ListCustomsItemsRequest$outboundSchema;
})(ListCustomsItemsRequest$ || (exports.ListCustomsItemsRequest$ = ListCustomsItemsRequest$ = {}));
function listCustomsItemsRequestToJSON(listCustomsItemsRequest) {
    return JSON.stringify(exports.ListCustomsItemsRequest$outboundSchema.parse(listCustomsItemsRequest));
}
exports.listCustomsItemsRequestToJSON = listCustomsItemsRequestToJSON;
function listCustomsItemsRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ListCustomsItemsRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListCustomsItemsRequest' from JSON`);
}
exports.listCustomsItemsRequestFromJSON = listCustomsItemsRequestFromJSON;
