// shippo/models/operations/listrefunds.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.listRefundsRequestFromJSON = exports.listRefundsRequestToJSON = exports.ListRefundsRequest$ = exports.ListRefundsRequest$outboundSchema = exports.ListRefundsRequest$inboundSchema = exports.listRefundsGlobalsFromJSON = exports.listRefundsGlobalsToJSON = exports.ListRefundsGlobals$ = exports.ListRefundsGlobals$outboundSchema = exports.ListRefundsGlobals$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.ListRefundsGlobals$inboundSchema = z.object({
    "SHIPPO-API-VERSION": z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "SHIPPO-API-VERSION": "shippoApiVersion",
    });
});
/** @internal */
exports.ListRefundsGlobals$outboundSchema = z.object({
    shippoApiVersion: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        shippoApiVersion: "SHIPPO-API-VERSION",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ListRefundsGlobals$;
(function (ListRefundsGlobals$) {
    /** @deprecated use `ListRefundsGlobals$inboundSchema` instead. */
    ListRefundsGlobals$.inboundSchema = exports.ListRefundsGlobals$inboundSchema;
    /** @deprecated use `ListRefundsGlobals$outboundSchema` instead. */
    ListRefundsGlobals$.outboundSchema = exports.ListRefundsGlobals$outboundSchema;
})(ListRefundsGlobals$ || (exports.ListRefundsGlobals$ = ListRefundsGlobals$ = {}));
function listRefundsGlobalsToJSON(listRefundsGlobals) {
    return JSON.stringify(exports.ListRefundsGlobals$outboundSchema.parse(listRefundsGlobals));
}
exports.listRefundsGlobalsToJSON = listRefundsGlobalsToJSON;
function listRefundsGlobalsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ListRefundsGlobals$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListRefundsGlobals' from JSON`);
}
exports.listRefundsGlobalsFromJSON = listRefundsGlobalsFromJSON;
/** @internal */
exports.ListRefundsRequest$inboundSchema = z.object({});
/** @internal */
exports.ListRefundsRequest$outboundSchema = z.object({});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ListRefundsRequest$;
(function (ListRefundsRequest$) {
    /** @deprecated use `ListRefundsRequest$inboundSchema` instead. */
    ListRefundsRequest$.inboundSchema = exports.ListRefundsRequest$inboundSchema;
    /** @deprecated use `ListRefundsRequest$outboundSchema` instead. */
    ListRefundsRequest$.outboundSchema = exports.ListRefundsRequest$outboundSchema;
})(ListRefundsRequest$ || (exports.ListRefundsRequest$ = ListRefundsRequest$ = {}));
function listRefundsRequestToJSON(listRefundsRequest) {
    return JSON.stringify(exports.ListRefundsRequest$outboundSchema.parse(listRefundsRequest));
}
exports.listRefundsRequestToJSON = listRefundsRequestToJSON;
function listRefundsRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ListRefundsRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListRefundsRequest' from JSON`);
}
exports.listRefundsRequestFromJSON = listRefundsRequestFromJSON;
