// shippo/models/operations/listshipmentrates.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.listShipmentRatesRequestFromJSON = exports.listShipmentRatesRequestToJSON = exports.ListShipmentRatesRequest$ = exports.ListShipmentRatesRequest$outboundSchema = exports.ListShipmentRatesRequest$inboundSchema = exports.listShipmentRatesGlobalsFromJSON = exports.listShipmentRatesGlobalsToJSON = exports.ListShipmentRatesGlobals$ = exports.ListShipmentRatesGlobals$outboundSchema = exports.ListShipmentRatesGlobals$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.ListShipmentRatesGlobals$inboundSchema = z.object({
    "SHIPPO-API-VERSION": z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "SHIPPO-API-VERSION": "shippoApiVersion",
    });
});
/** @internal */
exports.ListShipmentRatesGlobals$outboundSchema = z.object({
    shippoApiVersion: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        shippoApiVersion: "SHIPPO-API-VERSION",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ListShipmentRatesGlobals$;
(function (ListShipmentRatesGlobals$) {
    /** @deprecated use `ListShipmentRatesGlobals$inboundSchema` instead. */
    ListShipmentRatesGlobals$.inboundSchema = exports.ListShipmentRatesGlobals$inboundSchema;
    /** @deprecated use `ListShipmentRatesGlobals$outboundSchema` instead. */
    ListShipmentRatesGlobals$.outboundSchema = exports.ListShipmentRatesGlobals$outboundSchema;
})(ListShipmentRatesGlobals$ || (exports.ListShipmentRatesGlobals$ = ListShipmentRatesGlobals$ = {}));
function listShipmentRatesGlobalsToJSON(listShipmentRatesGlobals) {
    return JSON.stringify(exports.ListShipmentRatesGlobals$outboundSchema.parse(listShipmentRatesGlobals));
}
exports.listShipmentRatesGlobalsToJSON = listShipmentRatesGlobalsToJSON;
function listShipmentRatesGlobalsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ListShipmentRatesGlobals$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListShipmentRatesGlobals' from JSON`);
}
exports.listShipmentRatesGlobalsFromJSON = listShipmentRatesGlobalsFromJSON;
/** @internal */
exports.ListShipmentRatesRequest$inboundSchema = z.object({
    ShipmentId: z.string(),
    page: z.number().int().default(1),
    results: z.number().int().default(25),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "ShipmentId": "shipmentId",
    });
});
/** @internal */
exports.ListShipmentRatesRequest$outboundSchema = z.object({
    shipmentId: z.string(),
    page: z.number().int().default(1),
    results: z.number().int().default(25),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        shipmentId: "ShipmentId",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ListShipmentRatesRequest$;
(function (ListShipmentRatesRequest$) {
    /** @deprecated use `ListShipmentRatesRequest$inboundSchema` instead. */
    ListShipmentRatesRequest$.inboundSchema = exports.ListShipmentRatesRequest$inboundSchema;
    /** @deprecated use `ListShipmentRatesRequest$outboundSchema` instead. */
    ListShipmentRatesRequest$.outboundSchema = exports.ListShipmentRatesRequest$outboundSchema;
})(ListShipmentRatesRequest$ || (exports.ListShipmentRatesRequest$ = ListShipmentRatesRequest$ = {}));
function listShipmentRatesRequestToJSON(listShipmentRatesRequest) {
    return JSON.stringify(exports.ListShipmentRatesRequest$outboundSchema.parse(listShipmentRatesRequest));
}
exports.listShipmentRatesRequestToJSON = listShipmentRatesRequestToJSON;
function listShipmentRatesRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ListShipmentRatesRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ListShipmentRatesRequest' from JSON`);
}
exports.listShipmentRatesRequestFromJSON = listShipmentRatesRequestFromJSON;
