// shippo/models/operations/purchasebatch.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.purchaseBatchRequestFromJSON = exports.purchaseBatchRequestToJSON = exports.PurchaseBatchRequest$ = exports.PurchaseBatchRequest$outboundSchema = exports.PurchaseBatchRequest$inboundSchema = exports.purchaseBatchGlobalsFromJSON = exports.purchaseBatchGlobalsToJSON = exports.PurchaseBatchGlobals$ = exports.PurchaseBatchGlobals$outboundSchema = exports.PurchaseBatchGlobals$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.PurchaseBatchGlobals$inboundSchema = z.object({
    "SHIPPO-API-VERSION": z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "SHIPPO-API-VERSION": "shippoApiVersion",
    });
});
/** @internal */
exports.PurchaseBatchGlobals$outboundSchema = z.object({
    shippoApiVersion: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        shippoApiVersion: "SHIPPO-API-VERSION",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var PurchaseBatchGlobals$;
(function (PurchaseBatchGlobals$) {
    /** @deprecated use `PurchaseBatchGlobals$inboundSchema` instead. */
    PurchaseBatchGlobals$.inboundSchema = exports.PurchaseBatchGlobals$inboundSchema;
    /** @deprecated use `PurchaseBatchGlobals$outboundSchema` instead. */
    PurchaseBatchGlobals$.outboundSchema = exports.PurchaseBatchGlobals$outboundSchema;
})(PurchaseBatchGlobals$ || (exports.PurchaseBatchGlobals$ = PurchaseBatchGlobals$ = {}));
function purchaseBatchGlobalsToJSON(purchaseBatchGlobals) {
    return JSON.stringify(exports.PurchaseBatchGlobals$outboundSchema.parse(purchaseBatchGlobals));
}
exports.purchaseBatchGlobalsToJSON = purchaseBatchGlobalsToJSON;
function purchaseBatchGlobalsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.PurchaseBatchGlobals$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PurchaseBatchGlobals' from JSON`);
}
exports.purchaseBatchGlobalsFromJSON = purchaseBatchGlobalsFromJSON;
/** @internal */
exports.PurchaseBatchRequest$inboundSchema = z.object({
    BatchId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "BatchId": "batchId",
    });
});
/** @internal */
exports.PurchaseBatchRequest$outboundSchema = z.object({
    batchId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        batchId: "BatchId",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var PurchaseBatchRequest$;
(function (PurchaseBatchRequest$) {
    /** @deprecated use `PurchaseBatchRequest$inboundSchema` instead. */
    PurchaseBatchRequest$.inboundSchema = exports.PurchaseBatchRequest$inboundSchema;
    /** @deprecated use `PurchaseBatchRequest$outboundSchema` instead. */
    PurchaseBatchRequest$.outboundSchema = exports.PurchaseBatchRequest$outboundSchema;
})(PurchaseBatchRequest$ || (exports.PurchaseBatchRequest$ = PurchaseBatchRequest$ = {}));
function purchaseBatchRequestToJSON(purchaseBatchRequest) {
    return JSON.stringify(exports.PurchaseBatchRequest$outboundSchema.parse(purchaseBatchRequest));
}
exports.purchaseBatchRequestToJSON = purchaseBatchRequestToJSON;
function purchaseBatchRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.PurchaseBatchRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PurchaseBatchRequest' from JSON`);
}
exports.purchaseBatchRequestFromJSON = purchaseBatchRequestFromJSON;
