// shippo/models/operations/updateuserparceltemplate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.updateUserParcelTemplateRequestFromJSON = exports.updateUserParcelTemplateRequestToJSON = exports.UpdateUserParcelTemplateRequest$ = exports.UpdateUserParcelTemplateRequest$outboundSchema = exports.UpdateUserParcelTemplateRequest$inboundSchema = exports.updateUserParcelTemplateGlobalsFromJSON = exports.updateUserParcelTemplateGlobalsToJSON = exports.UpdateUserParcelTemplateGlobals$ = exports.UpdateUserParcelTemplateGlobals$outboundSchema = exports.UpdateUserParcelTemplateGlobals$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.UpdateUserParcelTemplateGlobals$inboundSchema = z.object({
    "SHIPPO-API-VERSION": z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "SHIPPO-API-VERSION": "shippoApiVersion",
    });
});
/** @internal */
exports.UpdateUserParcelTemplateGlobals$outboundSchema = z.object({
    shippoApiVersion: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        shippoApiVersion: "SHIPPO-API-VERSION",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var UpdateUserParcelTemplateGlobals$;
(function (UpdateUserParcelTemplateGlobals$) {
    /** @deprecated use `UpdateUserParcelTemplateGlobals$inboundSchema` instead. */
    UpdateUserParcelTemplateGlobals$.inboundSchema = exports.UpdateUserParcelTemplateGlobals$inboundSchema;
    /** @deprecated use `UpdateUserParcelTemplateGlobals$outboundSchema` instead. */
    UpdateUserParcelTemplateGlobals$.outboundSchema = exports.UpdateUserParcelTemplateGlobals$outboundSchema;
})(UpdateUserParcelTemplateGlobals$ || (exports.UpdateUserParcelTemplateGlobals$ = UpdateUserParcelTemplateGlobals$ = {}));
function updateUserParcelTemplateGlobalsToJSON(updateUserParcelTemplateGlobals) {
    return JSON.stringify(exports.UpdateUserParcelTemplateGlobals$outboundSchema.parse(updateUserParcelTemplateGlobals));
}
exports.updateUserParcelTemplateGlobalsToJSON = updateUserParcelTemplateGlobalsToJSON;
function updateUserParcelTemplateGlobalsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.UpdateUserParcelTemplateGlobals$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateUserParcelTemplateGlobals' from JSON`);
}
exports.updateUserParcelTemplateGlobalsFromJSON = updateUserParcelTemplateGlobalsFromJSON;
/** @internal */
exports.UpdateUserParcelTemplateRequest$inboundSchema = z.object({
    UserParcelTemplateObjectId: z.string(),
    UserParcelTemplateUpdateRequest: components
        .UserParcelTemplateUpdateRequest$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "UserParcelTemplateObjectId": "userParcelTemplateObjectId",
        "UserParcelTemplateUpdateRequest": "userParcelTemplateUpdateRequest",
    });
});
/** @internal */
exports.UpdateUserParcelTemplateRequest$outboundSchema = z.object({
    userParcelTemplateObjectId: z.string(),
    userParcelTemplateUpdateRequest: components
        .UserParcelTemplateUpdateRequest$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        userParcelTemplateObjectId: "UserParcelTemplateObjectId",
        userParcelTemplateUpdateRequest: "UserParcelTemplateUpdateRequest",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var UpdateUserParcelTemplateRequest$;
(function (UpdateUserParcelTemplateRequest$) {
    /** @deprecated use `UpdateUserParcelTemplateRequest$inboundSchema` instead. */
    UpdateUserParcelTemplateRequest$.inboundSchema = exports.UpdateUserParcelTemplateRequest$inboundSchema;
    /** @deprecated use `UpdateUserParcelTemplateRequest$outboundSchema` instead. */
    UpdateUserParcelTemplateRequest$.outboundSchema = exports.UpdateUserParcelTemplateRequest$outboundSchema;
})(UpdateUserParcelTemplateRequest$ || (exports.UpdateUserParcelTemplateRequest$ = UpdateUserParcelTemplateRequest$ = {}));
function updateUserParcelTemplateRequestToJSON(updateUserParcelTemplateRequest) {
    return JSON.stringify(exports.UpdateUserParcelTemplateRequest$outboundSchema.parse(updateUserParcelTemplateRequest));
}
exports.updateUserParcelTemplateRequestToJSON = updateUserParcelTemplateRequestToJSON;
function updateUserParcelTemplateRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.UpdateUserParcelTemplateRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateUserParcelTemplateRequest' from JSON`);
}
exports.updateUserParcelTemplateRequestFromJSON = updateUserParcelTemplateRequestFromJSON;
