// shippo/models/operations/updateshippoaccount.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.updateShippoAccountRequestFromJSON = exports.updateShippoAccountRequestToJSON = exports.UpdateShippoAccountRequest$ = exports.UpdateShippoAccountRequest$outboundSchema = exports.UpdateShippoAccountRequest$inboundSchema = exports.updateShippoAccountGlobalsFromJSON = exports.updateShippoAccountGlobalsToJSON = exports.UpdateShippoAccountGlobals$ = exports.UpdateShippoAccountGlobals$outboundSchema = exports.UpdateShippoAccountGlobals$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.UpdateShippoAccountGlobals$inboundSchema = z.object({
    "SHIPPO-API-VERSION": z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "SHIPPO-API-VERSION": "shippoApiVersion",
    });
});
/** @internal */
exports.UpdateShippoAccountGlobals$outboundSchema = z.object({
    shippoApiVersion: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        shippoApiVersion: "SHIPPO-API-VERSION",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var UpdateShippoAccountGlobals$;
(function (UpdateShippoAccountGlobals$) {
    /** @deprecated use `UpdateShippoAccountGlobals$inboundSchema` instead. */
    UpdateShippoAccountGlobals$.inboundSchema = exports.UpdateShippoAccountGlobals$inboundSchema;
    /** @deprecated use `UpdateShippoAccountGlobals$outboundSchema` instead. */
    UpdateShippoAccountGlobals$.outboundSchema = exports.UpdateShippoAccountGlobals$outboundSchema;
})(UpdateShippoAccountGlobals$ || (exports.UpdateShippoAccountGlobals$ = UpdateShippoAccountGlobals$ = {}));
function updateShippoAccountGlobalsToJSON(updateShippoAccountGlobals) {
    return JSON.stringify(exports.UpdateShippoAccountGlobals$outboundSchema.parse(updateShippoAccountGlobals));
}
exports.updateShippoAccountGlobalsToJSON = updateShippoAccountGlobalsToJSON;
function updateShippoAccountGlobalsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.UpdateShippoAccountGlobals$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateShippoAccountGlobals' from JSON`);
}
exports.updateShippoAccountGlobalsFromJSON = updateShippoAccountGlobalsFromJSON;
/** @internal */
exports.UpdateShippoAccountRequest$inboundSchema = z.object({
    ShippoAccountId: z.string(),
    ShippoAccountUpdateRequest: components
        .ShippoAccountUpdateRequest$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "ShippoAccountId": "shippoAccountId",
        "ShippoAccountUpdateRequest": "shippoAccountUpdateRequest",
    });
});
/** @internal */
exports.UpdateShippoAccountRequest$outboundSchema = z.object({
    shippoAccountId: z.string(),
    shippoAccountUpdateRequest: components
        .ShippoAccountUpdateRequest$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        shippoAccountId: "ShippoAccountId",
        shippoAccountUpdateRequest: "ShippoAccountUpdateRequest",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var UpdateShippoAccountRequest$;
(function (UpdateShippoAccountRequest$) {
    /** @deprecated use `UpdateShippoAccountRequest$inboundSchema` instead. */
    UpdateShippoAccountRequest$.inboundSchema = exports.UpdateShippoAccountRequest$inboundSchema;
    /** @deprecated use `UpdateShippoAccountRequest$outboundSchema` instead. */
    UpdateShippoAccountRequest$.outboundSchema = exports.UpdateShippoAccountRequest$outboundSchema;
})(UpdateShippoAccountRequest$ || (exports.UpdateShippoAccountRequest$ = UpdateShippoAccountRequest$ = {}));
function updateShippoAccountRequestToJSON(updateShippoAccountRequest) {
    return JSON.stringify(exports.UpdateShippoAccountRequest$outboundSchema.parse(updateShippoAccountRequest));
}
exports.updateShippoAccountRequestToJSON = updateShippoAccountRequestToJSON;
function updateShippoAccountRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.UpdateShippoAccountRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateShippoAccountRequest' from JSON`);
}
exports.updateShippoAccountRequestFromJSON = updateShippoAccountRequestFromJSON;
