// shippo/models/operations/validateaddress.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.validateAddressRequestFromJSON = exports.validateAddressRequestToJSON = exports.ValidateAddressRequest$ = exports.ValidateAddressRequest$outboundSchema = exports.ValidateAddressRequest$inboundSchema = exports.validateAddressGlobalsFromJSON = exports.validateAddressGlobalsToJSON = exports.ValidateAddressGlobals$ = exports.ValidateAddressGlobals$outboundSchema = exports.ValidateAddressGlobals$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.ValidateAddressGlobals$inboundSchema = z.object({
    "SHIPPO-API-VERSION": z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "SHIPPO-API-VERSION": "shippoApiVersion",
    });
});
/** @internal */
exports.ValidateAddressGlobals$outboundSchema = z.object({
    shippoApiVersion: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        shippoApiVersion: "SHIPPO-API-VERSION",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ValidateAddressGlobals$;
(function (ValidateAddressGlobals$) {
    /** @deprecated use `ValidateAddressGlobals$inboundSchema` instead. */
    ValidateAddressGlobals$.inboundSchema = exports.ValidateAddressGlobals$inboundSchema;
    /** @deprecated use `ValidateAddressGlobals$outboundSchema` instead. */
    ValidateAddressGlobals$.outboundSchema = exports.ValidateAddressGlobals$outboundSchema;
})(ValidateAddressGlobals$ || (exports.ValidateAddressGlobals$ = ValidateAddressGlobals$ = {}));
function validateAddressGlobalsToJSON(validateAddressGlobals) {
    return JSON.stringify(exports.ValidateAddressGlobals$outboundSchema.parse(validateAddressGlobals));
}
exports.validateAddressGlobalsToJSON = validateAddressGlobalsToJSON;
function validateAddressGlobalsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ValidateAddressGlobals$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ValidateAddressGlobals' from JSON`);
}
exports.validateAddressGlobalsFromJSON = validateAddressGlobalsFromJSON;
/** @internal */
exports.ValidateAddressRequest$inboundSchema = z.object({
    AddressId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "AddressId": "addressId",
    });
});
/** @internal */
exports.ValidateAddressRequest$outboundSchema = z.object({
    addressId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        addressId: "AddressId",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ValidateAddressRequest$;
(function (ValidateAddressRequest$) {
    /** @deprecated use `ValidateAddressRequest$inboundSchema` instead. */
    ValidateAddressRequest$.inboundSchema = exports.ValidateAddressRequest$inboundSchema;
    /** @deprecated use `ValidateAddressRequest$outboundSchema` instead. */
    ValidateAddressRequest$.outboundSchema = exports.ValidateAddressRequest$outboundSchema;
})(ValidateAddressRequest$ || (exports.ValidateAddressRequest$ = ValidateAddressRequest$ = {}));
function validateAddressRequestToJSON(validateAddressRequest) {
    return JSON.stringify(exports.ValidateAddressRequest$outboundSchema.parse(validateAddressRequest));
}
exports.validateAddressRequestToJSON = validateAddressRequestToJSON;
function validateAddressRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ValidateAddressRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ValidateAddressRequest' from JSON`);
}
exports.validateAddressRequestFromJSON = validateAddressRequestFromJSON;
