// shippo/models/components/addressimporter.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.addressImporterFromJSON = exports.addressImporterToJSON = exports.AddressImporter$ = exports.AddressImporter$outboundSchema = exports.AddressImporter$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.AddressImporter$inboundSchema = z.object({
    name: z.string().optional(),
    company: z.string().optional(),
    street1: z.string().optional(),
    street2: z.string().optional(),
    street3: z.string().optional(),
    street_no: z.string().optional(),
    city: z.string().optional(),
    state: z.string().optional(),
    zip: z.string().optional(),
    country: z.string().optional(),
    phone: z.string().optional(),
    email: z.string().optional(),
    is_residential: z.boolean().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "street_no": "streetNo",
        "is_residential": "isResidential",
    });
});
/** @internal */
exports.AddressImporter$outboundSchema = z.object({
    name: z.string().optional(),
    company: z.string().optional(),
    street1: z.string().optional(),
    street2: z.string().optional(),
    street3: z.string().optional(),
    streetNo: z.string().optional(),
    city: z.string().optional(),
    state: z.string().optional(),
    zip: z.string().optional(),
    country: z.string().optional(),
    phone: z.string().optional(),
    email: z.string().optional(),
    isResidential: z.boolean().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        streetNo: "street_no",
        isResidential: "is_residential",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AddressImporter$;
(function (AddressImporter$) {
    /** @deprecated use `AddressImporter$inboundSchema` instead. */
    AddressImporter$.inboundSchema = exports.AddressImporter$inboundSchema;
    /** @deprecated use `AddressImporter$outboundSchema` instead. */
    AddressImporter$.outboundSchema = exports.AddressImporter$outboundSchema;
})(AddressImporter$ || (exports.AddressImporter$ = AddressImporter$ = {}));
function addressImporterToJSON(addressImporter) {
    return JSON.stringify(exports.AddressImporter$outboundSchema.parse(addressImporter));
}
exports.addressImporterToJSON = addressImporterToJSON;
function addressImporterFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.AddressImporter$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'AddressImporter' from JSON`);
}
exports.addressImporterFromJSON = addressImporterFromJSON;
