// shippo/models/components/addressvalidationresults.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.addressValidationResultsFromJSON = exports.addressValidationResultsToJSON = exports.AddressValidationResults$ = exports.AddressValidationResults$outboundSchema = exports.AddressValidationResults$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const addressvalidationresultsmessage_js_1 = require("./addressvalidationresultsmessage.js");
/** @internal */
exports.AddressValidationResults$inboundSchema = z.object({
    is_valid: z.boolean().optional(),
    messages: z.array(addressvalidationresultsmessage_js_1.AddressValidationResultsMessage$inboundSchema).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "is_valid": "isValid",
    });
});
/** @internal */
exports.AddressValidationResults$outboundSchema = z.object({
    isValid: z.boolean().optional(),
    messages: z.array(addressvalidationresultsmessage_js_1.AddressValidationResultsMessage$outboundSchema).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        isValid: "is_valid",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AddressValidationResults$;
(function (AddressValidationResults$) {
    /** @deprecated use `AddressValidationResults$inboundSchema` instead. */
    AddressValidationResults$.inboundSchema = exports.AddressValidationResults$inboundSchema;
    /** @deprecated use `AddressValidationResults$outboundSchema` instead. */
    AddressValidationResults$.outboundSchema = exports.AddressValidationResults$outboundSchema;
})(AddressValidationResults$ || (exports.AddressValidationResults$ = AddressValidationResults$ = {}));
function addressValidationResultsToJSON(addressValidationResults) {
    return JSON.stringify(exports.AddressValidationResults$outboundSchema.parse(addressValidationResults));
}
exports.addressValidationResultsToJSON = addressValidationResultsToJSON;
function addressValidationResultsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.AddressValidationResults$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'AddressValidationResults' from JSON`);
}
exports.addressValidationResultsFromJSON = addressValidationResultsFromJSON;
