// shippo/models/components/addressvalidationresultsmessage.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.addressValidationResultsMessageFromJSON = exports.addressValidationResultsMessageToJSON = exports.AddressValidationResultsMessage$ = exports.AddressValidationResultsMessage$outboundSchema = exports.AddressValidationResultsMessage$inboundSchema = void 0;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.AddressValidationResultsMessage$inboundSchema = z.object({
    code: z.string().optional(),
    source: z.string().optional(),
    text: z.string().optional(),
    type: z.string().optional(),
});
/** @internal */
exports.AddressValidationResultsMessage$outboundSchema = z.object({
    code: z.string().optional(),
    source: z.string().optional(),
    text: z.string().optional(),
    type: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AddressValidationResultsMessage$;
(function (AddressValidationResultsMessage$) {
    /** @deprecated use `AddressValidationResultsMessage$inboundSchema` instead. */
    AddressValidationResultsMessage$.inboundSchema = exports.AddressValidationResultsMessage$inboundSchema;
    /** @deprecated use `AddressValidationResultsMessage$outboundSchema` instead. */
    AddressValidationResultsMessage$.outboundSchema = exports.AddressValidationResultsMessage$outboundSchema;
})(AddressValidationResultsMessage$ || (exports.AddressValidationResultsMessage$ = AddressValidationResultsMessage$ = {}));
function addressValidationResultsMessageToJSON(addressValidationResultsMessage) {
    return JSON.stringify(exports.AddressValidationResultsMessage$outboundSchema.parse(addressValidationResultsMessage));
}
exports.addressValidationResultsMessageToJSON = addressValidationResultsMessageToJSON;
function addressValidationResultsMessageFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.AddressValidationResultsMessage$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'AddressValidationResultsMessage' from JSON`);
}
exports.addressValidationResultsMessageFromJSON = addressValidationResultsMessageFromJSON;
