// shippo/models/components/addresspaginatedlist.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.addressPaginatedListFromJSON = exports.addressPaginatedListToJSON = exports.AddressPaginatedList$ = exports.AddressPaginatedList$outboundSchema = exports.AddressPaginatedList$inboundSchema = void 0;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const address_js_1 = require("./address.js");
/** @internal */
exports.AddressPaginatedList$inboundSchema = z.object({
    next: z.string().optional(),
    previous: z.string().optional(),
    results: z.array(address_js_1.Address$inboundSchema).optional(),
});
/** @internal */
exports.AddressPaginatedList$outboundSchema = z.object({
    next: z.string().optional(),
    previous: z.string().optional(),
    results: z.array(address_js_1.Address$outboundSchema).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AddressPaginatedList$;
(function (AddressPaginatedList$) {
    /** @deprecated use `AddressPaginatedList$inboundSchema` instead. */
    AddressPaginatedList$.inboundSchema = exports.AddressPaginatedList$inboundSchema;
    /** @deprecated use `AddressPaginatedList$outboundSchema` instead. */
    AddressPaginatedList$.outboundSchema = exports.AddressPaginatedList$outboundSchema;
})(AddressPaginatedList$ || (exports.AddressPaginatedList$ = AddressPaginatedList$ = {}));
function addressPaginatedListToJSON(addressPaginatedList) {
    return JSON.stringify(exports.AddressPaginatedList$outboundSchema.parse(addressPaginatedList));
}
exports.addressPaginatedListToJSON = addressPaginatedListToJSON;
function addressPaginatedListFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.AddressPaginatedList$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'AddressPaginatedList' from JSON`);
}
exports.addressPaginatedListFromJSON = addressPaginatedListFromJSON;
