// shippo/models/components/batchshipmentcreaterequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.batchShipmentCreateRequestFromJSON = exports.batchShipmentCreateRequestToJSON = exports.BatchShipmentCreateRequest$ = exports.BatchShipmentCreateRequest$outboundSchema = exports.BatchShipmentCreateRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const shipmentcreaterequest_js_1 = require("./shipmentcreaterequest.js");
/** @internal */
exports.BatchShipmentCreateRequest$inboundSchema = z.object({
    carrier_account: z.string().optional(),
    metadata: z.string().optional(),
    servicelevel_token: z.string().optional(),
    shipment: shipmentcreaterequest_js_1.ShipmentCreateRequest$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "carrier_account": "carrierAccount",
        "servicelevel_token": "servicelevelToken",
    });
});
/** @internal */
exports.BatchShipmentCreateRequest$outboundSchema = z.object({
    carrierAccount: z.string().optional(),
    metadata: z.string().optional(),
    servicelevelToken: z.string().optional(),
    shipment: shipmentcreaterequest_js_1.ShipmentCreateRequest$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        carrierAccount: "carrier_account",
        servicelevelToken: "servicelevel_token",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BatchShipmentCreateRequest$;
(function (BatchShipmentCreateRequest$) {
    /** @deprecated use `BatchShipmentCreateRequest$inboundSchema` instead. */
    BatchShipmentCreateRequest$.inboundSchema = exports.BatchShipmentCreateRequest$inboundSchema;
    /** @deprecated use `BatchShipmentCreateRequest$outboundSchema` instead. */
    BatchShipmentCreateRequest$.outboundSchema = exports.BatchShipmentCreateRequest$outboundSchema;
})(BatchShipmentCreateRequest$ || (exports.BatchShipmentCreateRequest$ = BatchShipmentCreateRequest$ = {}));
function batchShipmentCreateRequestToJSON(batchShipmentCreateRequest) {
    return JSON.stringify(exports.BatchShipmentCreateRequest$outboundSchema.parse(batchShipmentCreateRequest));
}
exports.batchShipmentCreateRequestToJSON = batchShipmentCreateRequestToJSON;
function batchShipmentCreateRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BatchShipmentCreateRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BatchShipmentCreateRequest' from JSON`);
}
exports.batchShipmentCreateRequestFromJSON = batchShipmentCreateRequestFromJSON;
