// shippo/models/components/batchshipmentpaginatedlist.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.batchShipmentPaginatedListFromJSON = exports.batchShipmentPaginatedListToJSON = exports.BatchShipmentPaginatedList$ = exports.BatchShipmentPaginatedList$outboundSchema = exports.BatchShipmentPaginatedList$inboundSchema = void 0;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const batchshipment_js_1 = require("./batchshipment.js");
/** @internal */
exports.BatchShipmentPaginatedList$inboundSchema = z.object({
    next: z.string().optional(),
    previous: z.string().optional(),
    results: z.array(batchshipment_js_1.BatchShipment$inboundSchema).optional(),
});
/** @internal */
exports.BatchShipmentPaginatedList$outboundSchema = z.object({
    next: z.string().optional(),
    previous: z.string().optional(),
    results: z.array(batchshipment_js_1.BatchShipment$outboundSchema).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BatchShipmentPaginatedList$;
(function (BatchShipmentPaginatedList$) {
    /** @deprecated use `BatchShipmentPaginatedList$inboundSchema` instead. */
    BatchShipmentPaginatedList$.inboundSchema = exports.BatchShipmentPaginatedList$inboundSchema;
    /** @deprecated use `BatchShipmentPaginatedList$outboundSchema` instead. */
    BatchShipmentPaginatedList$.outboundSchema = exports.BatchShipmentPaginatedList$outboundSchema;
})(BatchShipmentPaginatedList$ || (exports.BatchShipmentPaginatedList$ = BatchShipmentPaginatedList$ = {}));
function batchShipmentPaginatedListToJSON(batchShipmentPaginatedList) {
    return JSON.stringify(exports.BatchShipmentPaginatedList$outboundSchema.parse(batchShipmentPaginatedList));
}
exports.batchShipmentPaginatedListToJSON = batchShipmentPaginatedListToJSON;
function batchShipmentPaginatedListFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BatchShipmentPaginatedList$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BatchShipmentPaginatedList' from JSON`);
}
exports.batchShipmentPaginatedListFromJSON = batchShipmentPaginatedListFromJSON;
