// shippo/models/components/carrieraccountpaginatedlist.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.carrierAccountPaginatedListFromJSON = exports.carrierAccountPaginatedListToJSON = exports.CarrierAccountPaginatedList$ = exports.CarrierAccountPaginatedList$outboundSchema = exports.CarrierAccountPaginatedList$inboundSchema = void 0;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const carrieraccountwithextrainfo_js_1 = require("./carrieraccountwithextrainfo.js");
/** @internal */
exports.CarrierAccountPaginatedList$inboundSchema = z.object({
    next: z.string().optional(),
    previous: z.string().optional(),
    results: z.array(carrieraccountwithextrainfo_js_1.CarrierAccountWithExtraInfo$inboundSchema).optional(),
});
/** @internal */
exports.CarrierAccountPaginatedList$outboundSchema = z.object({
    next: z.string().optional(),
    previous: z.string().optional(),
    results: z.array(carrieraccountwithextrainfo_js_1.CarrierAccountWithExtraInfo$outboundSchema).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CarrierAccountPaginatedList$;
(function (CarrierAccountPaginatedList$) {
    /** @deprecated use `CarrierAccountPaginatedList$inboundSchema` instead. */
    CarrierAccountPaginatedList$.inboundSchema = exports.CarrierAccountPaginatedList$inboundSchema;
    /** @deprecated use `CarrierAccountPaginatedList$outboundSchema` instead. */
    CarrierAccountPaginatedList$.outboundSchema = exports.CarrierAccountPaginatedList$outboundSchema;
})(CarrierAccountPaginatedList$ || (exports.CarrierAccountPaginatedList$ = CarrierAccountPaginatedList$ = {}));
function carrierAccountPaginatedListToJSON(carrierAccountPaginatedList) {
    return JSON.stringify(exports.CarrierAccountPaginatedList$outboundSchema.parse(carrierAccountPaginatedList));
}
exports.carrierAccountPaginatedListToJSON = carrierAccountPaginatedListToJSON;
function carrierAccountPaginatedListFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CarrierAccountPaginatedList$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CarrierAccountPaginatedList' from JSON`);
}
exports.carrierAccountPaginatedListFromJSON = carrierAccountPaginatedListFromJSON;
