// shippo/models/components/carrieraccountfedexcreaterequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.carrierAccountFedExCreateRequestFromJSON = exports.carrierAccountFedExCreateRequestToJSON = exports.CarrierAccountFedExCreateRequest$ = exports.CarrierAccountFedExCreateRequest$outboundSchema = exports.CarrierAccountFedExCreateRequest$inboundSchema = exports.carrierAccountFedExCreateRequestParametersFromJSON = exports.carrierAccountFedExCreateRequestParametersToJSON = exports.CarrierAccountFedExCreateRequestParameters$ = exports.CarrierAccountFedExCreateRequestParameters$outboundSchema = exports.CarrierAccountFedExCreateRequestParameters$inboundSchema = void 0;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.CarrierAccountFedExCreateRequestParameters$inboundSchema = z.object({});
/** @internal */
exports.CarrierAccountFedExCreateRequestParameters$outboundSchema = z.object({});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CarrierAccountFedExCreateRequestParameters$;
(function (CarrierAccountFedExCreateRequestParameters$) {
    /** @deprecated use `CarrierAccountFedExCreateRequestParameters$inboundSchema` instead. */
    CarrierAccountFedExCreateRequestParameters$.inboundSchema = exports.CarrierAccountFedExCreateRequestParameters$inboundSchema;
    /** @deprecated use `CarrierAccountFedExCreateRequestParameters$outboundSchema` instead. */
    CarrierAccountFedExCreateRequestParameters$.outboundSchema = exports.CarrierAccountFedExCreateRequestParameters$outboundSchema;
})(CarrierAccountFedExCreateRequestParameters$ || (exports.CarrierAccountFedExCreateRequestParameters$ = CarrierAccountFedExCreateRequestParameters$ = {}));
function carrierAccountFedExCreateRequestParametersToJSON(carrierAccountFedExCreateRequestParameters) {
    return JSON.stringify(exports.CarrierAccountFedExCreateRequestParameters$outboundSchema.parse(carrierAccountFedExCreateRequestParameters));
}
exports.carrierAccountFedExCreateRequestParametersToJSON = carrierAccountFedExCreateRequestParametersToJSON;
function carrierAccountFedExCreateRequestParametersFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CarrierAccountFedExCreateRequestParameters$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CarrierAccountFedExCreateRequestParameters' from JSON`);
}
exports.carrierAccountFedExCreateRequestParametersFromJSON = carrierAccountFedExCreateRequestParametersFromJSON;
/** @internal */
exports.CarrierAccountFedExCreateRequest$inboundSchema = z.object({
    carrier: z.literal("fedex").optional(),
    parameters: z.lazy(() => exports.CarrierAccountFedExCreateRequestParameters$inboundSchema),
});
/** @internal */
exports.CarrierAccountFedExCreateRequest$outboundSchema = z.object({
    carrier: z.literal("fedex").default("fedex"),
    parameters: z.lazy(() => exports.CarrierAccountFedExCreateRequestParameters$outboundSchema),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CarrierAccountFedExCreateRequest$;
(function (CarrierAccountFedExCreateRequest$) {
    /** @deprecated use `CarrierAccountFedExCreateRequest$inboundSchema` instead. */
    CarrierAccountFedExCreateRequest$.inboundSchema = exports.CarrierAccountFedExCreateRequest$inboundSchema;
    /** @deprecated use `CarrierAccountFedExCreateRequest$outboundSchema` instead. */
    CarrierAccountFedExCreateRequest$.outboundSchema = exports.CarrierAccountFedExCreateRequest$outboundSchema;
})(CarrierAccountFedExCreateRequest$ || (exports.CarrierAccountFedExCreateRequest$ = CarrierAccountFedExCreateRequest$ = {}));
function carrierAccountFedExCreateRequestToJSON(carrierAccountFedExCreateRequest) {
    return JSON.stringify(exports.CarrierAccountFedExCreateRequest$outboundSchema.parse(carrierAccountFedExCreateRequest));
}
exports.carrierAccountFedExCreateRequestToJSON = carrierAccountFedExCreateRequestToJSON;
function carrierAccountFedExCreateRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CarrierAccountFedExCreateRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CarrierAccountFedExCreateRequest' from JSON`);
}
exports.carrierAccountFedExCreateRequestFromJSON = carrierAccountFedExCreateRequestFromJSON;
