// shippo/models/components/carrieraccountchronopostcreaterequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.carrierAccountChronopostCreateRequestFromJSON = exports.carrierAccountChronopostCreateRequestToJSON = exports.CarrierAccountChronopostCreateRequest$ = exports.CarrierAccountChronopostCreateRequest$outboundSchema = exports.CarrierAccountChronopostCreateRequest$inboundSchema = exports.carrierAccountChronopostCreateRequestParametersFromJSON = exports.carrierAccountChronopostCreateRequestParametersToJSON = exports.CarrierAccountChronopostCreateRequestParameters$ = exports.CarrierAccountChronopostCreateRequestParameters$outboundSchema = exports.CarrierAccountChronopostCreateRequestParameters$inboundSchema = void 0;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.CarrierAccountChronopostCreateRequestParameters$inboundSchema = z.object({});
/** @internal */
exports.CarrierAccountChronopostCreateRequestParameters$outboundSchema = z.object({});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CarrierAccountChronopostCreateRequestParameters$;
(function (CarrierAccountChronopostCreateRequestParameters$) {
    /** @deprecated use `CarrierAccountChronopostCreateRequestParameters$inboundSchema` instead. */
    CarrierAccountChronopostCreateRequestParameters$.inboundSchema = exports.CarrierAccountChronopostCreateRequestParameters$inboundSchema;
    /** @deprecated use `CarrierAccountChronopostCreateRequestParameters$outboundSchema` instead. */
    CarrierAccountChronopostCreateRequestParameters$.outboundSchema = exports.CarrierAccountChronopostCreateRequestParameters$outboundSchema;
})(CarrierAccountChronopostCreateRequestParameters$ || (exports.CarrierAccountChronopostCreateRequestParameters$ = CarrierAccountChronopostCreateRequestParameters$ = {}));
function carrierAccountChronopostCreateRequestParametersToJSON(carrierAccountChronopostCreateRequestParameters) {
    return JSON.stringify(exports.CarrierAccountChronopostCreateRequestParameters$outboundSchema.parse(carrierAccountChronopostCreateRequestParameters));
}
exports.carrierAccountChronopostCreateRequestParametersToJSON = carrierAccountChronopostCreateRequestParametersToJSON;
function carrierAccountChronopostCreateRequestParametersFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CarrierAccountChronopostCreateRequestParameters$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CarrierAccountChronopostCreateRequestParameters' from JSON`);
}
exports.carrierAccountChronopostCreateRequestParametersFromJSON = carrierAccountChronopostCreateRequestParametersFromJSON;
/** @internal */
exports.CarrierAccountChronopostCreateRequest$inboundSchema = z.object({
    carrier: z.literal("chronopost").optional(),
    parameters: z.lazy(() => exports.CarrierAccountChronopostCreateRequestParameters$inboundSchema),
});
/** @internal */
exports.CarrierAccountChronopostCreateRequest$outboundSchema = z.object({
    carrier: z.literal("chronopost").default("chronopost"),
    parameters: z.lazy(() => exports.CarrierAccountChronopostCreateRequestParameters$outboundSchema),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CarrierAccountChronopostCreateRequest$;
(function (CarrierAccountChronopostCreateRequest$) {
    /** @deprecated use `CarrierAccountChronopostCreateRequest$inboundSchema` instead. */
    CarrierAccountChronopostCreateRequest$.inboundSchema = exports.CarrierAccountChronopostCreateRequest$inboundSchema;
    /** @deprecated use `CarrierAccountChronopostCreateRequest$outboundSchema` instead. */
    CarrierAccountChronopostCreateRequest$.outboundSchema = exports.CarrierAccountChronopostCreateRequest$outboundSchema;
})(CarrierAccountChronopostCreateRequest$ || (exports.CarrierAccountChronopostCreateRequest$ = CarrierAccountChronopostCreateRequest$ = {}));
function carrierAccountChronopostCreateRequestToJSON(carrierAccountChronopostCreateRequest) {
    return JSON.stringify(exports.CarrierAccountChronopostCreateRequest$outboundSchema.parse(carrierAccountChronopostCreateRequest));
}
exports.carrierAccountChronopostCreateRequestToJSON = carrierAccountChronopostCreateRequestToJSON;
function carrierAccountChronopostCreateRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CarrierAccountChronopostCreateRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CarrierAccountChronopostCreateRequest' from JSON`);
}
exports.carrierAccountChronopostCreateRequestFromJSON = carrierAccountChronopostCreateRequestFromJSON;
