// shippo/models/components/carrieraccountcanadapostcreateparameters.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.carrierAccountCanadaPostCreateParametersFromJSON = exports.carrierAccountCanadaPostCreateParametersToJSON = exports.CarrierAccountCanadaPostCreateParameters$ = exports.CarrierAccountCanadaPostCreateParameters$outboundSchema = exports.CarrierAccountCanadaPostCreateParameters$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.CarrierAccountCanadaPostCreateParameters$inboundSchema = z.object({
    canada_post_terms: z.boolean(),
    company: z.string(),
    email: z.string(),
    full_name: z.string(),
    phone: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "canada_post_terms": "canadaPostTerms",
        "full_name": "fullName",
    });
});
/** @internal */
exports.CarrierAccountCanadaPostCreateParameters$outboundSchema = z.object({
    canadaPostTerms: z.boolean(),
    company: z.string(),
    email: z.string(),
    fullName: z.string(),
    phone: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        canadaPostTerms: "canada_post_terms",
        fullName: "full_name",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CarrierAccountCanadaPostCreateParameters$;
(function (CarrierAccountCanadaPostCreateParameters$) {
    /** @deprecated use `CarrierAccountCanadaPostCreateParameters$inboundSchema` instead. */
    CarrierAccountCanadaPostCreateParameters$.inboundSchema = exports.CarrierAccountCanadaPostCreateParameters$inboundSchema;
    /** @deprecated use `CarrierAccountCanadaPostCreateParameters$outboundSchema` instead. */
    CarrierAccountCanadaPostCreateParameters$.outboundSchema = exports.CarrierAccountCanadaPostCreateParameters$outboundSchema;
})(CarrierAccountCanadaPostCreateParameters$ || (exports.CarrierAccountCanadaPostCreateParameters$ = CarrierAccountCanadaPostCreateParameters$ = {}));
function carrierAccountCanadaPostCreateParametersToJSON(carrierAccountCanadaPostCreateParameters) {
    return JSON.stringify(exports.CarrierAccountCanadaPostCreateParameters$outboundSchema.parse(carrierAccountCanadaPostCreateParameters));
}
exports.carrierAccountCanadaPostCreateParametersToJSON = carrierAccountCanadaPostCreateParametersToJSON;
function carrierAccountCanadaPostCreateParametersFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CarrierAccountCanadaPostCreateParameters$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CarrierAccountCanadaPostCreateParameters' from JSON`);
}
exports.carrierAccountCanadaPostCreateParametersFromJSON = carrierAccountCanadaPostCreateParametersFromJSON;
