// shippo/models/components/carrierparceltemplate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.carrierParcelTemplateFromJSON = exports.carrierParcelTemplateToJSON = exports.CarrierParcelTemplate$ = exports.CarrierParcelTemplate$outboundSchema = exports.CarrierParcelTemplate$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const distanceunitenum_js_1 = require("./distanceunitenum.js");
/** @internal */
exports.CarrierParcelTemplate$inboundSchema = z.object({
    carrier: z.string().optional(),
    distance_unit: distanceunitenum_js_1.DistanceUnitEnum$inboundSchema.optional(),
    height: z.string().optional(),
    is_variable_dimensions: z.boolean().optional(),
    length: z.string().optional(),
    name: z.string().optional(),
    token: z.string().optional(),
    width: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "distance_unit": "distanceUnit",
        "is_variable_dimensions": "isVariableDimensions",
    });
});
/** @internal */
exports.CarrierParcelTemplate$outboundSchema = z.object({
    carrier: z.string().optional(),
    distanceUnit: distanceunitenum_js_1.DistanceUnitEnum$outboundSchema.optional(),
    height: z.string().optional(),
    isVariableDimensions: z.boolean().optional(),
    length: z.string().optional(),
    name: z.string().optional(),
    token: z.string().optional(),
    width: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        distanceUnit: "distance_unit",
        isVariableDimensions: "is_variable_dimensions",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CarrierParcelTemplate$;
(function (CarrierParcelTemplate$) {
    /** @deprecated use `CarrierParcelTemplate$inboundSchema` instead. */
    CarrierParcelTemplate$.inboundSchema = exports.CarrierParcelTemplate$inboundSchema;
    /** @deprecated use `CarrierParcelTemplate$outboundSchema` instead. */
    CarrierParcelTemplate$.outboundSchema = exports.CarrierParcelTemplate$outboundSchema;
})(CarrierParcelTemplate$ || (exports.CarrierParcelTemplate$ = CarrierParcelTemplate$ = {}));
function carrierParcelTemplateToJSON(carrierParcelTemplate) {
    return JSON.stringify(exports.CarrierParcelTemplate$outboundSchema.parse(carrierParcelTemplate));
}
exports.carrierParcelTemplateToJSON = carrierParcelTemplateToJSON;
function carrierParcelTemplateFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CarrierParcelTemplate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CarrierParcelTemplate' from JSON`);
}
exports.carrierParcelTemplateFromJSON = carrierParcelTemplateFromJSON;
