// shippo/models/components/carrieraccountupscreaterequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.carrierAccountUPSCreateRequestFromJSON = exports.carrierAccountUPSCreateRequestToJSON = exports.CarrierAccountUPSCreateRequest$ = exports.CarrierAccountUPSCreateRequest$outboundSchema = exports.CarrierAccountUPSCreateRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const carrieraccountupscreaterequestparameters_js_1 = require("./carrieraccountupscreaterequestparameters.js");
/** @internal */
exports.CarrierAccountUPSCreateRequest$inboundSchema = z.object({
    carrier: z.literal("ups").optional(),
    parameters: carrieraccountupscreaterequestparameters_js_1.CarrierAccountUPSCreateRequestParameters$inboundSchema.optional(),
});
/** @internal */
exports.CarrierAccountUPSCreateRequest$outboundSchema = z.object({
    carrier: z.literal("ups").default("ups"),
    parameters: carrieraccountupscreaterequestparameters_js_1.CarrierAccountUPSCreateRequestParameters$outboundSchema
        .optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CarrierAccountUPSCreateRequest$;
(function (CarrierAccountUPSCreateRequest$) {
    /** @deprecated use `CarrierAccountUPSCreateRequest$inboundSchema` instead. */
    CarrierAccountUPSCreateRequest$.inboundSchema = exports.CarrierAccountUPSCreateRequest$inboundSchema;
    /** @deprecated use `CarrierAccountUPSCreateRequest$outboundSchema` instead. */
    CarrierAccountUPSCreateRequest$.outboundSchema = exports.CarrierAccountUPSCreateRequest$outboundSchema;
})(CarrierAccountUPSCreateRequest$ || (exports.CarrierAccountUPSCreateRequest$ = CarrierAccountUPSCreateRequest$ = {}));
function carrierAccountUPSCreateRequestToJSON(carrierAccountUPSCreateRequest) {
    return JSON.stringify(exports.CarrierAccountUPSCreateRequest$outboundSchema.parse(carrierAccountUPSCreateRequest));
}
exports.carrierAccountUPSCreateRequestToJSON = carrierAccountUPSCreateRequestToJSON;
function carrierAccountUPSCreateRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CarrierAccountUPSCreateRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CarrierAccountUPSCreateRequest' from JSON`);
}
exports.carrierAccountUPSCreateRequestFromJSON = carrierAccountUPSCreateRequestFromJSON;
