// shippo/models/components/carrieraccountregistrationstatus.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.carrierAccountRegistrationStatusFromJSON = exports.carrierAccountRegistrationStatusToJSON = exports.CarrierAccountRegistrationStatus$ = exports.CarrierAccountRegistrationStatus$outboundSchema = exports.CarrierAccountRegistrationStatus$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.CarrierAccountRegistrationStatus$inboundSchema = z.object({
    carrier_account: z.string().optional(),
    object_created: z.string().datetime({ offset: true }).transform(v => new Date(v)).optional(),
    object_owner: z.string().optional(),
    object_updated: z.string().datetime({ offset: true }).transform(v => new Date(v)).optional(),
    status: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "carrier_account": "carrierAccount",
        "object_created": "objectCreated",
        "object_owner": "objectOwner",
        "object_updated": "objectUpdated",
    });
});
/** @internal */
exports.CarrierAccountRegistrationStatus$outboundSchema = z.object({
    carrierAccount: z.string().optional(),
    objectCreated: z.date().transform(v => v.toISOString()).optional(),
    objectOwner: z.string().optional(),
    objectUpdated: z.date().transform(v => v.toISOString()).optional(),
    status: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        carrierAccount: "carrier_account",
        objectCreated: "object_created",
        objectOwner: "object_owner",
        objectUpdated: "object_updated",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CarrierAccountRegistrationStatus$;
(function (CarrierAccountRegistrationStatus$) {
    /** @deprecated use `CarrierAccountRegistrationStatus$inboundSchema` instead. */
    CarrierAccountRegistrationStatus$.inboundSchema = exports.CarrierAccountRegistrationStatus$inboundSchema;
    /** @deprecated use `CarrierAccountRegistrationStatus$outboundSchema` instead. */
    CarrierAccountRegistrationStatus$.outboundSchema = exports.CarrierAccountRegistrationStatus$outboundSchema;
})(CarrierAccountRegistrationStatus$ || (exports.CarrierAccountRegistrationStatus$ = CarrierAccountRegistrationStatus$ = {}));
function carrierAccountRegistrationStatusToJSON(carrierAccountRegistrationStatus) {
    return JSON.stringify(exports.CarrierAccountRegistrationStatus$outboundSchema.parse(carrierAccountRegistrationStatus));
}
exports.carrierAccountRegistrationStatusToJSON = carrierAccountRegistrationStatusToJSON;
function carrierAccountRegistrationStatusFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CarrierAccountRegistrationStatus$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CarrierAccountRegistrationStatus' from JSON`);
}
exports.carrierAccountRegistrationStatusFromJSON = carrierAccountRegistrationStatusFromJSON;
