// shippo/models/components/carrieraccountdpdukcreaterequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.carrierAccountDPDUKCreateRequestFromJSON = exports.carrierAccountDPDUKCreateRequestToJSON = exports.CarrierAccountDPDUKCreateRequest$ = exports.CarrierAccountDPDUKCreateRequest$outboundSchema = exports.CarrierAccountDPDUKCreateRequest$inboundSchema = exports.carrierAccountDPDUKCreateRequestParametersFromJSON = exports.carrierAccountDPDUKCreateRequestParametersToJSON = exports.CarrierAccountDPDUKCreateRequestParameters$ = exports.CarrierAccountDPDUKCreateRequestParameters$outboundSchema = exports.CarrierAccountDPDUKCreateRequestParameters$inboundSchema = void 0;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.CarrierAccountDPDUKCreateRequestParameters$inboundSchema = z.object({});
/** @internal */
exports.CarrierAccountDPDUKCreateRequestParameters$outboundSchema = z.object({});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CarrierAccountDPDUKCreateRequestParameters$;
(function (CarrierAccountDPDUKCreateRequestParameters$) {
    /** @deprecated use `CarrierAccountDPDUKCreateRequestParameters$inboundSchema` instead. */
    CarrierAccountDPDUKCreateRequestParameters$.inboundSchema = exports.CarrierAccountDPDUKCreateRequestParameters$inboundSchema;
    /** @deprecated use `CarrierAccountDPDUKCreateRequestParameters$outboundSchema` instead. */
    CarrierAccountDPDUKCreateRequestParameters$.outboundSchema = exports.CarrierAccountDPDUKCreateRequestParameters$outboundSchema;
})(CarrierAccountDPDUKCreateRequestParameters$ || (exports.CarrierAccountDPDUKCreateRequestParameters$ = CarrierAccountDPDUKCreateRequestParameters$ = {}));
function carrierAccountDPDUKCreateRequestParametersToJSON(carrierAccountDPDUKCreateRequestParameters) {
    return JSON.stringify(exports.CarrierAccountDPDUKCreateRequestParameters$outboundSchema.parse(carrierAccountDPDUKCreateRequestParameters));
}
exports.carrierAccountDPDUKCreateRequestParametersToJSON = carrierAccountDPDUKCreateRequestParametersToJSON;
function carrierAccountDPDUKCreateRequestParametersFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CarrierAccountDPDUKCreateRequestParameters$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CarrierAccountDPDUKCreateRequestParameters' from JSON`);
}
exports.carrierAccountDPDUKCreateRequestParametersFromJSON = carrierAccountDPDUKCreateRequestParametersFromJSON;
/** @internal */
exports.CarrierAccountDPDUKCreateRequest$inboundSchema = z.object({
    carrier: z.literal("dpd_uk").optional(),
    parameters: z.lazy(() => exports.CarrierAccountDPDUKCreateRequestParameters$inboundSchema),
});
/** @internal */
exports.CarrierAccountDPDUKCreateRequest$outboundSchema = z.object({
    carrier: z.literal("dpd_uk").default("dpd_uk"),
    parameters: z.lazy(() => exports.CarrierAccountDPDUKCreateRequestParameters$outboundSchema),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CarrierAccountDPDUKCreateRequest$;
(function (CarrierAccountDPDUKCreateRequest$) {
    /** @deprecated use `CarrierAccountDPDUKCreateRequest$inboundSchema` instead. */
    CarrierAccountDPDUKCreateRequest$.inboundSchema = exports.CarrierAccountDPDUKCreateRequest$inboundSchema;
    /** @deprecated use `CarrierAccountDPDUKCreateRequest$outboundSchema` instead. */
    CarrierAccountDPDUKCreateRequest$.outboundSchema = exports.CarrierAccountDPDUKCreateRequest$outboundSchema;
})(CarrierAccountDPDUKCreateRequest$ || (exports.CarrierAccountDPDUKCreateRequest$ = CarrierAccountDPDUKCreateRequest$ = {}));
function carrierAccountDPDUKCreateRequestToJSON(carrierAccountDPDUKCreateRequest) {
    return JSON.stringify(exports.CarrierAccountDPDUKCreateRequest$outboundSchema.parse(carrierAccountDPDUKCreateRequest));
}
exports.carrierAccountDPDUKCreateRequestToJSON = carrierAccountDPDUKCreateRequestToJSON;
function carrierAccountDPDUKCreateRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CarrierAccountDPDUKCreateRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CarrierAccountDPDUKCreateRequest' from JSON`);
}
exports.carrierAccountDPDUKCreateRequestFromJSON = carrierAccountDPDUKCreateRequestFromJSON;
