// shippo/models/components/carrieraccountdhlexpresscreaterequestparameters.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.carrierAccountDHLExpressCreateRequestParametersFromJSON = exports.carrierAccountDHLExpressCreateRequestParametersToJSON = exports.CarrierAccountDHLExpressCreateRequestParameters$ = exports.CarrierAccountDHLExpressCreateRequestParameters$outboundSchema = exports.CarrierAccountDHLExpressCreateRequestParameters$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.CarrierAccountDHLExpressCreateRequestParameters$inboundSchema = z.object({
    user_accepted_terms_and_conditions: z.boolean(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "user_accepted_terms_and_conditions": "userAcceptedTermsAndConditions",
    });
});
/** @internal */
exports.CarrierAccountDHLExpressCreateRequestParameters$outboundSchema = z.object({
    userAcceptedTermsAndConditions: z.boolean(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        userAcceptedTermsAndConditions: "user_accepted_terms_and_conditions",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CarrierAccountDHLExpressCreateRequestParameters$;
(function (CarrierAccountDHLExpressCreateRequestParameters$) {
    /** @deprecated use `CarrierAccountDHLExpressCreateRequestParameters$inboundSchema` instead. */
    CarrierAccountDHLExpressCreateRequestParameters$.inboundSchema = exports.CarrierAccountDHLExpressCreateRequestParameters$inboundSchema;
    /** @deprecated use `CarrierAccountDHLExpressCreateRequestParameters$outboundSchema` instead. */
    CarrierAccountDHLExpressCreateRequestParameters$.outboundSchema = exports.CarrierAccountDHLExpressCreateRequestParameters$outboundSchema;
})(CarrierAccountDHLExpressCreateRequestParameters$ || (exports.CarrierAccountDHLExpressCreateRequestParameters$ = CarrierAccountDHLExpressCreateRequestParameters$ = {}));
function carrierAccountDHLExpressCreateRequestParametersToJSON(carrierAccountDHLExpressCreateRequestParameters) {
    return JSON.stringify(exports.CarrierAccountDHLExpressCreateRequestParameters$outboundSchema.parse(carrierAccountDHLExpressCreateRequestParameters));
}
exports.carrierAccountDHLExpressCreateRequestParametersToJSON = carrierAccountDHLExpressCreateRequestParametersToJSON;
function carrierAccountDHLExpressCreateRequestParametersFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CarrierAccountDHLExpressCreateRequestParameters$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CarrierAccountDHLExpressCreateRequestParameters' from JSON`);
}
exports.carrierAccountDHLExpressCreateRequestParametersFromJSON = carrierAccountDHLExpressCreateRequestParametersFromJSON;
