// shippo/models/components/carrieraccountsendlecreaterequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.carrierAccountSendleCreateRequestFromJSON = exports.carrierAccountSendleCreateRequestToJSON = exports.CarrierAccountSendleCreateRequest$ = exports.CarrierAccountSendleCreateRequest$outboundSchema = exports.CarrierAccountSendleCreateRequest$inboundSchema = exports.carrierAccountSendleCreateRequestParametersFromJSON = exports.carrierAccountSendleCreateRequestParametersToJSON = exports.CarrierAccountSendleCreateRequestParameters$ = exports.CarrierAccountSendleCreateRequestParameters$outboundSchema = exports.CarrierAccountSendleCreateRequestParameters$inboundSchema = void 0;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.CarrierAccountSendleCreateRequestParameters$inboundSchema = z.object({});
/** @internal */
exports.CarrierAccountSendleCreateRequestParameters$outboundSchema = z.object({});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CarrierAccountSendleCreateRequestParameters$;
(function (CarrierAccountSendleCreateRequestParameters$) {
    /** @deprecated use `CarrierAccountSendleCreateRequestParameters$inboundSchema` instead. */
    CarrierAccountSendleCreateRequestParameters$.inboundSchema = exports.CarrierAccountSendleCreateRequestParameters$inboundSchema;
    /** @deprecated use `CarrierAccountSendleCreateRequestParameters$outboundSchema` instead. */
    CarrierAccountSendleCreateRequestParameters$.outboundSchema = exports.CarrierAccountSendleCreateRequestParameters$outboundSchema;
})(CarrierAccountSendleCreateRequestParameters$ || (exports.CarrierAccountSendleCreateRequestParameters$ = CarrierAccountSendleCreateRequestParameters$ = {}));
function carrierAccountSendleCreateRequestParametersToJSON(carrierAccountSendleCreateRequestParameters) {
    return JSON.stringify(exports.CarrierAccountSendleCreateRequestParameters$outboundSchema.parse(carrierAccountSendleCreateRequestParameters));
}
exports.carrierAccountSendleCreateRequestParametersToJSON = carrierAccountSendleCreateRequestParametersToJSON;
function carrierAccountSendleCreateRequestParametersFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CarrierAccountSendleCreateRequestParameters$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CarrierAccountSendleCreateRequestParameters' from JSON`);
}
exports.carrierAccountSendleCreateRequestParametersFromJSON = carrierAccountSendleCreateRequestParametersFromJSON;
/** @internal */
exports.CarrierAccountSendleCreateRequest$inboundSchema = z.object({
    carrier: z.literal("sendle").optional(),
    parameters: z.lazy(() => exports.CarrierAccountSendleCreateRequestParameters$inboundSchema),
});
/** @internal */
exports.CarrierAccountSendleCreateRequest$outboundSchema = z.object({
    carrier: z.literal("sendle").default("sendle"),
    parameters: z.lazy(() => exports.CarrierAccountSendleCreateRequestParameters$outboundSchema),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CarrierAccountSendleCreateRequest$;
(function (CarrierAccountSendleCreateRequest$) {
    /** @deprecated use `CarrierAccountSendleCreateRequest$inboundSchema` instead. */
    CarrierAccountSendleCreateRequest$.inboundSchema = exports.CarrierAccountSendleCreateRequest$inboundSchema;
    /** @deprecated use `CarrierAccountSendleCreateRequest$outboundSchema` instead. */
    CarrierAccountSendleCreateRequest$.outboundSchema = exports.CarrierAccountSendleCreateRequest$outboundSchema;
})(CarrierAccountSendleCreateRequest$ || (exports.CarrierAccountSendleCreateRequest$ = CarrierAccountSendleCreateRequest$ = {}));
function carrierAccountSendleCreateRequestToJSON(carrierAccountSendleCreateRequest) {
    return JSON.stringify(exports.CarrierAccountSendleCreateRequest$outboundSchema.parse(carrierAccountSendleCreateRequest));
}
exports.carrierAccountSendleCreateRequestToJSON = carrierAccountSendleCreateRequestToJSON;
function carrierAccountSendleCreateRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CarrierAccountSendleCreateRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CarrierAccountSendleCreateRequest' from JSON`);
}
exports.carrierAccountSendleCreateRequestFromJSON = carrierAccountSendleCreateRequestFromJSON;
