// shippo/models/components/carrieraccountdeutschepostcreaterequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.carrierAccountDeutschePostCreateRequestFromJSON = exports.carrierAccountDeutschePostCreateRequestToJSON = exports.CarrierAccountDeutschePostCreateRequest$ = exports.CarrierAccountDeutschePostCreateRequest$outboundSchema = exports.CarrierAccountDeutschePostCreateRequest$inboundSchema = exports.carrierAccountDeutschePostCreateRequestParametersFromJSON = exports.carrierAccountDeutschePostCreateRequestParametersToJSON = exports.CarrierAccountDeutschePostCreateRequestParameters$ = exports.CarrierAccountDeutschePostCreateRequestParameters$outboundSchema = exports.CarrierAccountDeutschePostCreateRequestParameters$inboundSchema = void 0;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.CarrierAccountDeutschePostCreateRequestParameters$inboundSchema = z.object({});
/** @internal */
exports.CarrierAccountDeutschePostCreateRequestParameters$outboundSchema = z.object({});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CarrierAccountDeutschePostCreateRequestParameters$;
(function (CarrierAccountDeutschePostCreateRequestParameters$) {
    /** @deprecated use `CarrierAccountDeutschePostCreateRequestParameters$inboundSchema` instead. */
    CarrierAccountDeutschePostCreateRequestParameters$.inboundSchema = exports.CarrierAccountDeutschePostCreateRequestParameters$inboundSchema;
    /** @deprecated use `CarrierAccountDeutschePostCreateRequestParameters$outboundSchema` instead. */
    CarrierAccountDeutschePostCreateRequestParameters$.outboundSchema = exports.CarrierAccountDeutschePostCreateRequestParameters$outboundSchema;
})(CarrierAccountDeutschePostCreateRequestParameters$ || (exports.CarrierAccountDeutschePostCreateRequestParameters$ = CarrierAccountDeutschePostCreateRequestParameters$ = {}));
function carrierAccountDeutschePostCreateRequestParametersToJSON(carrierAccountDeutschePostCreateRequestParameters) {
    return JSON.stringify(exports.CarrierAccountDeutschePostCreateRequestParameters$outboundSchema.parse(carrierAccountDeutschePostCreateRequestParameters));
}
exports.carrierAccountDeutschePostCreateRequestParametersToJSON = carrierAccountDeutschePostCreateRequestParametersToJSON;
function carrierAccountDeutschePostCreateRequestParametersFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CarrierAccountDeutschePostCreateRequestParameters$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CarrierAccountDeutschePostCreateRequestParameters' from JSON`);
}
exports.carrierAccountDeutschePostCreateRequestParametersFromJSON = carrierAccountDeutschePostCreateRequestParametersFromJSON;
/** @internal */
exports.CarrierAccountDeutschePostCreateRequest$inboundSchema = z.object({
    carrier: z.literal("deutsche_post").optional(),
    parameters: z.lazy(() => exports.CarrierAccountDeutschePostCreateRequestParameters$inboundSchema),
});
/** @internal */
exports.CarrierAccountDeutschePostCreateRequest$outboundSchema = z.object({
    carrier: z.literal("deutsche_post").default("deutsche_post"),
    parameters: z.lazy(() => exports.CarrierAccountDeutschePostCreateRequestParameters$outboundSchema),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CarrierAccountDeutschePostCreateRequest$;
(function (CarrierAccountDeutschePostCreateRequest$) {
    /** @deprecated use `CarrierAccountDeutschePostCreateRequest$inboundSchema` instead. */
    CarrierAccountDeutschePostCreateRequest$.inboundSchema = exports.CarrierAccountDeutschePostCreateRequest$inboundSchema;
    /** @deprecated use `CarrierAccountDeutschePostCreateRequest$outboundSchema` instead. */
    CarrierAccountDeutschePostCreateRequest$.outboundSchema = exports.CarrierAccountDeutschePostCreateRequest$outboundSchema;
})(CarrierAccountDeutschePostCreateRequest$ || (exports.CarrierAccountDeutschePostCreateRequest$ = CarrierAccountDeutschePostCreateRequest$ = {}));
function carrierAccountDeutschePostCreateRequestToJSON(carrierAccountDeutschePostCreateRequest) {
    return JSON.stringify(exports.CarrierAccountDeutschePostCreateRequest$outboundSchema.parse(carrierAccountDeutschePostCreateRequest));
}
exports.carrierAccountDeutschePostCreateRequestToJSON = carrierAccountDeutschePostCreateRequestToJSON;
function carrierAccountDeutschePostCreateRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CarrierAccountDeutschePostCreateRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CarrierAccountDeutschePostCreateRequest' from JSON`);
}
exports.carrierAccountDeutschePostCreateRequestFromJSON = carrierAccountDeutschePostCreateRequestFromJSON;
