// shippo/models/components/carrieraccountservicelevel.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.carrierAccountServiceLevelFromJSON = exports.carrierAccountServiceLevelToJSON = exports.CarrierAccountServiceLevel$ = exports.CarrierAccountServiceLevel$outboundSchema = exports.CarrierAccountServiceLevel$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.CarrierAccountServiceLevel$inboundSchema = z.object({
    name: z.string().optional(),
    token: z.string().optional(),
    supports_return_labels: z.boolean().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "supports_return_labels": "supportsReturnLabels",
    });
});
/** @internal */
exports.CarrierAccountServiceLevel$outboundSchema = z.object({
    name: z.string().optional(),
    token: z.string().optional(),
    supportsReturnLabels: z.boolean().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        supportsReturnLabels: "supports_return_labels",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CarrierAccountServiceLevel$;
(function (CarrierAccountServiceLevel$) {
    /** @deprecated use `CarrierAccountServiceLevel$inboundSchema` instead. */
    CarrierAccountServiceLevel$.inboundSchema = exports.CarrierAccountServiceLevel$inboundSchema;
    /** @deprecated use `CarrierAccountServiceLevel$outboundSchema` instead. */
    CarrierAccountServiceLevel$.outboundSchema = exports.CarrierAccountServiceLevel$outboundSchema;
})(CarrierAccountServiceLevel$ || (exports.CarrierAccountServiceLevel$ = CarrierAccountServiceLevel$ = {}));
function carrierAccountServiceLevelToJSON(carrierAccountServiceLevel) {
    return JSON.stringify(exports.CarrierAccountServiceLevel$outboundSchema.parse(carrierAccountServiceLevel));
}
exports.carrierAccountServiceLevelToJSON = carrierAccountServiceLevelToJSON;
function carrierAccountServiceLevelFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CarrierAccountServiceLevel$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CarrierAccountServiceLevel' from JSON`);
}
exports.carrierAccountServiceLevelFromJSON = carrierAccountServiceLevelFromJSON;
