// shippo/models/components/carrieraccountdhlexpresscreaterequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.carrierAccountDHLExpressCreateRequestFromJSON = exports.carrierAccountDHLExpressCreateRequestToJSON = exports.CarrierAccountDHLExpressCreateRequest$ = exports.CarrierAccountDHLExpressCreateRequest$outboundSchema = exports.CarrierAccountDHLExpressCreateRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const carrieraccountdhlexpresscreaterequestparameters_js_1 = require("./carrieraccountdhlexpresscreaterequestparameters.js");
/** @internal */
exports.CarrierAccountDHLExpressCreateRequest$inboundSchema = z.object({
    carrier: z.literal("dhl_express").optional(),
    parameters: carrieraccountdhlexpresscreaterequestparameters_js_1.CarrierAccountDHLExpressCreateRequestParameters$inboundSchema,
});
/** @internal */
exports.CarrierAccountDHLExpressCreateRequest$outboundSchema = z.object({
    carrier: z.literal("dhl_express").default("dhl_express"),
    parameters: carrieraccountdhlexpresscreaterequestparameters_js_1.CarrierAccountDHLExpressCreateRequestParameters$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CarrierAccountDHLExpressCreateRequest$;
(function (CarrierAccountDHLExpressCreateRequest$) {
    /** @deprecated use `CarrierAccountDHLExpressCreateRequest$inboundSchema` instead. */
    CarrierAccountDHLExpressCreateRequest$.inboundSchema = exports.CarrierAccountDHLExpressCreateRequest$inboundSchema;
    /** @deprecated use `CarrierAccountDHLExpressCreateRequest$outboundSchema` instead. */
    CarrierAccountDHLExpressCreateRequest$.outboundSchema = exports.CarrierAccountDHLExpressCreateRequest$outboundSchema;
})(CarrierAccountDHLExpressCreateRequest$ || (exports.CarrierAccountDHLExpressCreateRequest$ = CarrierAccountDHLExpressCreateRequest$ = {}));
function carrierAccountDHLExpressCreateRequestToJSON(carrierAccountDHLExpressCreateRequest) {
    return JSON.stringify(exports.CarrierAccountDHLExpressCreateRequest$outboundSchema.parse(carrierAccountDHLExpressCreateRequest));
}
exports.carrierAccountDHLExpressCreateRequestToJSON = carrierAccountDHLExpressCreateRequestToJSON;
function carrierAccountDHLExpressCreateRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CarrierAccountDHLExpressCreateRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CarrierAccountDHLExpressCreateRequest' from JSON`);
}
exports.carrierAccountDHLExpressCreateRequestFromJSON = carrierAccountDHLExpressCreateRequestFromJSON;
