// shippo/models/components/carrieraccountposteitalianecreaterequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.carrierAccountPosteItalianeCreateRequestFromJSON = exports.carrierAccountPosteItalianeCreateRequestToJSON = exports.CarrierAccountPosteItalianeCreateRequest$ = exports.CarrierAccountPosteItalianeCreateRequest$outboundSchema = exports.CarrierAccountPosteItalianeCreateRequest$inboundSchema = exports.carrierAccountPosteItalianeCreateRequestParametersFromJSON = exports.carrierAccountPosteItalianeCreateRequestParametersToJSON = exports.CarrierAccountPosteItalianeCreateRequestParameters$ = exports.CarrierAccountPosteItalianeCreateRequestParameters$outboundSchema = exports.CarrierAccountPosteItalianeCreateRequestParameters$inboundSchema = void 0;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.CarrierAccountPosteItalianeCreateRequestParameters$inboundSchema = z.object({});
/** @internal */
exports.CarrierAccountPosteItalianeCreateRequestParameters$outboundSchema = z.object({});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CarrierAccountPosteItalianeCreateRequestParameters$;
(function (CarrierAccountPosteItalianeCreateRequestParameters$) {
    /** @deprecated use `CarrierAccountPosteItalianeCreateRequestParameters$inboundSchema` instead. */
    CarrierAccountPosteItalianeCreateRequestParameters$.inboundSchema = exports.CarrierAccountPosteItalianeCreateRequestParameters$inboundSchema;
    /** @deprecated use `CarrierAccountPosteItalianeCreateRequestParameters$outboundSchema` instead. */
    CarrierAccountPosteItalianeCreateRequestParameters$.outboundSchema = exports.CarrierAccountPosteItalianeCreateRequestParameters$outboundSchema;
})(CarrierAccountPosteItalianeCreateRequestParameters$ || (exports.CarrierAccountPosteItalianeCreateRequestParameters$ = CarrierAccountPosteItalianeCreateRequestParameters$ = {}));
function carrierAccountPosteItalianeCreateRequestParametersToJSON(carrierAccountPosteItalianeCreateRequestParameters) {
    return JSON.stringify(exports.CarrierAccountPosteItalianeCreateRequestParameters$outboundSchema.parse(carrierAccountPosteItalianeCreateRequestParameters));
}
exports.carrierAccountPosteItalianeCreateRequestParametersToJSON = carrierAccountPosteItalianeCreateRequestParametersToJSON;
function carrierAccountPosteItalianeCreateRequestParametersFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CarrierAccountPosteItalianeCreateRequestParameters$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CarrierAccountPosteItalianeCreateRequestParameters' from JSON`);
}
exports.carrierAccountPosteItalianeCreateRequestParametersFromJSON = carrierAccountPosteItalianeCreateRequestParametersFromJSON;
/** @internal */
exports.CarrierAccountPosteItalianeCreateRequest$inboundSchema = z.object({
    carrier: z.literal("poste_italiane").optional(),
    parameters: z.lazy(() => exports.CarrierAccountPosteItalianeCreateRequestParameters$inboundSchema),
});
/** @internal */
exports.CarrierAccountPosteItalianeCreateRequest$outboundSchema = z.object({
    carrier: z.literal("poste_italiane").default("poste_italiane"),
    parameters: z.lazy(() => exports.CarrierAccountPosteItalianeCreateRequestParameters$outboundSchema),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CarrierAccountPosteItalianeCreateRequest$;
(function (CarrierAccountPosteItalianeCreateRequest$) {
    /** @deprecated use `CarrierAccountPosteItalianeCreateRequest$inboundSchema` instead. */
    CarrierAccountPosteItalianeCreateRequest$.inboundSchema = exports.CarrierAccountPosteItalianeCreateRequest$inboundSchema;
    /** @deprecated use `CarrierAccountPosteItalianeCreateRequest$outboundSchema` instead. */
    CarrierAccountPosteItalianeCreateRequest$.outboundSchema = exports.CarrierAccountPosteItalianeCreateRequest$outboundSchema;
})(CarrierAccountPosteItalianeCreateRequest$ || (exports.CarrierAccountPosteItalianeCreateRequest$ = CarrierAccountPosteItalianeCreateRequest$ = {}));
function carrierAccountPosteItalianeCreateRequestToJSON(carrierAccountPosteItalianeCreateRequest) {
    return JSON.stringify(exports.CarrierAccountPosteItalianeCreateRequest$outboundSchema.parse(carrierAccountPosteItalianeCreateRequest));
}
exports.carrierAccountPosteItalianeCreateRequestToJSON = carrierAccountPosteItalianeCreateRequestToJSON;
function carrierAccountPosteItalianeCreateRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CarrierAccountPosteItalianeCreateRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CarrierAccountPosteItalianeCreateRequest' from JSON`);
}
exports.carrierAccountPosteItalianeCreateRequestFromJSON = carrierAccountPosteItalianeCreateRequestFromJSON;
