// shippo/models/components/carrieraccountcorreoscreaterequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.carrierAccountCorreosCreateRequestFromJSON = exports.carrierAccountCorreosCreateRequestToJSON = exports.CarrierAccountCorreosCreateRequest$ = exports.CarrierAccountCorreosCreateRequest$outboundSchema = exports.CarrierAccountCorreosCreateRequest$inboundSchema = exports.carrierAccountCorreosCreateRequestParametersFromJSON = exports.carrierAccountCorreosCreateRequestParametersToJSON = exports.CarrierAccountCorreosCreateRequestParameters$ = exports.CarrierAccountCorreosCreateRequestParameters$outboundSchema = exports.CarrierAccountCorreosCreateRequestParameters$inboundSchema = void 0;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.CarrierAccountCorreosCreateRequestParameters$inboundSchema = z.object({});
/** @internal */
exports.CarrierAccountCorreosCreateRequestParameters$outboundSchema = z.object({});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CarrierAccountCorreosCreateRequestParameters$;
(function (CarrierAccountCorreosCreateRequestParameters$) {
    /** @deprecated use `CarrierAccountCorreosCreateRequestParameters$inboundSchema` instead. */
    CarrierAccountCorreosCreateRequestParameters$.inboundSchema = exports.CarrierAccountCorreosCreateRequestParameters$inboundSchema;
    /** @deprecated use `CarrierAccountCorreosCreateRequestParameters$outboundSchema` instead. */
    CarrierAccountCorreosCreateRequestParameters$.outboundSchema = exports.CarrierAccountCorreosCreateRequestParameters$outboundSchema;
})(CarrierAccountCorreosCreateRequestParameters$ || (exports.CarrierAccountCorreosCreateRequestParameters$ = CarrierAccountCorreosCreateRequestParameters$ = {}));
function carrierAccountCorreosCreateRequestParametersToJSON(carrierAccountCorreosCreateRequestParameters) {
    return JSON.stringify(exports.CarrierAccountCorreosCreateRequestParameters$outboundSchema.parse(carrierAccountCorreosCreateRequestParameters));
}
exports.carrierAccountCorreosCreateRequestParametersToJSON = carrierAccountCorreosCreateRequestParametersToJSON;
function carrierAccountCorreosCreateRequestParametersFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CarrierAccountCorreosCreateRequestParameters$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CarrierAccountCorreosCreateRequestParameters' from JSON`);
}
exports.carrierAccountCorreosCreateRequestParametersFromJSON = carrierAccountCorreosCreateRequestParametersFromJSON;
/** @internal */
exports.CarrierAccountCorreosCreateRequest$inboundSchema = z.object({
    carrier: z.literal("correos").optional(),
    parameters: z.lazy(() => exports.CarrierAccountCorreosCreateRequestParameters$inboundSchema),
});
/** @internal */
exports.CarrierAccountCorreosCreateRequest$outboundSchema = z.object({
    carrier: z.literal("correos").default("correos"),
    parameters: z.lazy(() => exports.CarrierAccountCorreosCreateRequestParameters$outboundSchema),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CarrierAccountCorreosCreateRequest$;
(function (CarrierAccountCorreosCreateRequest$) {
    /** @deprecated use `CarrierAccountCorreosCreateRequest$inboundSchema` instead. */
    CarrierAccountCorreosCreateRequest$.inboundSchema = exports.CarrierAccountCorreosCreateRequest$inboundSchema;
    /** @deprecated use `CarrierAccountCorreosCreateRequest$outboundSchema` instead. */
    CarrierAccountCorreosCreateRequest$.outboundSchema = exports.CarrierAccountCorreosCreateRequest$outboundSchema;
})(CarrierAccountCorreosCreateRequest$ || (exports.CarrierAccountCorreosCreateRequest$ = CarrierAccountCorreosCreateRequest$ = {}));
function carrierAccountCorreosCreateRequestToJSON(carrierAccountCorreosCreateRequest) {
    return JSON.stringify(exports.CarrierAccountCorreosCreateRequest$outboundSchema.parse(carrierAccountCorreosCreateRequest));
}
exports.carrierAccountCorreosCreateRequestToJSON = carrierAccountCorreosCreateRequestToJSON;
function carrierAccountCorreosCreateRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CarrierAccountCorreosCreateRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CarrierAccountCorreosCreateRequest' from JSON`);
}
exports.carrierAccountCorreosCreateRequestFromJSON = carrierAccountCorreosCreateRequestFromJSON;
