// shippo/models/components/carrieraccountcanadapostcreaterequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.carrierAccountCanadaPostCreateRequestFromJSON = exports.carrierAccountCanadaPostCreateRequestToJSON = exports.CarrierAccountCanadaPostCreateRequest$ = exports.CarrierAccountCanadaPostCreateRequest$outboundSchema = exports.CarrierAccountCanadaPostCreateRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const carrieraccountcanadapostcreateparameters_js_1 = require("./carrieraccountcanadapostcreateparameters.js");
/** @internal */
exports.CarrierAccountCanadaPostCreateRequest$inboundSchema = z.object({
    carrier: z.literal("canada_post").optional(),
    parameters: carrieraccountcanadapostcreateparameters_js_1.CarrierAccountCanadaPostCreateParameters$inboundSchema,
});
/** @internal */
exports.CarrierAccountCanadaPostCreateRequest$outboundSchema = z.object({
    carrier: z.literal("canada_post").default("canada_post"),
    parameters: carrieraccountcanadapostcreateparameters_js_1.CarrierAccountCanadaPostCreateParameters$outboundSchema,
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CarrierAccountCanadaPostCreateRequest$;
(function (CarrierAccountCanadaPostCreateRequest$) {
    /** @deprecated use `CarrierAccountCanadaPostCreateRequest$inboundSchema` instead. */
    CarrierAccountCanadaPostCreateRequest$.inboundSchema = exports.CarrierAccountCanadaPostCreateRequest$inboundSchema;
    /** @deprecated use `CarrierAccountCanadaPostCreateRequest$outboundSchema` instead. */
    CarrierAccountCanadaPostCreateRequest$.outboundSchema = exports.CarrierAccountCanadaPostCreateRequest$outboundSchema;
})(CarrierAccountCanadaPostCreateRequest$ || (exports.CarrierAccountCanadaPostCreateRequest$ = CarrierAccountCanadaPostCreateRequest$ = {}));
function carrierAccountCanadaPostCreateRequestToJSON(carrierAccountCanadaPostCreateRequest) {
    return JSON.stringify(exports.CarrierAccountCanadaPostCreateRequest$outboundSchema.parse(carrierAccountCanadaPostCreateRequest));
}
exports.carrierAccountCanadaPostCreateRequestToJSON = carrierAccountCanadaPostCreateRequestToJSON;
function carrierAccountCanadaPostCreateRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CarrierAccountCanadaPostCreateRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CarrierAccountCanadaPostCreateRequest' from JSON`);
}
exports.carrierAccountCanadaPostCreateRequestFromJSON = carrierAccountCanadaPostCreateRequestFromJSON;
