// shippo/models/components/cod.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.codFromJSON = exports.codToJSON = exports.Cod$ = exports.Cod$outboundSchema = exports.Cod$inboundSchema = exports.PaymentMethod$ = exports.PaymentMethod$outboundSchema = exports.PaymentMethod$inboundSchema = exports.PaymentMethod = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/**
 * Secured funds include money orders, certified cheques and others (see
 *
 * @remarks
 * <a href="https://www.ups.com/content/us/en/shipping/time/service/value_added/cod.html">UPS</a> for details).
 * If no payment_method inputted the value defaults to "ANY".)
 */
exports.PaymentMethod = {
    SecuredFunds: "SECURED_FUNDS",
    Cash: "CASH",
    Any: "ANY",
};
/** @internal */
exports.PaymentMethod$inboundSchema = z.nativeEnum(exports.PaymentMethod);
/** @internal */
exports.PaymentMethod$outboundSchema = exports.PaymentMethod$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var PaymentMethod$;
(function (PaymentMethod$) {
    /** @deprecated use `PaymentMethod$inboundSchema` instead. */
    PaymentMethod$.inboundSchema = exports.PaymentMethod$inboundSchema;
    /** @deprecated use `PaymentMethod$outboundSchema` instead. */
    PaymentMethod$.outboundSchema = exports.PaymentMethod$outboundSchema;
})(PaymentMethod$ || (exports.PaymentMethod$ = PaymentMethod$ = {}));
/** @internal */
exports.Cod$inboundSchema = z
    .object({
    amount: z.string().optional(),
    currency: z.string().optional(),
    payment_method: exports.PaymentMethod$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "payment_method": "paymentMethod",
    });
});
/** @internal */
exports.Cod$outboundSchema = z
    .object({
    amount: z.string().optional(),
    currency: z.string().optional(),
    paymentMethod: exports.PaymentMethod$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        paymentMethod: "payment_method",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Cod$;
(function (Cod$) {
    /** @deprecated use `Cod$inboundSchema` instead. */
    Cod$.inboundSchema = exports.Cod$inboundSchema;
    /** @deprecated use `Cod$outboundSchema` instead. */
    Cod$.outboundSchema = exports.Cod$outboundSchema;
})(Cod$ || (exports.Cod$ = Cod$ = {}));
function codToJSON(cod) {
    return JSON.stringify(exports.Cod$outboundSchema.parse(cod));
}
exports.codToJSON = codToJSON;
function codFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Cod$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Cod' from JSON`);
}
exports.codFromJSON = codFromJSON;
