// shippo/models/components/corerate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.coreRateFromJSON = exports.coreRateToJSON = exports.CoreRate$ = exports.CoreRate$outboundSchema = exports.CoreRate$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.CoreRate$inboundSchema = z.object({
    amount: z.string().optional(),
    amount_local: z.string().optional(),
    currency: z.string().optional(),
    currency_local: z.string().optional(),
    object_id: z.string().optional(),
    provider: z.string().optional(),
    carrier_account: z.string().optional(),
    servicelevel_name: z.string().optional(),
    servicelevel_token: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "amount_local": "amountLocal",
        "currency_local": "currencyLocal",
        "object_id": "objectId",
        "carrier_account": "carrierAccount",
        "servicelevel_name": "servicelevelName",
        "servicelevel_token": "servicelevelToken",
    });
});
/** @internal */
exports.CoreRate$outboundSchema = z.object({
    amount: z.string().optional(),
    amountLocal: z.string().optional(),
    currency: z.string().optional(),
    currencyLocal: z.string().optional(),
    objectId: z.string().optional(),
    provider: z.string().optional(),
    carrierAccount: z.string().optional(),
    servicelevelName: z.string().optional(),
    servicelevelToken: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        amountLocal: "amount_local",
        currencyLocal: "currency_local",
        objectId: "object_id",
        carrierAccount: "carrier_account",
        servicelevelName: "servicelevel_name",
        servicelevelToken: "servicelevel_token",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CoreRate$;
(function (CoreRate$) {
    /** @deprecated use `CoreRate$inboundSchema` instead. */
    CoreRate$.inboundSchema = exports.CoreRate$inboundSchema;
    /** @deprecated use `CoreRate$outboundSchema` instead. */
    CoreRate$.outboundSchema = exports.CoreRate$outboundSchema;
})(CoreRate$ || (exports.CoreRate$ = CoreRate$ = {}));
function coreRateToJSON(coreRate) {
    return JSON.stringify(exports.CoreRate$outboundSchema.parse(coreRate));
}
exports.coreRateToJSON = coreRateToJSON;
function coreRateFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CoreRate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CoreRate' from JSON`);
}
exports.coreRateFromJSON = coreRateFromJSON;
