// shippo/models/components/customsexporteridentification.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.customsExporterIdentificationFromJSON = exports.customsExporterIdentificationToJSON = exports.CustomsExporterIdentification$ = exports.CustomsExporterIdentification$outboundSchema = exports.CustomsExporterIdentification$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const customstaxidentification_js_1 = require("./customstaxidentification.js");
/** @internal */
exports.CustomsExporterIdentification$inboundSchema = z.object({
    eori_number: z.string().optional(),
    tax_id: customstaxidentification_js_1.CustomsTaxIdentification$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "eori_number": "eoriNumber",
        "tax_id": "taxId",
    });
});
/** @internal */
exports.CustomsExporterIdentification$outboundSchema = z.object({
    eoriNumber: z.string().optional(),
    taxId: customstaxidentification_js_1.CustomsTaxIdentification$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        eoriNumber: "eori_number",
        taxId: "tax_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomsExporterIdentification$;
(function (CustomsExporterIdentification$) {
    /** @deprecated use `CustomsExporterIdentification$inboundSchema` instead. */
    CustomsExporterIdentification$.inboundSchema = exports.CustomsExporterIdentification$inboundSchema;
    /** @deprecated use `CustomsExporterIdentification$outboundSchema` instead. */
    CustomsExporterIdentification$.outboundSchema = exports.CustomsExporterIdentification$outboundSchema;
})(CustomsExporterIdentification$ || (exports.CustomsExporterIdentification$ = CustomsExporterIdentification$ = {}));
function customsExporterIdentificationToJSON(customsExporterIdentification) {
    return JSON.stringify(exports.CustomsExporterIdentification$outboundSchema.parse(customsExporterIdentification));
}
exports.customsExporterIdentificationToJSON = customsExporterIdentificationToJSON;
function customsExporterIdentificationFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomsExporterIdentification$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomsExporterIdentification' from JSON`);
}
exports.customsExporterIdentificationFromJSON = customsExporterIdentificationFromJSON;
