// shippo/models/components/customsdeclarationnondeliveryoptionenum.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomsDeclarationNonDeliveryOptionEnum$ = exports.CustomsDeclarationNonDeliveryOptionEnum$outboundSchema = exports.CustomsDeclarationNonDeliveryOptionEnum$inboundSchema = exports.CustomsDeclarationNonDeliveryOptionEnum = void 0;
const z = __importStar(require("zod"));
/**
 * Indicates how the carrier should proceed in case the shipment can't be delivered.
 *
 * @remarks
 * Allowed values available <a href="#tag/Customs-Declaration-Non-Delivery-Option">here</a>
 */
exports.CustomsDeclarationNonDeliveryOptionEnum = {
    Abandon: "ABANDON",
    Return: "RETURN",
};
/** @internal */
exports.CustomsDeclarationNonDeliveryOptionEnum$inboundSchema = z
    .nativeEnum(exports.CustomsDeclarationNonDeliveryOptionEnum);
/** @internal */
exports.CustomsDeclarationNonDeliveryOptionEnum$outboundSchema = exports.CustomsDeclarationNonDeliveryOptionEnum$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomsDeclarationNonDeliveryOptionEnum$;
(function (CustomsDeclarationNonDeliveryOptionEnum$) {
    /** @deprecated use `CustomsDeclarationNonDeliveryOptionEnum$inboundSchema` instead. */
    CustomsDeclarationNonDeliveryOptionEnum$.inboundSchema = exports.CustomsDeclarationNonDeliveryOptionEnum$inboundSchema;
    /** @deprecated use `CustomsDeclarationNonDeliveryOptionEnum$outboundSchema` instead. */
    CustomsDeclarationNonDeliveryOptionEnum$.outboundSchema = exports.CustomsDeclarationNonDeliveryOptionEnum$outboundSchema;
})(CustomsDeclarationNonDeliveryOptionEnum$ || (exports.CustomsDeclarationNonDeliveryOptionEnum$ = CustomsDeclarationNonDeliveryOptionEnum$ = {}));
