// shippo/models/components/customsdeclarationb13afilingoptionenum.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomsDeclarationB13AFilingOptionEnum$ = exports.CustomsDeclarationB13AFilingOptionEnum$outboundSchema = exports.CustomsDeclarationB13AFilingOptionEnum$inboundSchema = exports.CustomsDeclarationB13AFilingOptionEnum = void 0;
const z = __importStar(require("zod"));
/**
 * B13A Option details are obtained by filing a B13A Canada Export Declaration via the Canadian Export Reporting System (CERS).
 *
 * @remarks
 * <a href="https://www.cbsa-asfc.gc.ca/services/export/guide-eng.html" target="_blank" rel="noopener noreferrer"> More information on reporting commercial exports from Canada. </a>
 * Allowed values available <a href="#tag/Customs-Declaration-B13A-Filing-Option">here</a>
 */
exports.CustomsDeclarationB13AFilingOptionEnum = {
    FiledElectronically: "FILED_ELECTRONICALLY",
    SummaryReporting: "SUMMARY_REPORTING",
    NotRequired: "NOT_REQUIRED",
};
/** @internal */
exports.CustomsDeclarationB13AFilingOptionEnum$inboundSchema = z.nativeEnum(exports.CustomsDeclarationB13AFilingOptionEnum);
/** @internal */
exports.CustomsDeclarationB13AFilingOptionEnum$outboundSchema = exports.CustomsDeclarationB13AFilingOptionEnum$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomsDeclarationB13AFilingOptionEnum$;
(function (CustomsDeclarationB13AFilingOptionEnum$) {
    /** @deprecated use `CustomsDeclarationB13AFilingOptionEnum$inboundSchema` instead. */
    CustomsDeclarationB13AFilingOptionEnum$.inboundSchema = exports.CustomsDeclarationB13AFilingOptionEnum$inboundSchema;
    /** @deprecated use `CustomsDeclarationB13AFilingOptionEnum$outboundSchema` instead. */
    CustomsDeclarationB13AFilingOptionEnum$.outboundSchema = exports.CustomsDeclarationB13AFilingOptionEnum$outboundSchema;
})(CustomsDeclarationB13AFilingOptionEnum$ || (exports.CustomsDeclarationB13AFilingOptionEnum$ = CustomsDeclarationB13AFilingOptionEnum$ = {}));
