// shippo/models/components/customsdeclarationpaginatedlist.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.customsDeclarationPaginatedListFromJSON = exports.customsDeclarationPaginatedListToJSON = exports.CustomsDeclarationPaginatedList$ = exports.CustomsDeclarationPaginatedList$outboundSchema = exports.CustomsDeclarationPaginatedList$inboundSchema = void 0;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const customsdeclaration_js_1 = require("./customsdeclaration.js");
/** @internal */
exports.CustomsDeclarationPaginatedList$inboundSchema = z.object({
    next: z.string().optional(),
    previous: z.string().optional(),
    results: z.array(customsdeclaration_js_1.CustomsDeclaration$inboundSchema).optional(),
});
/** @internal */
exports.CustomsDeclarationPaginatedList$outboundSchema = z.object({
    next: z.string().optional(),
    previous: z.string().optional(),
    results: z.array(customsdeclaration_js_1.CustomsDeclaration$outboundSchema).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomsDeclarationPaginatedList$;
(function (CustomsDeclarationPaginatedList$) {
    /** @deprecated use `CustomsDeclarationPaginatedList$inboundSchema` instead. */
    CustomsDeclarationPaginatedList$.inboundSchema = exports.CustomsDeclarationPaginatedList$inboundSchema;
    /** @deprecated use `CustomsDeclarationPaginatedList$outboundSchema` instead. */
    CustomsDeclarationPaginatedList$.outboundSchema = exports.CustomsDeclarationPaginatedList$outboundSchema;
})(CustomsDeclarationPaginatedList$ || (exports.CustomsDeclarationPaginatedList$ = CustomsDeclarationPaginatedList$ = {}));
function customsDeclarationPaginatedListToJSON(customsDeclarationPaginatedList) {
    return JSON.stringify(exports.CustomsDeclarationPaginatedList$outboundSchema.parse(customsDeclarationPaginatedList));
}
exports.customsDeclarationPaginatedListToJSON = customsDeclarationPaginatedListToJSON;
function customsDeclarationPaginatedListFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomsDeclarationPaginatedList$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomsDeclarationPaginatedList' from JSON`);
}
exports.customsDeclarationPaginatedListFromJSON = customsDeclarationPaginatedListFromJSON;
