// shippo/models/components/customsitempaginatedlist.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.customsItemPaginatedListFromJSON = exports.customsItemPaginatedListToJSON = exports.CustomsItemPaginatedList$ = exports.CustomsItemPaginatedList$outboundSchema = exports.CustomsItemPaginatedList$inboundSchema = void 0;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const customsitem_js_1 = require("./customsitem.js");
/** @internal */
exports.CustomsItemPaginatedList$inboundSchema = z.object({
    next: z.string().optional(),
    previous: z.string().optional(),
    results: z.array(customsitem_js_1.CustomsItem$inboundSchema).optional(),
});
/** @internal */
exports.CustomsItemPaginatedList$outboundSchema = z.object({
    next: z.string().optional(),
    previous: z.string().optional(),
    results: z.array(customsitem_js_1.CustomsItem$outboundSchema).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomsItemPaginatedList$;
(function (CustomsItemPaginatedList$) {
    /** @deprecated use `CustomsItemPaginatedList$inboundSchema` instead. */
    CustomsItemPaginatedList$.inboundSchema = exports.CustomsItemPaginatedList$inboundSchema;
    /** @deprecated use `CustomsItemPaginatedList$outboundSchema` instead. */
    CustomsItemPaginatedList$.outboundSchema = exports.CustomsItemPaginatedList$outboundSchema;
})(CustomsItemPaginatedList$ || (exports.CustomsItemPaginatedList$ = CustomsItemPaginatedList$ = {}));
function customsItemPaginatedListToJSON(customsItemPaginatedList) {
    return JSON.stringify(exports.CustomsItemPaginatedList$outboundSchema.parse(customsItemPaginatedList));
}
exports.customsItemPaginatedListToJSON = customsItemPaginatedListToJSON;
function customsItemPaginatedListFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomsItemPaginatedList$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomsItemPaginatedList' from JSON`);
}
exports.customsItemPaginatedListFromJSON = customsItemPaginatedListFromJSON;
