// shippo/models/components/customsdeclarationincotermenum.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomsDeclarationIncotermEnum$ = exports.CustomsDeclarationIncotermEnum$outboundSchema = exports.CustomsDeclarationIncotermEnum$inboundSchema = exports.CustomsDeclarationIncotermEnum = void 0;
const z = __importStar(require("zod"));
/**
 * The incoterm reference of the shipment. FCA is available for DHL Express and FedEx only.
 *
 * @remarks
 * eDAP is available for DPD UK only. DAP is available for DHL Express and DPD UK.
 * If expecting DAP for other carriers, please use DDU.
 * Allowed values available <a href="#tag/Customs-Declaration-Incoterm">here</a>
 */
exports.CustomsDeclarationIncotermEnum = {
    Ddp: "DDP",
    Ddu: "DDU",
    Fca: "FCA",
    Dap: "DAP",
    EDAP: "eDAP",
};
/** @internal */
exports.CustomsDeclarationIncotermEnum$inboundSchema = z.nativeEnum(exports.CustomsDeclarationIncotermEnum);
/** @internal */
exports.CustomsDeclarationIncotermEnum$outboundSchema = exports.CustomsDeclarationIncotermEnum$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomsDeclarationIncotermEnum$;
(function (CustomsDeclarationIncotermEnum$) {
    /** @deprecated use `CustomsDeclarationIncotermEnum$inboundSchema` instead. */
    CustomsDeclarationIncotermEnum$.inboundSchema = exports.CustomsDeclarationIncotermEnum$inboundSchema;
    /** @deprecated use `CustomsDeclarationIncotermEnum$outboundSchema` instead. */
    CustomsDeclarationIncotermEnum$.outboundSchema = exports.CustomsDeclarationIncotermEnum$outboundSchema;
})(CustomsDeclarationIncotermEnum$ || (exports.CustomsDeclarationIncotermEnum$ = CustomsDeclarationIncotermEnum$ = {}));
