// shippo/models/components/customsitem.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.customsItemFromJSON = exports.customsItemToJSON = exports.CustomsItem$ = exports.CustomsItem$outboundSchema = exports.CustomsItem$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const objectstateenum_js_1 = require("./objectstateenum.js");
const weightunitenum_js_1 = require("./weightunitenum.js");
/** @internal */
exports.CustomsItem$inboundSchema = z.object({
    description: z.string(),
    eccn_ear99: z.string().optional(),
    mass_unit: weightunitenum_js_1.WeightUnitEnum$inboundSchema,
    metadata: z.string().optional(),
    net_weight: z.string(),
    origin_country: z.string(),
    quantity: z.number().int(),
    sku_code: z.string().optional(),
    hs_code: z.string().optional(),
    tariff_number: z.string().optional(),
    value_amount: z.string(),
    value_currency: z.string(),
    object_created: z.string().datetime({ offset: true }).transform(v => new Date(v)).optional(),
    object_id: z.string().optional(),
    object_owner: z.string().optional(),
    object_state: objectstateenum_js_1.ObjectStateEnum$inboundSchema.optional(),
    object_updated: z.string().datetime({ offset: true }).transform(v => new Date(v)).optional(),
    test: z.boolean().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "eccn_ear99": "eccnEar99",
        "mass_unit": "massUnit",
        "net_weight": "netWeight",
        "origin_country": "originCountry",
        "sku_code": "skuCode",
        "hs_code": "hsCode",
        "tariff_number": "tariffNumber",
        "value_amount": "valueAmount",
        "value_currency": "valueCurrency",
        "object_created": "objectCreated",
        "object_id": "objectId",
        "object_owner": "objectOwner",
        "object_state": "objectState",
        "object_updated": "objectUpdated",
    });
});
/** @internal */
exports.CustomsItem$outboundSchema = z.object({
    description: z.string(),
    eccnEar99: z.string().optional(),
    massUnit: weightunitenum_js_1.WeightUnitEnum$outboundSchema,
    metadata: z.string().optional(),
    netWeight: z.string(),
    originCountry: z.string(),
    quantity: z.number().int(),
    skuCode: z.string().optional(),
    hsCode: z.string().optional(),
    tariffNumber: z.string().optional(),
    valueAmount: z.string(),
    valueCurrency: z.string(),
    objectCreated: z.date().transform(v => v.toISOString()).optional(),
    objectId: z.string().optional(),
    objectOwner: z.string().optional(),
    objectState: objectstateenum_js_1.ObjectStateEnum$outboundSchema.optional(),
    objectUpdated: z.date().transform(v => v.toISOString()).optional(),
    test: z.boolean().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        eccnEar99: "eccn_ear99",
        massUnit: "mass_unit",
        netWeight: "net_weight",
        originCountry: "origin_country",
        skuCode: "sku_code",
        hsCode: "hs_code",
        tariffNumber: "tariff_number",
        valueAmount: "value_amount",
        valueCurrency: "value_currency",
        objectCreated: "object_created",
        objectId: "object_id",
        objectOwner: "object_owner",
        objectState: "object_state",
        objectUpdated: "object_updated",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomsItem$;
(function (CustomsItem$) {
    /** @deprecated use `CustomsItem$inboundSchema` instead. */
    CustomsItem$.inboundSchema = exports.CustomsItem$inboundSchema;
    /** @deprecated use `CustomsItem$outboundSchema` instead. */
    CustomsItem$.outboundSchema = exports.CustomsItem$outboundSchema;
})(CustomsItem$ || (exports.CustomsItem$ = CustomsItem$ = {}));
function customsItemToJSON(customsItem) {
    return JSON.stringify(exports.CustomsItem$outboundSchema.parse(customsItem));
}
exports.customsItemToJSON = customsItemToJSON;
function customsItemFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomsItem$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomsItem' from JSON`);
}
exports.customsItemFromJSON = customsItemFromJSON;
