// shippo/models/components/customsitemcreaterequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.customsItemCreateRequestFromJSON = exports.customsItemCreateRequestToJSON = exports.CustomsItemCreateRequest$ = exports.CustomsItemCreateRequest$outboundSchema = exports.CustomsItemCreateRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const weightunitenum_js_1 = require("./weightunitenum.js");
/** @internal */
exports.CustomsItemCreateRequest$inboundSchema = z.object({
    description: z.string(),
    eccn_ear99: z.string().optional(),
    mass_unit: weightunitenum_js_1.WeightUnitEnum$inboundSchema,
    metadata: z.string().optional(),
    net_weight: z.string(),
    origin_country: z.string(),
    quantity: z.number().int(),
    sku_code: z.string().optional(),
    hs_code: z.string().optional(),
    tariff_number: z.string().optional(),
    value_amount: z.string(),
    value_currency: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "eccn_ear99": "eccnEar99",
        "mass_unit": "massUnit",
        "net_weight": "netWeight",
        "origin_country": "originCountry",
        "sku_code": "skuCode",
        "hs_code": "hsCode",
        "tariff_number": "tariffNumber",
        "value_amount": "valueAmount",
        "value_currency": "valueCurrency",
    });
});
/** @internal */
exports.CustomsItemCreateRequest$outboundSchema = z.object({
    description: z.string(),
    eccnEar99: z.string().optional(),
    massUnit: weightunitenum_js_1.WeightUnitEnum$outboundSchema,
    metadata: z.string().optional(),
    netWeight: z.string(),
    originCountry: z.string(),
    quantity: z.number().int(),
    skuCode: z.string().optional(),
    hsCode: z.string().optional(),
    tariffNumber: z.string().optional(),
    valueAmount: z.string(),
    valueCurrency: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        eccnEar99: "eccn_ear99",
        massUnit: "mass_unit",
        netWeight: "net_weight",
        originCountry: "origin_country",
        skuCode: "sku_code",
        hsCode: "hs_code",
        tariffNumber: "tariff_number",
        valueAmount: "value_amount",
        valueCurrency: "value_currency",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomsItemCreateRequest$;
(function (CustomsItemCreateRequest$) {
    /** @deprecated use `CustomsItemCreateRequest$inboundSchema` instead. */
    CustomsItemCreateRequest$.inboundSchema = exports.CustomsItemCreateRequest$inboundSchema;
    /** @deprecated use `CustomsItemCreateRequest$outboundSchema` instead. */
    CustomsItemCreateRequest$.outboundSchema = exports.CustomsItemCreateRequest$outboundSchema;
})(CustomsItemCreateRequest$ || (exports.CustomsItemCreateRequest$ = CustomsItemCreateRequest$ = {}));
function customsItemCreateRequestToJSON(customsItemCreateRequest) {
    return JSON.stringify(exports.CustomsItemCreateRequest$outboundSchema.parse(customsItemCreateRequest));
}
exports.customsItemCreateRequestToJSON = customsItemCreateRequestToJSON;
function customsItemCreateRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomsItemCreateRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomsItemCreateRequest' from JSON`);
}
exports.customsItemCreateRequestFromJSON = customsItemCreateRequestFromJSON;
