// shippo/models/components/customsdeclarationeelpfcenum.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomsDeclarationEelPfcEnum$ = exports.CustomsDeclarationEelPfcEnum$outboundSchema = exports.CustomsDeclarationEelPfcEnum$inboundSchema = exports.CustomsDeclarationEelPfcEnum = void 0;
const z = __importStar(require("zod"));
/**
 * EEL / PFC type of the shipment. For most shipments from the US to CA, `NOEEI_30_36` is applicable; for most
 *
 * @remarks
 * other shipments from the US, `NOEEI_30_37_a` is applicable.
 * Allowed values available <a href="#tag/Customs-Declaration-EELPFC">here</a>
 */
exports.CustomsDeclarationEelPfcEnum = {
    NOEEI3037A: "NOEEI_30_37_a",
    NOEEI3037H: "NOEEI_30_37_h",
    NOEEI3037F: "NOEEI_30_37_f",
    Noeei3036: "NOEEI_30_36",
    AesItn: "AES_ITN",
};
/** @internal */
exports.CustomsDeclarationEelPfcEnum$inboundSchema = z.nativeEnum(exports.CustomsDeclarationEelPfcEnum);
/** @internal */
exports.CustomsDeclarationEelPfcEnum$outboundSchema = exports.CustomsDeclarationEelPfcEnum$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomsDeclarationEelPfcEnum$;
(function (CustomsDeclarationEelPfcEnum$) {
    /** @deprecated use `CustomsDeclarationEelPfcEnum$inboundSchema` instead. */
    CustomsDeclarationEelPfcEnum$.inboundSchema = exports.CustomsDeclarationEelPfcEnum$inboundSchema;
    /** @deprecated use `CustomsDeclarationEelPfcEnum$outboundSchema` instead. */
    CustomsDeclarationEelPfcEnum$.outboundSchema = exports.CustomsDeclarationEelPfcEnum$outboundSchema;
})(CustomsDeclarationEelPfcEnum$ || (exports.CustomsDeclarationEelPfcEnum$ = CustomsDeclarationEelPfcEnum$ = {}));
