// shippo/models/components/dangerousgoodsobject.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.dangerousGoodsObjectFromJSON = exports.dangerousGoodsObjectToJSON = exports.DangerousGoodsObject$ = exports.DangerousGoodsObject$outboundSchema = exports.DangerousGoodsObject$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const dangerousgoodsbiologicalmaterial_js_1 = require("./dangerousgoodsbiologicalmaterial.js");
const dangerousgoodslithiumbatteries_js_1 = require("./dangerousgoodslithiumbatteries.js");
/** @internal */
exports.DangerousGoodsObject$inboundSchema = z.object({
    contains: z.boolean().optional(),
    biological_material: dangerousgoodsbiologicalmaterial_js_1.DangerousGoodsBiologicalMaterial$inboundSchema
        .optional(),
    lithium_batteries: dangerousgoodslithiumbatteries_js_1.DangerousGoodsLithiumBatteries$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "biological_material": "biologicalMaterial",
        "lithium_batteries": "lithiumBatteries",
    });
});
/** @internal */
exports.DangerousGoodsObject$outboundSchema = z.object({
    contains: z.boolean().optional(),
    biologicalMaterial: dangerousgoodsbiologicalmaterial_js_1.DangerousGoodsBiologicalMaterial$outboundSchema
        .optional(),
    lithiumBatteries: dangerousgoodslithiumbatteries_js_1.DangerousGoodsLithiumBatteries$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        biologicalMaterial: "biological_material",
        lithiumBatteries: "lithium_batteries",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DangerousGoodsObject$;
(function (DangerousGoodsObject$) {
    /** @deprecated use `DangerousGoodsObject$inboundSchema` instead. */
    DangerousGoodsObject$.inboundSchema = exports.DangerousGoodsObject$inboundSchema;
    /** @deprecated use `DangerousGoodsObject$outboundSchema` instead. */
    DangerousGoodsObject$.outboundSchema = exports.DangerousGoodsObject$outboundSchema;
})(DangerousGoodsObject$ || (exports.DangerousGoodsObject$ = DangerousGoodsObject$ = {}));
function dangerousGoodsObjectToJSON(dangerousGoodsObject) {
    return JSON.stringify(exports.DangerousGoodsObject$outboundSchema.parse(dangerousGoodsObject));
}
exports.dangerousGoodsObjectToJSON = dangerousGoodsObjectToJSON;
function dangerousGoodsObjectFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.DangerousGoodsObject$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DangerousGoodsObject' from JSON`);
}
exports.dangerousGoodsObjectFromJSON = dangerousGoodsObjectFromJSON;
