// shippo/models/components/dangerousgoodsbiologicalmaterial.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.dangerousGoodsBiologicalMaterialFromJSON = exports.dangerousGoodsBiologicalMaterialToJSON = exports.DangerousGoodsBiologicalMaterial$ = exports.DangerousGoodsBiologicalMaterial$outboundSchema = exports.DangerousGoodsBiologicalMaterial$inboundSchema = void 0;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.DangerousGoodsBiologicalMaterial$inboundSchema = z.object({
    contains: z.boolean().optional(),
});
/** @internal */
exports.DangerousGoodsBiologicalMaterial$outboundSchema = z.object({
    contains: z.boolean().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DangerousGoodsBiologicalMaterial$;
(function (DangerousGoodsBiologicalMaterial$) {
    /** @deprecated use `DangerousGoodsBiologicalMaterial$inboundSchema` instead. */
    DangerousGoodsBiologicalMaterial$.inboundSchema = exports.DangerousGoodsBiologicalMaterial$inboundSchema;
    /** @deprecated use `DangerousGoodsBiologicalMaterial$outboundSchema` instead. */
    DangerousGoodsBiologicalMaterial$.outboundSchema = exports.DangerousGoodsBiologicalMaterial$outboundSchema;
})(DangerousGoodsBiologicalMaterial$ || (exports.DangerousGoodsBiologicalMaterial$ = DangerousGoodsBiologicalMaterial$ = {}));
function dangerousGoodsBiologicalMaterialToJSON(dangerousGoodsBiologicalMaterial) {
    return JSON.stringify(exports.DangerousGoodsBiologicalMaterial$outboundSchema.parse(dangerousGoodsBiologicalMaterial));
}
exports.dangerousGoodsBiologicalMaterialToJSON = dangerousGoodsBiologicalMaterialToJSON;
function dangerousGoodsBiologicalMaterialFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.DangerousGoodsBiologicalMaterial$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DangerousGoodsBiologicalMaterial' from JSON`);
}
exports.dangerousGoodsBiologicalMaterialFromJSON = dangerousGoodsBiologicalMaterialFromJSON;
