// shippo/models/components/dangerousgoodslithiumbatteries.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.dangerousGoodsLithiumBatteriesFromJSON = exports.dangerousGoodsLithiumBatteriesToJSON = exports.DangerousGoodsLithiumBatteries$ = exports.DangerousGoodsLithiumBatteries$outboundSchema = exports.DangerousGoodsLithiumBatteries$inboundSchema = void 0;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.DangerousGoodsLithiumBatteries$inboundSchema = z.object({
    contains: z.boolean().optional(),
});
/** @internal */
exports.DangerousGoodsLithiumBatteries$outboundSchema = z.object({
    contains: z.boolean().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DangerousGoodsLithiumBatteries$;
(function (DangerousGoodsLithiumBatteries$) {
    /** @deprecated use `DangerousGoodsLithiumBatteries$inboundSchema` instead. */
    DangerousGoodsLithiumBatteries$.inboundSchema = exports.DangerousGoodsLithiumBatteries$inboundSchema;
    /** @deprecated use `DangerousGoodsLithiumBatteries$outboundSchema` instead. */
    DangerousGoodsLithiumBatteries$.outboundSchema = exports.DangerousGoodsLithiumBatteries$outboundSchema;
})(DangerousGoodsLithiumBatteries$ || (exports.DangerousGoodsLithiumBatteries$ = DangerousGoodsLithiumBatteries$ = {}));
function dangerousGoodsLithiumBatteriesToJSON(dangerousGoodsLithiumBatteries) {
    return JSON.stringify(exports.DangerousGoodsLithiumBatteries$outboundSchema.parse(dangerousGoodsLithiumBatteries));
}
exports.dangerousGoodsLithiumBatteriesToJSON = dangerousGoodsLithiumBatteriesToJSON;
function dangerousGoodsLithiumBatteriesFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.DangerousGoodsLithiumBatteries$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DangerousGoodsLithiumBatteries' from JSON`);
}
exports.dangerousGoodsLithiumBatteriesFromJSON = dangerousGoodsLithiumBatteriesFromJSON;
