// shippo/models/components/defaultparceltemplate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.defaultParcelTemplateFromJSON = exports.defaultParcelTemplateToJSON = exports.DefaultParcelTemplate$ = exports.DefaultParcelTemplate$outboundSchema = exports.DefaultParcelTemplate$inboundSchema = void 0;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const userparceltemplate_js_1 = require("./userparceltemplate.js");
/** @internal */
exports.DefaultParcelTemplate$inboundSchema = z.object({
    result: userparceltemplate_js_1.UserParcelTemplate$inboundSchema.optional(),
});
/** @internal */
exports.DefaultParcelTemplate$outboundSchema = z.object({
    result: userparceltemplate_js_1.UserParcelTemplate$outboundSchema.optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DefaultParcelTemplate$;
(function (DefaultParcelTemplate$) {
    /** @deprecated use `DefaultParcelTemplate$inboundSchema` instead. */
    DefaultParcelTemplate$.inboundSchema = exports.DefaultParcelTemplate$inboundSchema;
    /** @deprecated use `DefaultParcelTemplate$outboundSchema` instead. */
    DefaultParcelTemplate$.outboundSchema = exports.DefaultParcelTemplate$outboundSchema;
})(DefaultParcelTemplate$ || (exports.DefaultParcelTemplate$ = DefaultParcelTemplate$ = {}));
function defaultParcelTemplateToJSON(defaultParcelTemplate) {
    return JSON.stringify(exports.DefaultParcelTemplate$outboundSchema.parse(defaultParcelTemplate));
}
exports.defaultParcelTemplateToJSON = defaultParcelTemplateToJSON;
function defaultParcelTemplateFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.DefaultParcelTemplate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DefaultParcelTemplate' from JSON`);
}
exports.defaultParcelTemplateFromJSON = defaultParcelTemplateFromJSON;
