// shippo/models/components/defaultparceltemplateupdaterequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.defaultParcelTemplateUpdateRequestFromJSON = exports.defaultParcelTemplateUpdateRequestToJSON = exports.DefaultParcelTemplateUpdateRequest$ = exports.DefaultParcelTemplateUpdateRequest$outboundSchema = exports.DefaultParcelTemplateUpdateRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.DefaultParcelTemplateUpdateRequest$inboundSchema = z.object({
    object_id: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "object_id": "objectId",
    });
});
/** @internal */
exports.DefaultParcelTemplateUpdateRequest$outboundSchema = z.object({
    objectId: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        objectId: "object_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DefaultParcelTemplateUpdateRequest$;
(function (DefaultParcelTemplateUpdateRequest$) {
    /** @deprecated use `DefaultParcelTemplateUpdateRequest$inboundSchema` instead. */
    DefaultParcelTemplateUpdateRequest$.inboundSchema = exports.DefaultParcelTemplateUpdateRequest$inboundSchema;
    /** @deprecated use `DefaultParcelTemplateUpdateRequest$outboundSchema` instead. */
    DefaultParcelTemplateUpdateRequest$.outboundSchema = exports.DefaultParcelTemplateUpdateRequest$outboundSchema;
})(DefaultParcelTemplateUpdateRequest$ || (exports.DefaultParcelTemplateUpdateRequest$ = DefaultParcelTemplateUpdateRequest$ = {}));
function defaultParcelTemplateUpdateRequestToJSON(defaultParcelTemplateUpdateRequest) {
    return JSON.stringify(exports.DefaultParcelTemplateUpdateRequest$outboundSchema.parse(defaultParcelTemplateUpdateRequest));
}
exports.defaultParcelTemplateUpdateRequestToJSON = defaultParcelTemplateUpdateRequestToJSON;
function defaultParcelTemplateUpdateRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.DefaultParcelTemplateUpdateRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DefaultParcelTemplateUpdateRequest' from JSON`);
}
exports.defaultParcelTemplateUpdateRequestFromJSON = defaultParcelTemplateUpdateRequestFromJSON;
