// shippo/models/components/dryice.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.dryIceFromJSON = exports.dryIceToJSON = exports.DryIce$ = exports.DryIce$outboundSchema = exports.DryIce$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.DryIce$inboundSchema = z
    .object({
    contains_dry_ice: z.boolean().optional(),
    weight: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "contains_dry_ice": "containsDryIce",
    });
});
/** @internal */
exports.DryIce$outboundSchema = z.object({
    containsDryIce: z.boolean().optional(),
    weight: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        containsDryIce: "contains_dry_ice",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DryIce$;
(function (DryIce$) {
    /** @deprecated use `DryIce$inboundSchema` instead. */
    DryIce$.inboundSchema = exports.DryIce$inboundSchema;
    /** @deprecated use `DryIce$outboundSchema` instead. */
    DryIce$.outboundSchema = exports.DryIce$outboundSchema;
})(DryIce$ || (exports.DryIce$ = DryIce$ = {}));
function dryIceToJSON(dryIce) {
    return JSON.stringify(exports.DryIce$outboundSchema.parse(dryIce));
}
exports.dryIceToJSON = dryIceToJSON;
function dryIceFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.DryIce$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DryIce' from JSON`);
}
exports.dryIceFromJSON = dryIceFromJSON;
