// shippo/models/components/fedexconnectexistingownaccountparameters.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.fedExConnectExistingOwnAccountParametersFromJSON = exports.fedExConnectExistingOwnAccountParametersToJSON = exports.FedExConnectExistingOwnAccountParameters$ = exports.FedExConnectExistingOwnAccountParameters$outboundSchema = exports.FedExConnectExistingOwnAccountParameters$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.FedExConnectExistingOwnAccountParameters$inboundSchema = z.object({
    first_name: z.string(),
    last_name: z.string(),
    phone_number: z.string(),
    from_address_st: z.string(),
    from_address_city: z.string(),
    from_address_state: z.string(),
    from_address_zip: z.string(),
    from_address_country_iso2: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "first_name": "firstName",
        "last_name": "lastName",
        "phone_number": "phoneNumber",
        "from_address_st": "fromAddressSt",
        "from_address_city": "fromAddressCity",
        "from_address_state": "fromAddressState",
        "from_address_zip": "fromAddressZip",
        "from_address_country_iso2": "fromAddressCountryIso2",
    });
});
/** @internal */
exports.FedExConnectExistingOwnAccountParameters$outboundSchema = z.object({
    firstName: z.string(),
    lastName: z.string(),
    phoneNumber: z.string(),
    fromAddressSt: z.string(),
    fromAddressCity: z.string(),
    fromAddressState: z.string(),
    fromAddressZip: z.string(),
    fromAddressCountryIso2: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        firstName: "first_name",
        lastName: "last_name",
        phoneNumber: "phone_number",
        fromAddressSt: "from_address_st",
        fromAddressCity: "from_address_city",
        fromAddressState: "from_address_state",
        fromAddressZip: "from_address_zip",
        fromAddressCountryIso2: "from_address_country_iso2",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var FedExConnectExistingOwnAccountParameters$;
(function (FedExConnectExistingOwnAccountParameters$) {
    /** @deprecated use `FedExConnectExistingOwnAccountParameters$inboundSchema` instead. */
    FedExConnectExistingOwnAccountParameters$.inboundSchema = exports.FedExConnectExistingOwnAccountParameters$inboundSchema;
    /** @deprecated use `FedExConnectExistingOwnAccountParameters$outboundSchema` instead. */
    FedExConnectExistingOwnAccountParameters$.outboundSchema = exports.FedExConnectExistingOwnAccountParameters$outboundSchema;
})(FedExConnectExistingOwnAccountParameters$ || (exports.FedExConnectExistingOwnAccountParameters$ = FedExConnectExistingOwnAccountParameters$ = {}));
function fedExConnectExistingOwnAccountParametersToJSON(fedExConnectExistingOwnAccountParameters) {
    return JSON.stringify(exports.FedExConnectExistingOwnAccountParameters$outboundSchema.parse(fedExConnectExistingOwnAccountParameters));
}
exports.fedExConnectExistingOwnAccountParametersToJSON = fedExConnectExistingOwnAccountParametersToJSON;
function fedExConnectExistingOwnAccountParametersFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.FedExConnectExistingOwnAccountParameters$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'FedExConnectExistingOwnAccountParameters' from JSON`);
}
exports.fedExConnectExistingOwnAccountParametersFromJSON = fedExConnectExistingOwnAccountParametersFromJSON;
