// shippo/models/components/instanttransactioncreaterequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.instantTransactionCreateRequestFromJSON = exports.instantTransactionCreateRequestToJSON = exports.InstantTransactionCreateRequest$ = exports.InstantTransactionCreateRequest$outboundSchema = exports.InstantTransactionCreateRequest$inboundSchema = exports.LabelFileType$ = exports.LabelFileType$outboundSchema = exports.LabelFileType$inboundSchema = exports.LabelFileType = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const shipmentcreaterequest_js_1 = require("./shipmentcreaterequest.js");
exports.LabelFileType = {
    Png: "PNG",
    PNG23x75: "PNG_2.3x7.5",
    Pdf: "PDF",
    PDF23x75: "PDF_2.3x7.5",
    PDF4x6: "PDF_4x6",
    PDF4x8: "PDF_4x8",
    PdfA4: "PDF_A4",
    PdfA6: "PDF_A6",
    Zplii: "ZPLII",
};
/** @internal */
exports.LabelFileType$inboundSchema = z.nativeEnum(exports.LabelFileType);
/** @internal */
exports.LabelFileType$outboundSchema = exports.LabelFileType$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var LabelFileType$;
(function (LabelFileType$) {
    /** @deprecated use `LabelFileType$inboundSchema` instead. */
    LabelFileType$.inboundSchema = exports.LabelFileType$inboundSchema;
    /** @deprecated use `LabelFileType$outboundSchema` instead. */
    LabelFileType$.outboundSchema = exports.LabelFileType$outboundSchema;
})(LabelFileType$ || (exports.LabelFileType$ = LabelFileType$ = {}));
/** @internal */
exports.InstantTransactionCreateRequest$inboundSchema = z.object({
    async: z.boolean().default(false),
    carrier_account: z.string(),
    label_file_type: exports.LabelFileType$inboundSchema.optional(),
    metadata: z.string().optional(),
    servicelevel_token: z.string(),
    shipment: shipmentcreaterequest_js_1.ShipmentCreateRequest$inboundSchema,
    order: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "carrier_account": "carrierAccount",
        "label_file_type": "labelFileType",
        "servicelevel_token": "servicelevelToken",
    });
});
/** @internal */
exports.InstantTransactionCreateRequest$outboundSchema = z.object({
    async: z.boolean().default(false),
    carrierAccount: z.string(),
    labelFileType: exports.LabelFileType$outboundSchema.optional(),
    metadata: z.string().optional(),
    servicelevelToken: z.string(),
    shipment: shipmentcreaterequest_js_1.ShipmentCreateRequest$outboundSchema,
    order: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        carrierAccount: "carrier_account",
        labelFileType: "label_file_type",
        servicelevelToken: "servicelevel_token",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var InstantTransactionCreateRequest$;
(function (InstantTransactionCreateRequest$) {
    /** @deprecated use `InstantTransactionCreateRequest$inboundSchema` instead. */
    InstantTransactionCreateRequest$.inboundSchema = exports.InstantTransactionCreateRequest$inboundSchema;
    /** @deprecated use `InstantTransactionCreateRequest$outboundSchema` instead. */
    InstantTransactionCreateRequest$.outboundSchema = exports.InstantTransactionCreateRequest$outboundSchema;
})(InstantTransactionCreateRequest$ || (exports.InstantTransactionCreateRequest$ = InstantTransactionCreateRequest$ = {}));
function instantTransactionCreateRequestToJSON(instantTransactionCreateRequest) {
    return JSON.stringify(exports.InstantTransactionCreateRequest$outboundSchema.parse(instantTransactionCreateRequest));
}
exports.instantTransactionCreateRequestToJSON = instantTransactionCreateRequestToJSON;
function instantTransactionCreateRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.InstantTransactionCreateRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'InstantTransactionCreateRequest' from JSON`);
}
exports.instantTransactionCreateRequestFromJSON = instantTransactionCreateRequestFromJSON;
