// shippo/models/components/insurance.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.insuranceFromJSON = exports.insuranceToJSON = exports.Insurance$ = exports.Insurance$outboundSchema = exports.Insurance$inboundSchema = exports.Provider$ = exports.Provider$outboundSchema = exports.Provider$inboundSchema = exports.Provider = void 0;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
/**
 * To have insurance cover provided by a carrier directly instead of Shippo's provider (XCover), set `provider` to `FEDEX`, `UPS`, or `ONTRAC`.
 */
exports.Provider = {
    Fedex: "FEDEX",
    Ups: "UPS",
    Ontrac: "ONTRAC",
};
/** @internal */
exports.Provider$inboundSchema = z
    .nativeEnum(exports.Provider);
/** @internal */
exports.Provider$outboundSchema = exports.Provider$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Provider$;
(function (Provider$) {
    /** @deprecated use `Provider$inboundSchema` instead. */
    Provider$.inboundSchema = exports.Provider$inboundSchema;
    /** @deprecated use `Provider$outboundSchema` instead. */
    Provider$.outboundSchema = exports.Provider$outboundSchema;
})(Provider$ || (exports.Provider$ = Provider$ = {}));
/** @internal */
exports.Insurance$inboundSchema = z.object({
    amount: z.string().optional(),
    content: z.string().optional(),
    currency: z.string().optional(),
    provider: exports.Provider$inboundSchema.optional(),
});
/** @internal */
exports.Insurance$outboundSchema = z.object({
    amount: z.string().optional(),
    content: z.string().optional(),
    currency: z.string().optional(),
    provider: exports.Provider$outboundSchema.optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Insurance$;
(function (Insurance$) {
    /** @deprecated use `Insurance$inboundSchema` instead. */
    Insurance$.inboundSchema = exports.Insurance$inboundSchema;
    /** @deprecated use `Insurance$outboundSchema` instead. */
    Insurance$.outboundSchema = exports.Insurance$outboundSchema;
})(Insurance$ || (exports.Insurance$ = Insurance$ = {}));
function insuranceToJSON(insurance) {
    return JSON.stringify(exports.Insurance$outboundSchema.parse(insurance));
}
exports.insuranceToJSON = insuranceToJSON;
function insuranceFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Insurance$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Insurance' from JSON`);
}
exports.insuranceFromJSON = insuranceFromJSON;
