// shippo/models/components/invoicenumber.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.invoiceNumberFromJSON = exports.invoiceNumberToJSON = exports.InvoiceNumber$ = exports.InvoiceNumber$outboundSchema = exports.InvoiceNumber$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.InvoiceNumber$inboundSchema = z.object({
    prefix: z.string().optional(),
    value: z.string().optional(),
    ref_sort: z.number().int().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "ref_sort": "refSort",
    });
});
/** @internal */
exports.InvoiceNumber$outboundSchema = z.object({
    prefix: z.string().optional(),
    value: z.string().optional(),
    refSort: z.number().int().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        refSort: "ref_sort",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var InvoiceNumber$;
(function (InvoiceNumber$) {
    /** @deprecated use `InvoiceNumber$inboundSchema` instead. */
    InvoiceNumber$.inboundSchema = exports.InvoiceNumber$inboundSchema;
    /** @deprecated use `InvoiceNumber$outboundSchema` instead. */
    InvoiceNumber$.outboundSchema = exports.InvoiceNumber$outboundSchema;
})(InvoiceNumber$ || (exports.InvoiceNumber$ = InvoiceNumber$ = {}));
function invoiceNumberToJSON(invoiceNumber) {
    return JSON.stringify(exports.InvoiceNumber$outboundSchema.parse(invoiceNumber));
}
exports.invoiceNumberToJSON = invoiceNumberToJSON;
function invoiceNumberFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.InvoiceNumber$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'InvoiceNumber' from JSON`);
}
exports.invoiceNumberFromJSON = invoiceNumberFromJSON;
