// shippo/models/components/liveratecreaterequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.liveRateCreateRequestFromJSON = exports.liveRateCreateRequestToJSON = exports.LiveRateCreateRequest$ = exports.LiveRateCreateRequest$outboundSchema = exports.LiveRateCreateRequest$inboundSchema = exports.liveRateCreateRequestParcelFromJSON = exports.liveRateCreateRequestParcelToJSON = exports.LiveRateCreateRequestParcel$ = exports.LiveRateCreateRequestParcel$outboundSchema = exports.LiveRateCreateRequestParcel$inboundSchema = exports.liveRateCreateRequestAddressToFromJSON = exports.liveRateCreateRequestAddressToToJSON = exports.LiveRateCreateRequestAddressTo$ = exports.LiveRateCreateRequestAddressTo$outboundSchema = exports.LiveRateCreateRequestAddressTo$inboundSchema = exports.liveRateCreateRequestAddressFromFromJSON = exports.liveRateCreateRequestAddressFromToJSON = exports.LiveRateCreateRequestAddressFrom$ = exports.LiveRateCreateRequestAddressFrom$outboundSchema = exports.LiveRateCreateRequestAddressFrom$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const addresscompletecreaterequest_js_1 = require("./addresscompletecreaterequest.js");
const lineitem_js_1 = require("./lineitem.js");
const parcel_js_1 = require("./parcel.js");
/** @internal */
exports.LiveRateCreateRequestAddressFrom$inboundSchema = z.union([addresscompletecreaterequest_js_1.AddressCompleteCreateRequest$inboundSchema, z.string()]);
/** @internal */
exports.LiveRateCreateRequestAddressFrom$outboundSchema = z.union([addresscompletecreaterequest_js_1.AddressCompleteCreateRequest$outboundSchema, z.string()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var LiveRateCreateRequestAddressFrom$;
(function (LiveRateCreateRequestAddressFrom$) {
    /** @deprecated use `LiveRateCreateRequestAddressFrom$inboundSchema` instead. */
    LiveRateCreateRequestAddressFrom$.inboundSchema = exports.LiveRateCreateRequestAddressFrom$inboundSchema;
    /** @deprecated use `LiveRateCreateRequestAddressFrom$outboundSchema` instead. */
    LiveRateCreateRequestAddressFrom$.outboundSchema = exports.LiveRateCreateRequestAddressFrom$outboundSchema;
})(LiveRateCreateRequestAddressFrom$ || (exports.LiveRateCreateRequestAddressFrom$ = LiveRateCreateRequestAddressFrom$ = {}));
function liveRateCreateRequestAddressFromToJSON(liveRateCreateRequestAddressFrom) {
    return JSON.stringify(exports.LiveRateCreateRequestAddressFrom$outboundSchema.parse(liveRateCreateRequestAddressFrom));
}
exports.liveRateCreateRequestAddressFromToJSON = liveRateCreateRequestAddressFromToJSON;
function liveRateCreateRequestAddressFromFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.LiveRateCreateRequestAddressFrom$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'LiveRateCreateRequestAddressFrom' from JSON`);
}
exports.liveRateCreateRequestAddressFromFromJSON = liveRateCreateRequestAddressFromFromJSON;
/** @internal */
exports.LiveRateCreateRequestAddressTo$inboundSchema = z.union([addresscompletecreaterequest_js_1.AddressCompleteCreateRequest$inboundSchema, z.string()]);
/** @internal */
exports.LiveRateCreateRequestAddressTo$outboundSchema = z.union([addresscompletecreaterequest_js_1.AddressCompleteCreateRequest$outboundSchema, z.string()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var LiveRateCreateRequestAddressTo$;
(function (LiveRateCreateRequestAddressTo$) {
    /** @deprecated use `LiveRateCreateRequestAddressTo$inboundSchema` instead. */
    LiveRateCreateRequestAddressTo$.inboundSchema = exports.LiveRateCreateRequestAddressTo$inboundSchema;
    /** @deprecated use `LiveRateCreateRequestAddressTo$outboundSchema` instead. */
    LiveRateCreateRequestAddressTo$.outboundSchema = exports.LiveRateCreateRequestAddressTo$outboundSchema;
})(LiveRateCreateRequestAddressTo$ || (exports.LiveRateCreateRequestAddressTo$ = LiveRateCreateRequestAddressTo$ = {}));
function liveRateCreateRequestAddressToToJSON(liveRateCreateRequestAddressTo) {
    return JSON.stringify(exports.LiveRateCreateRequestAddressTo$outboundSchema.parse(liveRateCreateRequestAddressTo));
}
exports.liveRateCreateRequestAddressToToJSON = liveRateCreateRequestAddressToToJSON;
function liveRateCreateRequestAddressToFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.LiveRateCreateRequestAddressTo$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'LiveRateCreateRequestAddressTo' from JSON`);
}
exports.liveRateCreateRequestAddressToFromJSON = liveRateCreateRequestAddressToFromJSON;
/** @internal */
exports.LiveRateCreateRequestParcel$inboundSchema = z.union([parcel_js_1.Parcel$inboundSchema, z.string()]);
/** @internal */
exports.LiveRateCreateRequestParcel$outboundSchema = z.union([parcel_js_1.Parcel$outboundSchema, z.string()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var LiveRateCreateRequestParcel$;
(function (LiveRateCreateRequestParcel$) {
    /** @deprecated use `LiveRateCreateRequestParcel$inboundSchema` instead. */
    LiveRateCreateRequestParcel$.inboundSchema = exports.LiveRateCreateRequestParcel$inboundSchema;
    /** @deprecated use `LiveRateCreateRequestParcel$outboundSchema` instead. */
    LiveRateCreateRequestParcel$.outboundSchema = exports.LiveRateCreateRequestParcel$outboundSchema;
})(LiveRateCreateRequestParcel$ || (exports.LiveRateCreateRequestParcel$ = LiveRateCreateRequestParcel$ = {}));
function liveRateCreateRequestParcelToJSON(liveRateCreateRequestParcel) {
    return JSON.stringify(exports.LiveRateCreateRequestParcel$outboundSchema.parse(liveRateCreateRequestParcel));
}
exports.liveRateCreateRequestParcelToJSON = liveRateCreateRequestParcelToJSON;
function liveRateCreateRequestParcelFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.LiveRateCreateRequestParcel$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'LiveRateCreateRequestParcel' from JSON`);
}
exports.liveRateCreateRequestParcelFromJSON = liveRateCreateRequestParcelFromJSON;
/** @internal */
exports.LiveRateCreateRequest$inboundSchema = z.object({
    address_from: z.union([
        addresscompletecreaterequest_js_1.AddressCompleteCreateRequest$inboundSchema,
        z.string(),
    ]).optional(),
    address_to: z.union([addresscompletecreaterequest_js_1.AddressCompleteCreateRequest$inboundSchema, z.string()]),
    line_items: z.array(lineitem_js_1.LineItem$inboundSchema),
    parcel: z.union([parcel_js_1.Parcel$inboundSchema, z.string()]).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "address_from": "addressFrom",
        "address_to": "addressTo",
        "line_items": "lineItems",
    });
});
/** @internal */
exports.LiveRateCreateRequest$outboundSchema = z.object({
    addressFrom: z.union([
        addresscompletecreaterequest_js_1.AddressCompleteCreateRequest$outboundSchema,
        z.string(),
    ]).optional(),
    addressTo: z.union([addresscompletecreaterequest_js_1.AddressCompleteCreateRequest$outboundSchema, z.string()]),
    lineItems: z.array(lineitem_js_1.LineItem$outboundSchema),
    parcel: z.union([parcel_js_1.Parcel$outboundSchema, z.string()]).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        addressFrom: "address_from",
        addressTo: "address_to",
        lineItems: "line_items",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var LiveRateCreateRequest$;
(function (LiveRateCreateRequest$) {
    /** @deprecated use `LiveRateCreateRequest$inboundSchema` instead. */
    LiveRateCreateRequest$.inboundSchema = exports.LiveRateCreateRequest$inboundSchema;
    /** @deprecated use `LiveRateCreateRequest$outboundSchema` instead. */
    LiveRateCreateRequest$.outboundSchema = exports.LiveRateCreateRequest$outboundSchema;
})(LiveRateCreateRequest$ || (exports.LiveRateCreateRequest$ = LiveRateCreateRequest$ = {}));
function liveRateCreateRequestToJSON(liveRateCreateRequest) {
    return JSON.stringify(exports.LiveRateCreateRequest$outboundSchema.parse(liveRateCreateRequest));
}
exports.liveRateCreateRequestToJSON = liveRateCreateRequestToJSON;
function liveRateCreateRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.LiveRateCreateRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'LiveRateCreateRequest' from JSON`);
}
exports.liveRateCreateRequestFromJSON = liveRateCreateRequestFromJSON;
