// shippo/models/components/liveratepaginatedlist.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.liveRatePaginatedListFromJSON = exports.liveRatePaginatedListToJSON = exports.LiveRatePaginatedList$ = exports.LiveRatePaginatedList$outboundSchema = exports.LiveRatePaginatedList$inboundSchema = void 0;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const liverate_js_1 = require("./liverate.js");
/** @internal */
exports.LiveRatePaginatedList$inboundSchema = z.object({
    next: z.string().optional(),
    previous: z.string().optional(),
    count: z.number().int().optional(),
    results: z.array(liverate_js_1.LiveRate$inboundSchema).optional(),
});
/** @internal */
exports.LiveRatePaginatedList$outboundSchema = z.object({
    next: z.string().optional(),
    previous: z.string().optional(),
    count: z.number().int().optional(),
    results: z.array(liverate_js_1.LiveRate$outboundSchema).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var LiveRatePaginatedList$;
(function (LiveRatePaginatedList$) {
    /** @deprecated use `LiveRatePaginatedList$inboundSchema` instead. */
    LiveRatePaginatedList$.inboundSchema = exports.LiveRatePaginatedList$inboundSchema;
    /** @deprecated use `LiveRatePaginatedList$outboundSchema` instead. */
    LiveRatePaginatedList$.outboundSchema = exports.LiveRatePaginatedList$outboundSchema;
})(LiveRatePaginatedList$ || (exports.LiveRatePaginatedList$ = LiveRatePaginatedList$ = {}));
function liveRatePaginatedListToJSON(liveRatePaginatedList) {
    return JSON.stringify(exports.LiveRatePaginatedList$outboundSchema.parse(liveRatePaginatedList));
}
exports.liveRatePaginatedListToJSON = liveRatePaginatedListToJSON;
function liveRatePaginatedListFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.LiveRatePaginatedList$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'LiveRatePaginatedList' from JSON`);
}
exports.liveRatePaginatedListFromJSON = liveRatePaginatedListFromJSON;
