// shippo/models/components/liverate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.liveRateFromJSON = exports.liveRateToJSON = exports.LiveRate$ = exports.LiveRate$outboundSchema = exports.LiveRate$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.LiveRate$inboundSchema = z.object({
    amount: z.string().optional(),
    amount_local: z.string().optional(),
    currency: z.string().optional(),
    currency_local: z.string().optional(),
    estimated_days: z.number().int().optional(),
    title: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "amount_local": "amountLocal",
        "currency_local": "currencyLocal",
        "estimated_days": "estimatedDays",
    });
});
/** @internal */
exports.LiveRate$outboundSchema = z.object({
    amount: z.string().optional(),
    amountLocal: z.string().optional(),
    currency: z.string().optional(),
    currencyLocal: z.string().optional(),
    estimatedDays: z.number().int().optional(),
    title: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        amountLocal: "amount_local",
        currencyLocal: "currency_local",
        estimatedDays: "estimated_days",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var LiveRate$;
(function (LiveRate$) {
    /** @deprecated use `LiveRate$inboundSchema` instead. */
    LiveRate$.inboundSchema = exports.LiveRate$inboundSchema;
    /** @deprecated use `LiveRate$outboundSchema` instead. */
    LiveRate$.outboundSchema = exports.LiveRate$outboundSchema;
})(LiveRate$ || (exports.LiveRate$ = LiveRate$ = {}));
function liveRateToJSON(liveRate) {
    return JSON.stringify(exports.LiveRate$outboundSchema.parse(liveRate));
}
exports.liveRateToJSON = liveRateToJSON;
function liveRateFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.LiveRate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'LiveRate' from JSON`);
}
exports.liveRateFromJSON = liveRateFromJSON;
