// shippo/models/components/manifestcreaterequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.manifestCreateRequestFromJSON = exports.manifestCreateRequestToJSON = exports.ManifestCreateRequest$ = exports.ManifestCreateRequest$outboundSchema = exports.ManifestCreateRequest$inboundSchema = exports.manifestCreateRequestAddressFromFromJSON = exports.manifestCreateRequestAddressFromToJSON = exports.ManifestCreateRequestAddressFrom$ = exports.ManifestCreateRequestAddressFrom$outboundSchema = exports.ManifestCreateRequestAddressFrom$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const addresscreaterequest_js_1 = require("./addresscreaterequest.js");
/** @internal */
exports.ManifestCreateRequestAddressFrom$inboundSchema = z.union([addresscreaterequest_js_1.AddressCreateRequest$inboundSchema, z.string()]);
/** @internal */
exports.ManifestCreateRequestAddressFrom$outboundSchema = z.union([addresscreaterequest_js_1.AddressCreateRequest$outboundSchema, z.string()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ManifestCreateRequestAddressFrom$;
(function (ManifestCreateRequestAddressFrom$) {
    /** @deprecated use `ManifestCreateRequestAddressFrom$inboundSchema` instead. */
    ManifestCreateRequestAddressFrom$.inboundSchema = exports.ManifestCreateRequestAddressFrom$inboundSchema;
    /** @deprecated use `ManifestCreateRequestAddressFrom$outboundSchema` instead. */
    ManifestCreateRequestAddressFrom$.outboundSchema = exports.ManifestCreateRequestAddressFrom$outboundSchema;
})(ManifestCreateRequestAddressFrom$ || (exports.ManifestCreateRequestAddressFrom$ = ManifestCreateRequestAddressFrom$ = {}));
function manifestCreateRequestAddressFromToJSON(manifestCreateRequestAddressFrom) {
    return JSON.stringify(exports.ManifestCreateRequestAddressFrom$outboundSchema.parse(manifestCreateRequestAddressFrom));
}
exports.manifestCreateRequestAddressFromToJSON = manifestCreateRequestAddressFromToJSON;
function manifestCreateRequestAddressFromFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ManifestCreateRequestAddressFrom$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ManifestCreateRequestAddressFrom' from JSON`);
}
exports.manifestCreateRequestAddressFromFromJSON = manifestCreateRequestAddressFromFromJSON;
/** @internal */
exports.ManifestCreateRequest$inboundSchema = z.object({
    carrier_account: z.string(),
    shipment_date: z.string(),
    transactions: z.array(z.string()).optional(),
    address_from: z.union([addresscreaterequest_js_1.AddressCreateRequest$inboundSchema, z.string()]),
    async: z.boolean().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "carrier_account": "carrierAccount",
        "shipment_date": "shipmentDate",
        "address_from": "addressFrom",
    });
});
/** @internal */
exports.ManifestCreateRequest$outboundSchema = z.object({
    carrierAccount: z.string(),
    shipmentDate: z.string(),
    transactions: z.array(z.string()).optional(),
    addressFrom: z.union([addresscreaterequest_js_1.AddressCreateRequest$outboundSchema, z.string()]),
    async: z.boolean().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        carrierAccount: "carrier_account",
        shipmentDate: "shipment_date",
        addressFrom: "address_from",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ManifestCreateRequest$;
(function (ManifestCreateRequest$) {
    /** @deprecated use `ManifestCreateRequest$inboundSchema` instead. */
    ManifestCreateRequest$.inboundSchema = exports.ManifestCreateRequest$inboundSchema;
    /** @deprecated use `ManifestCreateRequest$outboundSchema` instead. */
    ManifestCreateRequest$.outboundSchema = exports.ManifestCreateRequest$outboundSchema;
})(ManifestCreateRequest$ || (exports.ManifestCreateRequest$ = ManifestCreateRequest$ = {}));
function manifestCreateRequestToJSON(manifestCreateRequest) {
    return JSON.stringify(exports.ManifestCreateRequest$outboundSchema.parse(manifestCreateRequest));
}
exports.manifestCreateRequestToJSON = manifestCreateRequestToJSON;
function manifestCreateRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ManifestCreateRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ManifestCreateRequest' from JSON`);
}
exports.manifestCreateRequestFromJSON = manifestCreateRequestFromJSON;
