// shippo/models/components/ordercreaterequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.orderCreateRequestFromJSON = exports.orderCreateRequestToJSON = exports.OrderCreateRequest$ = exports.OrderCreateRequest$outboundSchema = exports.OrderCreateRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const addresscreaterequest_js_1 = require("./addresscreaterequest.js");
const lineitembase_js_1 = require("./lineitembase.js");
const orderstatusenum_js_1 = require("./orderstatusenum.js");
const weightunitenum_js_1 = require("./weightunitenum.js");
/** @internal */
exports.OrderCreateRequest$inboundSchema = z.object({
    currency: z.string().optional(),
    notes: z.string().optional(),
    order_number: z.string().optional(),
    order_status: orderstatusenum_js_1.OrderStatusEnum$inboundSchema.optional(),
    placed_at: z.string(),
    shipping_cost: z.string().optional(),
    shipping_cost_currency: z.string().optional(),
    shipping_method: z.string().optional(),
    subtotal_price: z.string().optional(),
    total_price: z.string().optional(),
    total_tax: z.string().optional(),
    weight: z.string().optional(),
    weight_unit: weightunitenum_js_1.WeightUnitEnum$inboundSchema.optional(),
    from_address: addresscreaterequest_js_1.AddressCreateRequest$inboundSchema.optional(),
    to_address: addresscreaterequest_js_1.AddressCreateRequest$inboundSchema,
    line_items: z.array(lineitembase_js_1.LineItemBase$inboundSchema).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "order_number": "orderNumber",
        "order_status": "orderStatus",
        "placed_at": "placedAt",
        "shipping_cost": "shippingCost",
        "shipping_cost_currency": "shippingCostCurrency",
        "shipping_method": "shippingMethod",
        "subtotal_price": "subtotalPrice",
        "total_price": "totalPrice",
        "total_tax": "totalTax",
        "weight_unit": "weightUnit",
        "from_address": "fromAddress",
        "to_address": "toAddress",
        "line_items": "lineItems",
    });
});
/** @internal */
exports.OrderCreateRequest$outboundSchema = z.object({
    currency: z.string().optional(),
    notes: z.string().optional(),
    orderNumber: z.string().optional(),
    orderStatus: orderstatusenum_js_1.OrderStatusEnum$outboundSchema.optional(),
    placedAt: z.string(),
    shippingCost: z.string().optional(),
    shippingCostCurrency: z.string().optional(),
    shippingMethod: z.string().optional(),
    subtotalPrice: z.string().optional(),
    totalPrice: z.string().optional(),
    totalTax: z.string().optional(),
    weight: z.string().optional(),
    weightUnit: weightunitenum_js_1.WeightUnitEnum$outboundSchema.optional(),
    fromAddress: addresscreaterequest_js_1.AddressCreateRequest$outboundSchema.optional(),
    toAddress: addresscreaterequest_js_1.AddressCreateRequest$outboundSchema,
    lineItems: z.array(lineitembase_js_1.LineItemBase$outboundSchema).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        orderNumber: "order_number",
        orderStatus: "order_status",
        placedAt: "placed_at",
        shippingCost: "shipping_cost",
        shippingCostCurrency: "shipping_cost_currency",
        shippingMethod: "shipping_method",
        subtotalPrice: "subtotal_price",
        totalPrice: "total_price",
        totalTax: "total_tax",
        weightUnit: "weight_unit",
        fromAddress: "from_address",
        toAddress: "to_address",
        lineItems: "line_items",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var OrderCreateRequest$;
(function (OrderCreateRequest$) {
    /** @deprecated use `OrderCreateRequest$inboundSchema` instead. */
    OrderCreateRequest$.inboundSchema = exports.OrderCreateRequest$inboundSchema;
    /** @deprecated use `OrderCreateRequest$outboundSchema` instead. */
    OrderCreateRequest$.outboundSchema = exports.OrderCreateRequest$outboundSchema;
})(OrderCreateRequest$ || (exports.OrderCreateRequest$ = OrderCreateRequest$ = {}));
function orderCreateRequestToJSON(orderCreateRequest) {
    return JSON.stringify(exports.OrderCreateRequest$outboundSchema.parse(orderCreateRequest));
}
exports.orderCreateRequestToJSON = orderCreateRequestToJSON;
function orderCreateRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.OrderCreateRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'OrderCreateRequest' from JSON`);
}
exports.orderCreateRequestFromJSON = orderCreateRequestFromJSON;
