// shippo/models/components/orderstatusenum.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrderStatusEnum$ = exports.OrderStatusEnum$outboundSchema = exports.OrderStatusEnum$inboundSchema = exports.OrderStatusEnum = void 0;
const z = __importStar(require("zod"));
/**
 * Current state of the order. See the <a href="https://docs.goshippo.com/docs/orders/orders/">orders tutorial</a>
 *
 * @remarks
 * for the logic of how the status is handled.
 */
exports.OrderStatusEnum = {
    Unknown: "UNKNOWN",
    Awaitpay: "AWAITPAY",
    Paid: "PAID",
    Refunded: "REFUNDED",
    Cancelled: "CANCELLED",
    PartiallyFulfilled: "PARTIALLY_FULFILLED",
    Shipped: "SHIPPED",
};
/** @internal */
exports.OrderStatusEnum$inboundSchema = z.nativeEnum(exports.OrderStatusEnum);
/** @internal */
exports.OrderStatusEnum$outboundSchema = exports.OrderStatusEnum$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var OrderStatusEnum$;
(function (OrderStatusEnum$) {
    /** @deprecated use `OrderStatusEnum$inboundSchema` instead. */
    OrderStatusEnum$.inboundSchema = exports.OrderStatusEnum$inboundSchema;
    /** @deprecated use `OrderStatusEnum$outboundSchema` instead. */
    OrderStatusEnum$.outboundSchema = exports.OrderStatusEnum$outboundSchema;
})(OrderStatusEnum$ || (exports.OrderStatusEnum$ = OrderStatusEnum$ = {}));
