// shippo/models/components/ordershopappenum.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrderShopAppEnum$ = exports.OrderShopAppEnum$outboundSchema = exports.OrderShopAppEnum$inboundSchema = exports.OrderShopAppEnum = void 0;
const z = __importStar(require("zod"));
/**
 * Platform the order was created on and, if applicable, imported from.
 *
 * @remarks
 * Orders created via the Shippo API or dashboard will have the value "Shippo".
 */
exports.OrderShopAppEnum = {
    Amazon: "Amazon",
    Bigcommerce: "Bigcommerce",
    CSVImport: "CSV_Import",
    EBay: "eBay",
    EPages: "ePages",
    Etsy: "Etsy",
    Godaddy: "Godaddy",
    Magento: "Magento",
    Shippo: "Shippo",
    Shopify: "Shopify",
    Spreecommerce: "Spreecommerce",
    StripeRelay: "StripeRelay",
    Walmart: "Walmart",
    Weebly: "Weebly",
    WooCommerce: "WooCommerce",
};
/** @internal */
exports.OrderShopAppEnum$inboundSchema = z.nativeEnum(exports.OrderShopAppEnum);
/** @internal */
exports.OrderShopAppEnum$outboundSchema = exports.OrderShopAppEnum$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var OrderShopAppEnum$;
(function (OrderShopAppEnum$) {
    /** @deprecated use `OrderShopAppEnum$inboundSchema` instead. */
    OrderShopAppEnum$.inboundSchema = exports.OrderShopAppEnum$inboundSchema;
    /** @deprecated use `OrderShopAppEnum$outboundSchema` instead. */
    OrderShopAppEnum$.outboundSchema = exports.OrderShopAppEnum$outboundSchema;
})(OrderShopAppEnum$ || (exports.OrderShopAppEnum$ = OrderShopAppEnum$ = {}));
