// shippo/models/components/parceltemplatearamexaustraliaenum.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ParcelTemplateAramexAustraliaEnum$ = exports.ParcelTemplateAramexAustraliaEnum$outboundSchema = exports.ParcelTemplateAramexAustraliaEnum$inboundSchema = exports.ParcelTemplateAramexAustraliaEnum = void 0;
const z = __importStar(require("zod"));
/**
 * |Token | Name | Dimensions|
 *
 * @remarks
 * |:---|:---|:---|
 * | Fastway_Australia_Satchel_A2 | Satchel A2 | 594.00 x 420.00 x 48.00 mm|
 * | Fastway_Australia_Satchel_A3 | Satchel A3 | 420.00 x 297.00 x 64.00 mm|
 * | Fastway_Australia_Satchel_A4 | Satchel A4 | 297.00 x 210.00 x 64.00 mm|
 * | Fastway_Australia_Satchel_A5 | Satchel A5 | 210.00 x 148.00 x 64.00 mm|
 */
exports.ParcelTemplateAramexAustraliaEnum = {
    FastwayAustraliaSatchelA2: "Fastway_Australia_Satchel_A2",
    FastwayAustraliaSatchelA3: "Fastway_Australia_Satchel_A3",
    FastwayAustraliaSatchelA4: "Fastway_Australia_Satchel_A4",
    FastwayAustraliaSatchelA5: "Fastway_Australia_Satchel_A5",
};
/** @internal */
exports.ParcelTemplateAramexAustraliaEnum$inboundSchema = z.nativeEnum(exports.ParcelTemplateAramexAustraliaEnum);
/** @internal */
exports.ParcelTemplateAramexAustraliaEnum$outboundSchema = exports.ParcelTemplateAramexAustraliaEnum$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ParcelTemplateAramexAustraliaEnum$;
(function (ParcelTemplateAramexAustraliaEnum$) {
    /** @deprecated use `ParcelTemplateAramexAustraliaEnum$inboundSchema` instead. */
    ParcelTemplateAramexAustraliaEnum$.inboundSchema = exports.ParcelTemplateAramexAustraliaEnum$inboundSchema;
    /** @deprecated use `ParcelTemplateAramexAustraliaEnum$outboundSchema` instead. */
    ParcelTemplateAramexAustraliaEnum$.outboundSchema = exports.ParcelTemplateAramexAustraliaEnum$outboundSchema;
})(ParcelTemplateAramexAustraliaEnum$ || (exports.ParcelTemplateAramexAustraliaEnum$ = ParcelTemplateAramexAustraliaEnum$ = {}));
