// shippo/models/components/parcelinsurance.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.parcelInsuranceFromJSON = exports.parcelInsuranceToJSON = exports.ParcelInsurance$ = exports.ParcelInsurance$outboundSchema = exports.ParcelInsurance$inboundSchema = exports.ParcelInsuranceProvider$ = exports.ParcelInsuranceProvider$outboundSchema = exports.ParcelInsuranceProvider$inboundSchema = exports.ParcelInsuranceProvider = void 0;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
/**
 * To have insurance cover provided by a carrier directly instead of Shippo's provider (XCover), set provider to `FEDEX`, `UPS`, or `ONTRAC`.
 */
exports.ParcelInsuranceProvider = {
    Fedex: "FEDEX",
    Ups: "UPS",
    Ontrac: "ONTRAC",
};
/** @internal */
exports.ParcelInsuranceProvider$inboundSchema = z.nativeEnum(exports.ParcelInsuranceProvider);
/** @internal */
exports.ParcelInsuranceProvider$outboundSchema = exports.ParcelInsuranceProvider$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ParcelInsuranceProvider$;
(function (ParcelInsuranceProvider$) {
    /** @deprecated use `ParcelInsuranceProvider$inboundSchema` instead. */
    ParcelInsuranceProvider$.inboundSchema = exports.ParcelInsuranceProvider$inboundSchema;
    /** @deprecated use `ParcelInsuranceProvider$outboundSchema` instead. */
    ParcelInsuranceProvider$.outboundSchema = exports.ParcelInsuranceProvider$outboundSchema;
})(ParcelInsuranceProvider$ || (exports.ParcelInsuranceProvider$ = ParcelInsuranceProvider$ = {}));
/** @internal */
exports.ParcelInsurance$inboundSchema = z.object({
    amount: z.string().optional(),
    content: z.string().optional(),
    currency: z.string().optional(),
    provider: exports.ParcelInsuranceProvider$inboundSchema.optional(),
});
/** @internal */
exports.ParcelInsurance$outboundSchema = z.object({
    amount: z.string().optional(),
    content: z.string().optional(),
    currency: z.string().optional(),
    provider: exports.ParcelInsuranceProvider$outboundSchema.optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ParcelInsurance$;
(function (ParcelInsurance$) {
    /** @deprecated use `ParcelInsurance$inboundSchema` instead. */
    ParcelInsurance$.inboundSchema = exports.ParcelInsurance$inboundSchema;
    /** @deprecated use `ParcelInsurance$outboundSchema` instead. */
    ParcelInsurance$.outboundSchema = exports.ParcelInsurance$outboundSchema;
})(ParcelInsurance$ || (exports.ParcelInsurance$ = ParcelInsurance$ = {}));
function parcelInsuranceToJSON(parcelInsurance) {
    return JSON.stringify(exports.ParcelInsurance$outboundSchema.parse(parcelInsurance));
}
exports.parcelInsuranceToJSON = parcelInsuranceToJSON;
function parcelInsuranceFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ParcelInsurance$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ParcelInsurance' from JSON`);
}
exports.parcelInsuranceFromJSON = parcelInsuranceFromJSON;
