// shippo/models/components/parceltemplatedhlecommerceenum.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ParcelTemplateDHLeCommerceEnum$ = exports.ParcelTemplateDHLeCommerceEnum$outboundSchema = exports.ParcelTemplateDHLeCommerceEnum$inboundSchema = exports.ParcelTemplateDHLeCommerceEnum = void 0;
const z = __importStar(require("zod"));
/**
 * |Token | Name | Dimensions|
 *
 * @remarks
 * |:---|:---|:---|
 * | DHLeC_Irregular | Irregular Shipment | 10.00 x 10.00 x 10.00 in|
 * | DHLeC_SM_Flats | Flats | 27.00 x 17.00 x 17.00 in|
 */
exports.ParcelTemplateDHLeCommerceEnum = {
    DHLeCIrregular: "DHLeC_Irregular",
    DHLeCSMFlats: "DHLeC_SM_Flats",
};
/** @internal */
exports.ParcelTemplateDHLeCommerceEnum$inboundSchema = z.nativeEnum(exports.ParcelTemplateDHLeCommerceEnum);
/** @internal */
exports.ParcelTemplateDHLeCommerceEnum$outboundSchema = exports.ParcelTemplateDHLeCommerceEnum$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ParcelTemplateDHLeCommerceEnum$;
(function (ParcelTemplateDHLeCommerceEnum$) {
    /** @deprecated use `ParcelTemplateDHLeCommerceEnum$inboundSchema` instead. */
    ParcelTemplateDHLeCommerceEnum$.inboundSchema = exports.ParcelTemplateDHLeCommerceEnum$inboundSchema;
    /** @deprecated use `ParcelTemplateDHLeCommerceEnum$outboundSchema` instead. */
    ParcelTemplateDHLeCommerceEnum$.outboundSchema = exports.ParcelTemplateDHLeCommerceEnum$outboundSchema;
})(ParcelTemplateDHLeCommerceEnum$ || (exports.ParcelTemplateDHLeCommerceEnum$ = ParcelTemplateDHLeCommerceEnum$ = {}));
