// shippo/models/components/parcelcreatefromtemplaterequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.parcelCreateFromTemplateRequestFromJSON = exports.parcelCreateFromTemplateRequestToJSON = exports.ParcelCreateFromTemplateRequest$ = exports.ParcelCreateFromTemplateRequest$outboundSchema = exports.ParcelCreateFromTemplateRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const parcelextra_js_1 = require("./parcelextra.js");
const parceltemplateenumset_js_1 = require("./parceltemplateenumset.js");
const weightunitenum_js_1 = require("./weightunitenum.js");
/** @internal */
exports.ParcelCreateFromTemplateRequest$inboundSchema = z.object({
    extra: parcelextra_js_1.ParcelExtra$inboundSchema.optional(),
    metadata: z.string().optional(),
    mass_unit: weightunitenum_js_1.WeightUnitEnum$inboundSchema,
    weight: z.string(),
    template: parceltemplateenumset_js_1.ParcelTemplateEnumSet$inboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "mass_unit": "massUnit",
    });
});
/** @internal */
exports.ParcelCreateFromTemplateRequest$outboundSchema = z.object({
    extra: parcelextra_js_1.ParcelExtra$outboundSchema.optional(),
    metadata: z.string().optional(),
    massUnit: weightunitenum_js_1.WeightUnitEnum$outboundSchema,
    weight: z.string(),
    template: parceltemplateenumset_js_1.ParcelTemplateEnumSet$outboundSchema,
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        massUnit: "mass_unit",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ParcelCreateFromTemplateRequest$;
(function (ParcelCreateFromTemplateRequest$) {
    /** @deprecated use `ParcelCreateFromTemplateRequest$inboundSchema` instead. */
    ParcelCreateFromTemplateRequest$.inboundSchema = exports.ParcelCreateFromTemplateRequest$inboundSchema;
    /** @deprecated use `ParcelCreateFromTemplateRequest$outboundSchema` instead. */
    ParcelCreateFromTemplateRequest$.outboundSchema = exports.ParcelCreateFromTemplateRequest$outboundSchema;
})(ParcelCreateFromTemplateRequest$ || (exports.ParcelCreateFromTemplateRequest$ = ParcelCreateFromTemplateRequest$ = {}));
function parcelCreateFromTemplateRequestToJSON(parcelCreateFromTemplateRequest) {
    return JSON.stringify(exports.ParcelCreateFromTemplateRequest$outboundSchema.parse(parcelCreateFromTemplateRequest));
}
exports.parcelCreateFromTemplateRequestToJSON = parcelCreateFromTemplateRequestToJSON;
function parcelCreateFromTemplateRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ParcelCreateFromTemplateRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ParcelCreateFromTemplateRequest' from JSON`);
}
exports.parcelCreateFromTemplateRequestFromJSON = parcelCreateFromTemplateRequestFromJSON;
