// shippo/models/components/ponumber.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.poNumberFromJSON = exports.poNumberToJSON = exports.PoNumber$ = exports.PoNumber$outboundSchema = exports.PoNumber$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.PoNumber$inboundSchema = z.object({
    prefix: z.string().optional(),
    value: z.string().optional(),
    ref_sort: z.number().int().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "ref_sort": "refSort",
    });
});
/** @internal */
exports.PoNumber$outboundSchema = z.object({
    prefix: z.string().optional(),
    value: z.string().optional(),
    refSort: z.number().int().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        refSort: "ref_sort",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var PoNumber$;
(function (PoNumber$) {
    /** @deprecated use `PoNumber$inboundSchema` instead. */
    PoNumber$.inboundSchema = exports.PoNumber$inboundSchema;
    /** @deprecated use `PoNumber$outboundSchema` instead. */
    PoNumber$.outboundSchema = exports.PoNumber$outboundSchema;
})(PoNumber$ || (exports.PoNumber$ = PoNumber$ = {}));
function poNumberToJSON(poNumber) {
    return JSON.stringify(exports.PoNumber$outboundSchema.parse(poNumber));
}
exports.poNumberToJSON = poNumberToJSON;
function poNumberFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.PoNumber$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PoNumber' from JSON`);
}
exports.poNumberFromJSON = poNumberFromJSON;
