// shippo/models/components/refund.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.refundFromJSON = exports.refundToJSON = exports.Refund$ = exports.Refund$outboundSchema = exports.Refund$inboundSchema = exports.RefundStatus$ = exports.RefundStatus$outboundSchema = exports.RefundStatus$inboundSchema = exports.RefundStatus = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/**
 * Indicates the status of the Refund.
 */
exports.RefundStatus = {
    Queued: "QUEUED",
    Pending: "PENDING",
    Success: "SUCCESS",
    Error: "ERROR",
};
/** @internal */
exports.RefundStatus$inboundSchema = z.nativeEnum(exports.RefundStatus);
/** @internal */
exports.RefundStatus$outboundSchema = exports.RefundStatus$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var RefundStatus$;
(function (RefundStatus$) {
    /** @deprecated use `RefundStatus$inboundSchema` instead. */
    RefundStatus$.inboundSchema = exports.RefundStatus$inboundSchema;
    /** @deprecated use `RefundStatus$outboundSchema` instead. */
    RefundStatus$.outboundSchema = exports.RefundStatus$outboundSchema;
})(RefundStatus$ || (exports.RefundStatus$ = RefundStatus$ = {}));
/** @internal */
exports.Refund$inboundSchema = z
    .object({
    object_created: z.string().datetime({ offset: true }).transform(v => new Date(v)).optional(),
    object_id: z.string().optional(),
    object_owner: z.string().optional(),
    object_updated: z.string().datetime({ offset: true }).transform(v => new Date(v)).optional(),
    status: exports.RefundStatus$inboundSchema.optional(),
    test: z.boolean().optional(),
    transaction: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "object_created": "objectCreated",
        "object_id": "objectId",
        "object_owner": "objectOwner",
        "object_updated": "objectUpdated",
    });
});
/** @internal */
exports.Refund$outboundSchema = z.object({
    objectCreated: z.date().transform(v => v.toISOString()).optional(),
    objectId: z.string().optional(),
    objectOwner: z.string().optional(),
    objectUpdated: z.date().transform(v => v.toISOString()).optional(),
    status: exports.RefundStatus$outboundSchema.optional(),
    test: z.boolean().optional(),
    transaction: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        objectCreated: "object_created",
        objectId: "object_id",
        objectOwner: "object_owner",
        objectUpdated: "object_updated",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Refund$;
(function (Refund$) {
    /** @deprecated use `Refund$inboundSchema` instead. */
    Refund$.inboundSchema = exports.Refund$inboundSchema;
    /** @deprecated use `Refund$outboundSchema` instead. */
    Refund$.outboundSchema = exports.Refund$outboundSchema;
})(Refund$ || (exports.Refund$ = Refund$ = {}));
function refundToJSON(refund) {
    return JSON.stringify(exports.Refund$outboundSchema.parse(refund));
}
exports.refundToJSON = refundToJSON;
function refundFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Refund$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Refund' from JSON`);
}
exports.refundFromJSON = refundFromJSON;
