// shippo/models/components/refundpaginatedlist.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.refundPaginatedListFromJSON = exports.refundPaginatedListToJSON = exports.RefundPaginatedList$ = exports.RefundPaginatedList$outboundSchema = exports.RefundPaginatedList$inboundSchema = void 0;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const refund_js_1 = require("./refund.js");
/** @internal */
exports.RefundPaginatedList$inboundSchema = z.object({
    next: z.string().optional(),
    previous: z.string().optional(),
    results: z.array(refund_js_1.Refund$inboundSchema).optional(),
});
/** @internal */
exports.RefundPaginatedList$outboundSchema = z.object({
    next: z.string().optional(),
    previous: z.string().optional(),
    results: z.array(refund_js_1.Refund$outboundSchema).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var RefundPaginatedList$;
(function (RefundPaginatedList$) {
    /** @deprecated use `RefundPaginatedList$inboundSchema` instead. */
    RefundPaginatedList$.inboundSchema = exports.RefundPaginatedList$inboundSchema;
    /** @deprecated use `RefundPaginatedList$outboundSchema` instead. */
    RefundPaginatedList$.outboundSchema = exports.RefundPaginatedList$outboundSchema;
})(RefundPaginatedList$ || (exports.RefundPaginatedList$ = RefundPaginatedList$ = {}));
function refundPaginatedListToJSON(refundPaginatedList) {
    return JSON.stringify(exports.RefundPaginatedList$outboundSchema.parse(refundPaginatedList));
}
exports.refundPaginatedListToJSON = refundPaginatedListToJSON;
function refundPaginatedListFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.RefundPaginatedList$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RefundPaginatedList' from JSON`);
}
exports.refundPaginatedListFromJSON = refundPaginatedListFromJSON;
