// shippo/models/components/rmanumber.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.rmaNumberFromJSON = exports.rmaNumberToJSON = exports.RmaNumber$ = exports.RmaNumber$outboundSchema = exports.RmaNumber$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.RmaNumber$inboundSchema = z.object({
    prefix: z.string().optional(),
    value: z.string().optional(),
    ref_sort: z.number().int().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "ref_sort": "refSort",
    });
});
/** @internal */
exports.RmaNumber$outboundSchema = z.object({
    prefix: z.string().optional(),
    value: z.string().optional(),
    refSort: z.number().int().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        refSort: "ref_sort",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var RmaNumber$;
(function (RmaNumber$) {
    /** @deprecated use `RmaNumber$inboundSchema` instead. */
    RmaNumber$.inboundSchema = exports.RmaNumber$inboundSchema;
    /** @deprecated use `RmaNumber$outboundSchema` instead. */
    RmaNumber$.outboundSchema = exports.RmaNumber$outboundSchema;
})(RmaNumber$ || (exports.RmaNumber$ = RmaNumber$ = {}));
function rmaNumberToJSON(rmaNumber) {
    return JSON.stringify(exports.RmaNumber$outboundSchema.parse(rmaNumber));
}
exports.rmaNumberToJSON = rmaNumberToJSON;
function rmaNumberFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.RmaNumber$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RmaNumber' from JSON`);
}
exports.rmaNumberFromJSON = rmaNumberFromJSON;
