// shippo/models/components/serviceleveluspsenum.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceLevelUSPSEnum$ = exports.ServiceLevelUSPSEnum$outboundSchema = exports.ServiceLevelUSPSEnum$inboundSchema = exports.ServiceLevelUSPSEnum = void 0;
const z = __importStar(require("zod"));
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | usps_priority | Priority Mail|
 * | usps_priority_express | Priority Mail Express|
 * | usps_media_mail | Media Mail, only for existing Shippo customers with grandfathered Media Mail option.|
 * | usps_priority_mail_international | Priority Mail International|
 * | usps_priority_mail_express_international | Priority Mail Express International|
 * | usps_first_class_package_international_service | First Class Package International|
 * | usps_ground_advantage | Ground Advantage |
 */
exports.ServiceLevelUSPSEnum = {
    UspsPriority: "usps_priority",
    UspsPriorityExpress: "usps_priority_express",
    UspsMediaMail: "usps_media_mail",
    UspsPriorityMailInternational: "usps_priority_mail_international",
    UspsPriorityMailExpressInternational: "usps_priority_mail_express_international",
    UspsFirstClassPackageInternationalService: "usps_first_class_package_international_service",
    UspsGroundAdvantage: "usps_ground_advantage",
};
/** @internal */
exports.ServiceLevelUSPSEnum$inboundSchema = z.nativeEnum(exports.ServiceLevelUSPSEnum);
/** @internal */
exports.ServiceLevelUSPSEnum$outboundSchema = exports.ServiceLevelUSPSEnum$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ServiceLevelUSPSEnum$;
(function (ServiceLevelUSPSEnum$) {
    /** @deprecated use `ServiceLevelUSPSEnum$inboundSchema` instead. */
    ServiceLevelUSPSEnum$.inboundSchema = exports.ServiceLevelUSPSEnum$inboundSchema;
    /** @deprecated use `ServiceLevelUSPSEnum$outboundSchema` instead. */
    ServiceLevelUSPSEnum$.outboundSchema = exports.ServiceLevelUSPSEnum$outboundSchema;
})(ServiceLevelUSPSEnum$ || (exports.ServiceLevelUSPSEnum$ = ServiceLevelUSPSEnum$ = {}));
