// shippo/models/components/servicelevelcolissimoenum.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceLevelColissimoEnum$ = exports.ServiceLevelColissimoEnum$outboundSchema = exports.ServiceLevelColissimoEnum$inboundSchema = exports.ServiceLevelColissimoEnum = void 0;
const z = __importStar(require("zod"));
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | colissimo_home | Domicile|
 * | colissimo_pick_up_point | Point Retrait|
 * | colissimo_return_mainland_france | Retour France|
 */
exports.ServiceLevelColissimoEnum = {
    ColissimoHome: "colissimo_home",
    ColissimoPickUpPoint: "colissimo_pick_up_point",
    ColissimoReturnMainlandFrance: "colissimo_return_mainland_france",
};
/** @internal */
exports.ServiceLevelColissimoEnum$inboundSchema = z.nativeEnum(exports.ServiceLevelColissimoEnum);
/** @internal */
exports.ServiceLevelColissimoEnum$outboundSchema = exports.ServiceLevelColissimoEnum$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ServiceLevelColissimoEnum$;
(function (ServiceLevelColissimoEnum$) {
    /** @deprecated use `ServiceLevelColissimoEnum$inboundSchema` instead. */
    ServiceLevelColissimoEnum$.inboundSchema = exports.ServiceLevelColissimoEnum$inboundSchema;
    /** @deprecated use `ServiceLevelColissimoEnum$outboundSchema` instead. */
    ServiceLevelColissimoEnum$.outboundSchema = exports.ServiceLevelColissimoEnum$outboundSchema;
})(ServiceLevelColissimoEnum$ || (exports.ServiceLevelColissimoEnum$ = ServiceLevelColissimoEnum$ = {}));
