// shippo/models/components/serviceleveljitsuenum.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceLevelJitsuEnum$ = exports.ServiceLevelJitsuEnum$outboundSchema = exports.ServiceLevelJitsuEnum$inboundSchema = exports.ServiceLevelJitsuEnum = void 0;
const z = __importStar(require("zod"));
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | axlehire_same_day | Same Day|
 * | axlehire_next_day | Next Day|
 */
exports.ServiceLevelJitsuEnum = {
    AxlehireSameDay: "axlehire_same_day",
    AxlehireNextDay: "axlehire_next_day",
};
/** @internal */
exports.ServiceLevelJitsuEnum$inboundSchema = z.nativeEnum(exports.ServiceLevelJitsuEnum);
/** @internal */
exports.ServiceLevelJitsuEnum$outboundSchema = exports.ServiceLevelJitsuEnum$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ServiceLevelJitsuEnum$;
(function (ServiceLevelJitsuEnum$) {
    /** @deprecated use `ServiceLevelJitsuEnum$inboundSchema` instead. */
    ServiceLevelJitsuEnum$.inboundSchema = exports.ServiceLevelJitsuEnum$inboundSchema;
    /** @deprecated use `ServiceLevelJitsuEnum$outboundSchema` instead. */
    ServiceLevelJitsuEnum$.outboundSchema = exports.ServiceLevelJitsuEnum$outboundSchema;
})(ServiceLevelJitsuEnum$ || (exports.ServiceLevelJitsuEnum$ = ServiceLevelJitsuEnum$ = {}));
