// shippo/models/components/servicegrouptypeenum.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceGroupTypeEnum$ = exports.ServiceGroupTypeEnum$outboundSchema = exports.ServiceGroupTypeEnum$inboundSchema = exports.ServiceGroupTypeEnum = void 0;
const z = __importStar(require("zod"));
/**
 * The type of the service group.<br>
 *
 * @remarks
 * `LIVE_RATE` - Shippo will make a rating request and return real-time rates for the shipping group, only falling back to the specified flat rate amount if no rates match a service level in the service group.<br>
 * `FLAT_RATE` - Returns a shipping option with the specified flat rate amount.<br>
 * `FREE_SHIPPING` - Returns a shipping option with a price of $0 only if the total cost of items exceeds the amount defined by `free_shipping_threshold_min`
 */
exports.ServiceGroupTypeEnum = {
    LiveRate: "LIVE_RATE",
    FlatRate: "FLAT_RATE",
    FreeShipping: "FREE_SHIPPING",
};
/** @internal */
exports.ServiceGroupTypeEnum$inboundSchema = z.nativeEnum(exports.ServiceGroupTypeEnum);
/** @internal */
exports.ServiceGroupTypeEnum$outboundSchema = exports.ServiceGroupTypeEnum$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ServiceGroupTypeEnum$;
(function (ServiceGroupTypeEnum$) {
    /** @deprecated use `ServiceGroupTypeEnum$inboundSchema` instead. */
    ServiceGroupTypeEnum$.inboundSchema = exports.ServiceGroupTypeEnum$inboundSchema;
    /** @deprecated use `ServiceGroupTypeEnum$outboundSchema` instead. */
    ServiceGroupTypeEnum$.outboundSchema = exports.ServiceGroupTypeEnum$outboundSchema;
})(ServiceGroupTypeEnum$ || (exports.ServiceGroupTypeEnum$ = ServiceGroupTypeEnum$ = {}));
