// shippo/models/components/servicelevelpurolatorenum.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceLevelPurolatorEnum$ = exports.ServiceLevelPurolatorEnum$outboundSchema = exports.ServiceLevelPurolatorEnum$inboundSchema = exports.ServiceLevelPurolatorEnum = void 0;
const z = __importStar(require("zod"));
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | purolator_ground | Ground|
 * | purolator_ground9_am | Ground 9am|
 * | purolator_ground1030_am | Ground 10:30am|
 * | purolator_ground_distribution | Ground Distribution|
 * | purolator_ground_evening | Ground Evening|
 * | purolator_ground_us | Ground US|
 * | purolator_express | Express|
 * | purolator_express9_am | Express 9am|
 * | purolator_express1030_am | Express 10am|
 * | purolator_express_evening | Express Evening|
 * | purolator_express_us | Express US|
 * | purolator_express_us9_am | Express US 9am|
 * | purolator_express_us1030_am | Express US 10:30am|
 * | purolator_express_us1200 | Express US 12pm|
 * | purolator_express_international | Express International|
 * | purolator_express_international9_am | Express International 9am|
 * | purolator_express_international1030_am | Express International 10:30am|
 * | purolator_express_international1200 | Express International 12pm|
 */
exports.ServiceLevelPurolatorEnum = {
    PurolatorGround: "purolator_ground",
    PurolatorGround9Am: "purolator_ground9_am",
    PurolatorGround1030Am: "purolator_ground1030_am",
    PurolatorGroundDistribution: "purolator_ground_distribution",
    PurolatorGroundEvening: "purolator_ground_evening",
    PurolatorGroundUs: "purolator_ground_us",
    PurolatorExpress: "purolator_express",
    PurolatorExpress9Am: "purolator_express9_am",
    PurolatorExpress1030Am: "purolator_express1030_am",
    PurolatorExpressEvening: "purolator_express_evening",
    PurolatorExpressUs: "purolator_express_us",
    PurolatorExpressUs9Am: "purolator_express_us9_am",
    PurolatorExpressUs1030Am: "purolator_express_us1030_am",
    PurolatorExpressUs1200: "purolator_express_us1200",
    PurolatorExpressInternational: "purolator_express_international",
    PurolatorExpressInternational9Am: "purolator_express_international9_am",
    PurolatorExpressInternational1030Am: "purolator_express_international1030_am",
    PurolatorExpressInternational1200: "purolator_express_international1200",
};
/** @internal */
exports.ServiceLevelPurolatorEnum$inboundSchema = z.nativeEnum(exports.ServiceLevelPurolatorEnum);
/** @internal */
exports.ServiceLevelPurolatorEnum$outboundSchema = exports.ServiceLevelPurolatorEnum$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ServiceLevelPurolatorEnum$;
(function (ServiceLevelPurolatorEnum$) {
    /** @deprecated use `ServiceLevelPurolatorEnum$inboundSchema` instead. */
    ServiceLevelPurolatorEnum$.inboundSchema = exports.ServiceLevelPurolatorEnum$inboundSchema;
    /** @deprecated use `ServiceLevelPurolatorEnum$outboundSchema` instead. */
    ServiceLevelPurolatorEnum$.outboundSchema = exports.ServiceLevelPurolatorEnum$outboundSchema;
})(ServiceLevelPurolatorEnum$ || (exports.ServiceLevelPurolatorEnum$ = ServiceLevelPurolatorEnum$ = {}));
