// shippo/models/components/servicelevelaustraliapostenum.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceLevelAustraliaPostEnum$ = exports.ServiceLevelAustraliaPostEnum$outboundSchema = exports.ServiceLevelAustraliaPostEnum$inboundSchema = exports.ServiceLevelAustraliaPostEnum = void 0;
const z = __importStar(require("zod"));
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | australia_post_express_post | Express Post|
 * | australia_post_parcel_post | Parcel Post|
 * | australia_post_pack_and_track_international | Pack and Track International|
 * | australia_post_international_airmail | International Airmail|
 * | australia_post_express_post_international | Express Post International|
 * | australia_post_express_courier_international | Express Courier International|
 * | australia_post_international_express | International Express|
 * | australia_post_international_standard | International Standard|
 * | australia_post_international_economy | International Economy|
 * | australia_post_parcel_post_return | Parcel Post Return|
 */
exports.ServiceLevelAustraliaPostEnum = {
    AustraliaPostExpressPost: "australia_post_express_post",
    AustraliaPostParcelPost: "australia_post_parcel_post",
    AustraliaPostPackAndTrackInternational: "australia_post_pack_and_track_international",
    AustraliaPostInternationalAirmail: "australia_post_international_airmail",
    AustraliaPostExpressPostInternational: "australia_post_express_post_international",
    AustraliaPostExpressCourierInternational: "australia_post_express_courier_international",
    AustraliaPostInternationalExpress: "australia_post_international_express",
    AustraliaPostInternationalStandard: "australia_post_international_standard",
    AustraliaPostInternationalEconomy: "australia_post_international_economy",
    AustraliaPostParcelPostReturn: "australia_post_parcel_post_return",
};
/** @internal */
exports.ServiceLevelAustraliaPostEnum$inboundSchema = z.nativeEnum(exports.ServiceLevelAustraliaPostEnum);
/** @internal */
exports.ServiceLevelAustraliaPostEnum$outboundSchema = exports.ServiceLevelAustraliaPostEnum$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ServiceLevelAustraliaPostEnum$;
(function (ServiceLevelAustraliaPostEnum$) {
    /** @deprecated use `ServiceLevelAustraliaPostEnum$inboundSchema` instead. */
    ServiceLevelAustraliaPostEnum$.inboundSchema = exports.ServiceLevelAustraliaPostEnum$inboundSchema;
    /** @deprecated use `ServiceLevelAustraliaPostEnum$outboundSchema` instead. */
    ServiceLevelAustraliaPostEnum$.outboundSchema = exports.ServiceLevelAustraliaPostEnum$outboundSchema;
})(ServiceLevelAustraliaPostEnum$ || (exports.ServiceLevelAustraliaPostEnum$ = ServiceLevelAustraliaPostEnum$ = {}));
