// shippo/models/components/servicelevelevriukenum.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceLevelEvriUKEnum$ = exports.ServiceLevelEvriUKEnum$outboundSchema = exports.ServiceLevelEvriUKEnum$inboundSchema = exports.ServiceLevelEvriUKEnum = void 0;
const z = __importStar(require("zod"));
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | hermes_uk_courier_service | Courier Collection|
 * | hermes_uk_parcelshop_dropoff | ParcelShop Drop-Off|
 * | hermes_uk_parcelshop_dropoff_nextday | ParcelShop Drop-Off Next Day|
 * | hermes_uk_postable | Postable|
 * | hermes_uk_postable_nextday | Postable Next Day|
 */
exports.ServiceLevelEvriUKEnum = {
    HermesUkCourierService: "hermes_uk_courier_service",
    HermesUkParcelshopDropoff: "hermes_uk_parcelshop_dropoff",
    HermesUkParcelshopDropoffNextday: "hermes_uk_parcelshop_dropoff_nextday",
    HermesUkPostable: "hermes_uk_postable",
    HermesUkPostableNextday: "hermes_uk_postable_nextday",
};
/** @internal */
exports.ServiceLevelEvriUKEnum$inboundSchema = z.nativeEnum(exports.ServiceLevelEvriUKEnum);
/** @internal */
exports.ServiceLevelEvriUKEnum$outboundSchema = exports.ServiceLevelEvriUKEnum$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ServiceLevelEvriUKEnum$;
(function (ServiceLevelEvriUKEnum$) {
    /** @deprecated use `ServiceLevelEvriUKEnum$inboundSchema` instead. */
    ServiceLevelEvriUKEnum$.inboundSchema = exports.ServiceLevelEvriUKEnum$inboundSchema;
    /** @deprecated use `ServiceLevelEvriUKEnum$outboundSchema` instead. */
    ServiceLevelEvriUKEnum$.outboundSchema = exports.ServiceLevelEvriUKEnum$outboundSchema;
})(ServiceLevelEvriUKEnum$ || (exports.ServiceLevelEvriUKEnum$ = ServiceLevelEvriUKEnum$ = {}));
