// shippo/models/components/servicelevelglsusenum.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceLevelGLSUSEnum$ = exports.ServiceLevelGLSUSEnum$outboundSchema = exports.ServiceLevelGLSUSEnum$inboundSchema = exports.ServiceLevelGLSUSEnum = void 0;
const z = __importStar(require("zod"));
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | gls_us_cps | GSO Ground|
 * | gls_us_eps | Early Priority Overnight|
 * | gls_us_ess | Early Saturday Delivery|
 * | gls_us_nps | Noon Priority Overnight|
 * | gls_us_pds | Priority Overnight|
 * | gls_us_sds | Saturday Delivery|
 */
exports.ServiceLevelGLSUSEnum = {
    GlsUsCps: "gls_us_cps",
    GlsUsEps: "gls_us_eps",
    GlsUsEss: "gls_us_ess",
    GlsUsNps: "gls_us_nps",
    GlsUsPds: "gls_us_pds",
    GlsUsSds: "gls_us_sds",
};
/** @internal */
exports.ServiceLevelGLSUSEnum$inboundSchema = z.nativeEnum(exports.ServiceLevelGLSUSEnum);
/** @internal */
exports.ServiceLevelGLSUSEnum$outboundSchema = exports.ServiceLevelGLSUSEnum$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ServiceLevelGLSUSEnum$;
(function (ServiceLevelGLSUSEnum$) {
    /** @deprecated use `ServiceLevelGLSUSEnum$inboundSchema` instead. */
    ServiceLevelGLSUSEnum$.inboundSchema = exports.ServiceLevelGLSUSEnum$inboundSchema;
    /** @deprecated use `ServiceLevelGLSUSEnum$outboundSchema` instead. */
    ServiceLevelGLSUSEnum$.outboundSchema = exports.ServiceLevelGLSUSEnum$outboundSchema;
})(ServiceLevelGLSUSEnum$ || (exports.ServiceLevelGLSUSEnum$ = ServiceLevelGLSUSEnum$ = {}));
