// shippo/models/components/servicelevelcanadapostenum.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceLevelCanadaPostEnum$ = exports.ServiceLevelCanadaPostEnum$outboundSchema = exports.ServiceLevelCanadaPostEnum$inboundSchema = exports.ServiceLevelCanadaPostEnum = void 0;
const z = __importStar(require("zod"));
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | canada_post_regular_parcel | Regular Parcel|
 * | canada_post_expedited_parcel | Expedited Parcel|
 * | canada_post_priority | Priority|
 * | canada_post_xpresspost | Xpresspost|
 * | canada_post_xpresspost_international | Xpresspost International|
 * | canada_post_xpresspost_usa | Xpresspost USA|
 * | canada_post_expedited_parcel_usa | Expedited Parcel USA|
 * | canada_post_tracked_packet_usa | Tracked Packet USA|
 * | canada_post_small_packet_usa_air | Small Packet USA Air|
 * | canada_post_tracked_packet_international | Tracked Packet International|
 * | canada_post_small_packet_international_air | Small Package International Air|
 * | canada_post_small_packet_international_surface  | Small Packet International Surface
 * | canada_post_international_parcel_surface  | International Parcel Surface
 * | canada_post_international_parcel_air  | International Parcel Air
 */
exports.ServiceLevelCanadaPostEnum = {
    CanadaPostRegularParcel: "canada_post_regular_parcel",
    CanadaPostExpeditedParcel: "canada_post_expedited_parcel",
    CanadaPostPriority: "canada_post_priority",
    CanadaPostXpresspost: "canada_post_xpresspost",
    CanadaPostXpresspostInternational: "canada_post_xpresspost_international",
    CanadaPostXpresspostUsa: "canada_post_xpresspost_usa",
    CanadaPostExpeditedParcelUsa: "canada_post_expedited_parcel_usa",
    CanadaPostTrackedPacketUsa: "canada_post_tracked_packet_usa",
    CanadaPostSmallPacketUsaAir: "canada_post_small_packet_usa_air",
    CanadaPostTrackedPacketInternational: "canada_post_tracked_packet_international",
    CanadaPostSmallPacketInternationalAir: "canada_post_small_packet_international_air",
    CanadaPostSmallPacketInternationalSurface: "canada_post_small_packet_international_surface",
    CanadaPostInternationalParcelSurface: "canada_post_international_parcel_surface",
    CanadaPostInternationalParcelAir: "canada_post_international_parcel_air",
};
/** @internal */
exports.ServiceLevelCanadaPostEnum$inboundSchema = z.nativeEnum(exports.ServiceLevelCanadaPostEnum);
/** @internal */
exports.ServiceLevelCanadaPostEnum$outboundSchema = exports.ServiceLevelCanadaPostEnum$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ServiceLevelCanadaPostEnum$;
(function (ServiceLevelCanadaPostEnum$) {
    /** @deprecated use `ServiceLevelCanadaPostEnum$inboundSchema` instead. */
    ServiceLevelCanadaPostEnum$.inboundSchema = exports.ServiceLevelCanadaPostEnum$inboundSchema;
    /** @deprecated use `ServiceLevelCanadaPostEnum$outboundSchema` instead. */
    ServiceLevelCanadaPostEnum$.outboundSchema = exports.ServiceLevelCanadaPostEnum$outboundSchema;
})(ServiceLevelCanadaPostEnum$ || (exports.ServiceLevelCanadaPostEnum$ = ServiceLevelCanadaPostEnum$ = {}));
