// shippo/models/components/servicegroupaccountandservicelevel.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.serviceGroupAccountAndServiceLevelFromJSON = exports.serviceGroupAccountAndServiceLevelToJSON = exports.ServiceGroupAccountAndServiceLevel$ = exports.ServiceGroupAccountAndServiceLevel$outboundSchema = exports.ServiceGroupAccountAndServiceLevel$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.ServiceGroupAccountAndServiceLevel$inboundSchema = z.object({
    account_object_id: z.string().optional(),
    service_level_token: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "account_object_id": "accountObjectId",
        "service_level_token": "serviceLevelToken",
    });
});
/** @internal */
exports.ServiceGroupAccountAndServiceLevel$outboundSchema = z.object({
    accountObjectId: z.string().optional(),
    serviceLevelToken: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accountObjectId: "account_object_id",
        serviceLevelToken: "service_level_token",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ServiceGroupAccountAndServiceLevel$;
(function (ServiceGroupAccountAndServiceLevel$) {
    /** @deprecated use `ServiceGroupAccountAndServiceLevel$inboundSchema` instead. */
    ServiceGroupAccountAndServiceLevel$.inboundSchema = exports.ServiceGroupAccountAndServiceLevel$inboundSchema;
    /** @deprecated use `ServiceGroupAccountAndServiceLevel$outboundSchema` instead. */
    ServiceGroupAccountAndServiceLevel$.outboundSchema = exports.ServiceGroupAccountAndServiceLevel$outboundSchema;
})(ServiceGroupAccountAndServiceLevel$ || (exports.ServiceGroupAccountAndServiceLevel$ = ServiceGroupAccountAndServiceLevel$ = {}));
function serviceGroupAccountAndServiceLevelToJSON(serviceGroupAccountAndServiceLevel) {
    return JSON.stringify(exports.ServiceGroupAccountAndServiceLevel$outboundSchema.parse(serviceGroupAccountAndServiceLevel));
}
exports.serviceGroupAccountAndServiceLevelToJSON = serviceGroupAccountAndServiceLevelToJSON;
function serviceGroupAccountAndServiceLevelFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ServiceGroupAccountAndServiceLevel$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ServiceGroupAccountAndServiceLevel' from JSON`);
}
exports.serviceGroupAccountAndServiceLevelFromJSON = serviceGroupAccountAndServiceLevelFromJSON;
