// shippo/models/components/servicelevelroyalmailenum.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceLevelRoyalMailEnum$ = exports.ServiceLevelRoyalMailEnum$outboundSchema = exports.ServiceLevelRoyalMailEnum$inboundSchema = exports.ServiceLevelRoyalMailEnum = void 0;
const z = __importStar(require("zod"));
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | royal_mail_tracked_letter_boxable_24_no_signature | Royal Mail Tracked Letter-Boxable 24 No Signature|
 * | royal_mail_tracked_letter_boxable_48_no_signature | Royal Mail Tracked Letter-Boxable 48 No Signature|
 * | royal_mail_tracked_24_returns | Royal Mail Tracked Returns 24|
 * | royal_mail_tracked_48_returns | Royal Mail Tracked Returns 48|
 * | royal_mail_special_delivery_guaranteed_1pm | Special Delivery Guaranteed by 1pm
 * | royal_mail_special_delivery_guaranteed_9am | Special Delivery Guaranteed by 9am
 * | royal_mail_24 | RoyalMail 24
 * | royal_mail_48 | RoyalMail 48
 * | royal_mail_signed_for_first_class | Signed for First Class
 * | royal_mail_signed_for_second_class | Signed for Second Class
 * | royal_mail_standard_first_class | Standard First Class
 * | royal_mail_standard_second_class | Standard Second Class
 * | royal_mail_tracked_24_no_signature | Royal Mail Tracked 24 No Signature
 * | royal_mail_tracked_48_no_signature | Royal Mail Ttacked 48 No Signature
 * | royal_mail_intl_bus_mail_lrg_ltr_zone_sort_pri | International Business Mail Large Letter Zone Sort Priority
 * | royal_mail_intl_bus_parcels_tracked_zone_sort | International Business Parcels Tracked Zone Sort
 * | royal_mail_intl_bus_parcels_tracked_country_priced | International Business Parcels Tracked Country Priced
 * | royal_mail_intl_bus_parcels_tracked_signed_zone_srt | International Business Parcels & Signed Zone Sort
 * | royal_mail_24_flat_rate | Royal Mail 24 Parcel Flat Rate
 * | royal_mail_48_flat_rate | Royal Mail 48 Parcel Flat Rate
 */
exports.ServiceLevelRoyalMailEnum = {
    RoyalMailTrackedLetterBoxable24NoSignature: "royal_mail_tracked_letter_boxable_24_no_signature",
    RoyalMailTrackedLetterBoxable48NoSignature: "royal_mail_tracked_letter_boxable_48_no_signature",
    RoyalMailTracked24Returns: "royal_mail_tracked_24_returns",
    RoyalMailTracked48Returns: "royal_mail_tracked_48_returns",
    RoyalMailSpecialDeliveryGuaranteed1pm: "royal_mail_special_delivery_guaranteed_1pm",
    RoyalMailSpecialDeliveryGuaranteed9am: "royal_mail_special_delivery_guaranteed_9am",
    RoyalMail24: "royal_mail_24",
    RoyalMail48: "royal_mail_48",
    RoyalMailSignedForFirstClass: "royal_mail_signed_for_first_class",
    RoyalMailSignedForSecondClass: "royal_mail_signed_for_second_class",
    RoyalMailStandardFirstClass: "royal_mail_standard_first_class",
    RoyalMailStandardSecondClass: "royal_mail_standard_second_class",
    RoyalMailTracked24NoSignature: "royal_mail_tracked_24_no_signature",
    RoyalMailTracked48NoSignature: "royal_mail_tracked_48_no_signature",
    RoyalMailIntlBusMailLrgLtrZoneSortPri: "royal_mail_intl_bus_mail_lrg_ltr_zone_sort_pri",
    RoyalMailIntlBusParcelsTrackedZoneSort: "royal_mail_intl_bus_parcels_tracked_zone_sort",
    RoyalMailIntlBusParcelsTrackedCountryPriced: "royal_mail_intl_bus_parcels_tracked_country_priced",
    RoyalMailIntlBusParcelsTrackedSignedZoneSrt: "royal_mail_intl_bus_parcels_tracked_signed_zone_srt",
    RoyalMail24FlatRate: "royal_mail_24_flat_rate",
    RoyalMail48FlatRate: "royal_mail_48_flat_rate",
};
/** @internal */
exports.ServiceLevelRoyalMailEnum$inboundSchema = z.nativeEnum(exports.ServiceLevelRoyalMailEnum);
/** @internal */
exports.ServiceLevelRoyalMailEnum$outboundSchema = exports.ServiceLevelRoyalMailEnum$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ServiceLevelRoyalMailEnum$;
(function (ServiceLevelRoyalMailEnum$) {
    /** @deprecated use `ServiceLevelRoyalMailEnum$inboundSchema` instead. */
    ServiceLevelRoyalMailEnum$.inboundSchema = exports.ServiceLevelRoyalMailEnum$inboundSchema;
    /** @deprecated use `ServiceLevelRoyalMailEnum$outboundSchema` instead. */
    ServiceLevelRoyalMailEnum$.outboundSchema = exports.ServiceLevelRoyalMailEnum$outboundSchema;
})(ServiceLevelRoyalMailEnum$ || (exports.ServiceLevelRoyalMailEnum$ = ServiceLevelRoyalMailEnum$ = {}));
