// shippo/models/components/serviceleveldhlgermanyenum.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceLevelDHLGermanyEnum$ = exports.ServiceLevelDHLGermanyEnum$outboundSchema = exports.ServiceLevelDHLGermanyEnum$inboundSchema = exports.ServiceLevelDHLGermanyEnum = void 0;
const z = __importStar(require("zod"));
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | dhl_germany_europaket | DHL Germany Europaket|
 * | dhl_germany_paket | DHL Germany Paket|
 * | dhl_germany_paket_connect | DHL Germany Paket Connect|
 * | dhl_germany_paket_international | DHL Germany Paket International|
 * | dhl_germany_paket_priority | DHL Germany Paket Priority|
 * | dhl_germany_paket_sameday | DHL Germany Paket Sameday|
 */
exports.ServiceLevelDHLGermanyEnum = {
    DhlGermanyEuropaket: "dhl_germany_europaket",
    DhlGermanyPaket: "dhl_germany_paket",
    DhlGermanyPaketConnect: "dhl_germany_paket_connect",
    DhlGermanyPaketInternational: "dhl_germany_paket_international",
    DhlGermanyPaketPriority: "dhl_germany_paket_priority",
    DhlGermanyPaketSameday: "dhl_germany_paket_sameday",
};
/** @internal */
exports.ServiceLevelDHLGermanyEnum$inboundSchema = z.nativeEnum(exports.ServiceLevelDHLGermanyEnum);
/** @internal */
exports.ServiceLevelDHLGermanyEnum$outboundSchema = exports.ServiceLevelDHLGermanyEnum$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ServiceLevelDHLGermanyEnum$;
(function (ServiceLevelDHLGermanyEnum$) {
    /** @deprecated use `ServiceLevelDHLGermanyEnum$inboundSchema` instead. */
    ServiceLevelDHLGermanyEnum$.inboundSchema = exports.ServiceLevelDHLGermanyEnum$inboundSchema;
    /** @deprecated use `ServiceLevelDHLGermanyEnum$outboundSchema` instead. */
    ServiceLevelDHLGermanyEnum$.outboundSchema = exports.ServiceLevelDHLGermanyEnum$outboundSchema;
})(ServiceLevelDHLGermanyEnum$ || (exports.ServiceLevelDHLGermanyEnum$ = ServiceLevelDHLGermanyEnum$ = {}));
