// shippo/models/components/serviceleveldeutschepostenum.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceLevelDeutschePostEnum$ = exports.ServiceLevelDeutschePostEnum$outboundSchema = exports.ServiceLevelDeutschePostEnum$inboundSchema = exports.ServiceLevelDeutschePostEnum = void 0;
const z = __importStar(require("zod"));
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | deutsche_post_postkarte | Postkarte|
 * | deutsche_post_standardbrief | Standardbrief|
 * | deutsche_post_kompaktbrief | Kompaktbrief|
 * | deutsche_post_grossbrief | Grossbrief|
 * | deutsche_post_maxibrief | Maxibrief|
 * | deutsche_post_maxibrief_plus | Maxibrief Plus|
 * | deutsche_post_warenpost_international_xs | Warenpost International XS|
 * | deutsche_post_warenpost_international_s | Warenpost International S|
 * | deutsche_post_warenpost_international_m | Warenpost International M|
 * | deutsche_post_warenpost_international_l | Warenpost International L|
 */
exports.ServiceLevelDeutschePostEnum = {
    DeutschePostPostkarte: "deutsche_post_postkarte",
    DeutschePostStandardbrief: "deutsche_post_standardbrief",
    DeutschePostKompaktbrief: "deutsche_post_kompaktbrief",
    DeutschePostGrossbrief: "deutsche_post_grossbrief",
    DeutschePostMaxibrief: "deutsche_post_maxibrief",
    DeutschePostMaxibriefPlus: "deutsche_post_maxibrief_plus",
    DeutschePostWarenpostInternationalXs: "deutsche_post_warenpost_international_xs",
    DeutschePostWarenpostInternationalS: "deutsche_post_warenpost_international_s",
    DeutschePostWarenpostInternationalM: "deutsche_post_warenpost_international_m",
    DeutschePostWarenpostInternationalL: "deutsche_post_warenpost_international_l",
};
/** @internal */
exports.ServiceLevelDeutschePostEnum$inboundSchema = z.nativeEnum(exports.ServiceLevelDeutschePostEnum);
/** @internal */
exports.ServiceLevelDeutschePostEnum$outboundSchema = exports.ServiceLevelDeutschePostEnum$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ServiceLevelDeutschePostEnum$;
(function (ServiceLevelDeutschePostEnum$) {
    /** @deprecated use `ServiceLevelDeutschePostEnum$inboundSchema` instead. */
    ServiceLevelDeutschePostEnum$.inboundSchema = exports.ServiceLevelDeutschePostEnum$inboundSchema;
    /** @deprecated use `ServiceLevelDeutschePostEnum$outboundSchema` instead. */
    ServiceLevelDeutschePostEnum$.outboundSchema = exports.ServiceLevelDeutschePostEnum$outboundSchema;
})(ServiceLevelDeutschePostEnum$ || (exports.ServiceLevelDeutschePostEnum$ = ServiceLevelDeutschePostEnum$ = {}));
