// shippo/models/components/servicelevelontracenum.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceLevelOnTracEnum$ = exports.ServiceLevelOnTracEnum$outboundSchema = exports.ServiceLevelOnTracEnum$inboundSchema = exports.ServiceLevelOnTracEnum = void 0;
const z = __importStar(require("zod"));
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | ontrac_ground | Ground|
 * | ontrac_sunrise_gold | Sunrise Gold|
 * | ontrac_sunrise | Sunrise|
 */
exports.ServiceLevelOnTracEnum = {
    OntracGround: "ontrac_ground",
    OntracSunriseGold: "ontrac_sunrise_gold",
    OntracSunrise: "ontrac_sunrise",
};
/** @internal */
exports.ServiceLevelOnTracEnum$inboundSchema = z.nativeEnum(exports.ServiceLevelOnTracEnum);
/** @internal */
exports.ServiceLevelOnTracEnum$outboundSchema = exports.ServiceLevelOnTracEnum$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ServiceLevelOnTracEnum$;
(function (ServiceLevelOnTracEnum$) {
    /** @deprecated use `ServiceLevelOnTracEnum$inboundSchema` instead. */
    ServiceLevelOnTracEnum$.inboundSchema = exports.ServiceLevelOnTracEnum$inboundSchema;
    /** @deprecated use `ServiceLevelOnTracEnum$outboundSchema` instead. */
    ServiceLevelOnTracEnum$.outboundSchema = exports.ServiceLevelOnTracEnum$outboundSchema;
})(ServiceLevelOnTracEnum$ || (exports.ServiceLevelOnTracEnum$ = ServiceLevelOnTracEnum$ = {}));
