// shippo/models/components/servicelevelparcelforceenum.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceLevelParcelforceEnum$ = exports.ServiceLevelParcelforceEnum$outboundSchema = exports.ServiceLevelParcelforceEnum$inboundSchema = exports.ServiceLevelParcelforceEnum = void 0;
const z = __importStar(require("zod"));
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | parcelforce_express48 | Express 48|
 * | parcelforce_express24 | Express 24|
 * | parcelforce_expressam | Express AM|
 */
exports.ServiceLevelParcelforceEnum = {
    ParcelforceExpress48: "parcelforce_express48",
    ParcelforceExpress24: "parcelforce_express24",
    ParcelforceExpressam: "parcelforce_expressam",
};
/** @internal */
exports.ServiceLevelParcelforceEnum$inboundSchema = z.nativeEnum(exports.ServiceLevelParcelforceEnum);
/** @internal */
exports.ServiceLevelParcelforceEnum$outboundSchema = exports.ServiceLevelParcelforceEnum$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ServiceLevelParcelforceEnum$;
(function (ServiceLevelParcelforceEnum$) {
    /** @deprecated use `ServiceLevelParcelforceEnum$inboundSchema` instead. */
    ServiceLevelParcelforceEnum$.inboundSchema = exports.ServiceLevelParcelforceEnum$inboundSchema;
    /** @deprecated use `ServiceLevelParcelforceEnum$outboundSchema` instead. */
    ServiceLevelParcelforceEnum$.outboundSchema = exports.ServiceLevelParcelforceEnum$outboundSchema;
})(ServiceLevelParcelforceEnum$ || (exports.ServiceLevelParcelforceEnum$ = ServiceLevelParcelforceEnum$ = {}));
