// shippo/models/components/serviceleveldhlecommerceenum.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceLevelDHLeCommerceEnum$ = exports.ServiceLevelDHLeCommerceEnum$outboundSchema = exports.ServiceLevelDHLeCommerceEnum$inboundSchema = exports.ServiceLevelDHLeCommerceEnum = void 0;
const z = __importStar(require("zod"));
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | dhl_ecommerce_marketing_parcel_expedited | Marketing Parcel Expedited|
 * | dhl_ecommerce_globalmail_business_ipa | GlobalMail Business IPA|
 * | dhl_ecommerce_parcel_international_direct | Parcel International Direct|
 * | dhl_ecommerce_parcels_expedited_max | Parcels Expedited Max|
 * | dhl_ecommerce_bpm_ground | Bounded Printed Matter Ground|
 * | dhl_ecommerce_priority_expedited | Priority Expedited|
 * | dhl_ecommerce_globalmail_packet_ipa | GlobalMail Packet IPA|
 * | dhl_ecommerce_globalmail_packet_isal | GlobalMail Packet ISAL|
 * | dhl_ecommerce_easy_return_plus | Easy Return Plus|
 * | dhl_ecommerce_marketing_parcel_ground | Marketing Parcel Ground|
 * | dhl_ecommerce_first_class_parcel_expedited | First Class Parcel Expedited|
 * | dhl_ecommerce_globalmail_business_priority | GlobalMail Business Priority|
 * | dhl_ecommerce_parcels_expedited | Parcels Expedited|
 * | dhl_ecommerce_globalmail_business_isal | GlobalMail Business ISAL|
 * | dhl_ecommerce_parcel_plus_expedited_max | Parcel Plus Expedited Max|
 * | dhl_ecommerce_globalmail_packet_plus | GlobalMail Packet Plus|
 * | dhl_ecommerce_parcels_ground | Parcels Ground|
 * | dhl_ecommerce_expedited | Expedited|
 * | dhl_ecommerce_parcel_plus_ground | Parcel Plus Ground|
 * | dhl_ecommerce_parcel_international_standard | Parcel International Standard|
 * | dhl_ecommerce_bpm_expedited | Bounded Printed Matter Expedited|
 * | dhl_ecommerce_parcel_international_expedited | Parcel International Expedited|
 * | dhl_ecommerce_globalmail_packet_priority | GlobalMail Packet Priority|
 * | dhl_ecommerce_easy_return_light | Easy Return Light|
 * | dhl_ecommerce_parcel_plus_expedited | Parcel Plus Expedited|
 * | dhl_ecommerce_globalmail_business_standard | GlobalMail Packet Standard|
 * | dhl_ecommerce_ground | Ground|
 * | dhl_ecommerce_globalmail_packet_standard | GlobalMail Business Standard|
 */
exports.ServiceLevelDHLeCommerceEnum = {
    DhlEcommerceMarketingParcelExpedited: "dhl_ecommerce_marketing_parcel_expedited",
    DhlEcommerceGlobalmailBusinessIpa: "dhl_ecommerce_globalmail_business_ipa",
    DhlEcommerceParcelInternationalDirect: "dhl_ecommerce_parcel_international_direct",
    DhlEcommerceParcelsExpeditedMax: "dhl_ecommerce_parcels_expedited_max",
    DhlEcommerceBpmGround: "dhl_ecommerce_bpm_ground",
    DhlEcommercePriorityExpedited: "dhl_ecommerce_priority_expedited",
    DhlEcommerceGlobalmailPacketIpa: "dhl_ecommerce_globalmail_packet_ipa",
    DhlEcommerceGlobalmailPacketIsal: "dhl_ecommerce_globalmail_packet_isal",
    DhlEcommerceEasyReturnPlus: "dhl_ecommerce_easy_return_plus",
    DhlEcommerceMarketingParcelGround: "dhl_ecommerce_marketing_parcel_ground",
    DhlEcommerceFirstClassParcelExpedited: "dhl_ecommerce_first_class_parcel_expedited",
    DhlEcommerceGlobalmailBusinessPriority: "dhl_ecommerce_globalmail_business_priority",
    DhlEcommerceParcelsExpedited: "dhl_ecommerce_parcels_expedited",
    DhlEcommerceGlobalmailBusinessIsal: "dhl_ecommerce_globalmail_business_isal",
    DhlEcommerceParcelPlusExpeditedMax: "dhl_ecommerce_parcel_plus_expedited_max",
    DhlEcommerceGlobalmailPacketPlus: "dhl_ecommerce_globalmail_packet_plus",
    DhlEcommerceParcelsGround: "dhl_ecommerce_parcels_ground",
    DhlEcommerceExpedited: "dhl_ecommerce_expedited",
    DhlEcommerceParcelPlusGround: "dhl_ecommerce_parcel_plus_ground",
    DhlEcommerceParcelInternationalStandard: "dhl_ecommerce_parcel_international_standard",
    DhlEcommerceBpmExpedited: "dhl_ecommerce_bpm_expedited",
    DhlEcommerceParcelInternationalExpedited: "dhl_ecommerce_parcel_international_expedited",
    DhlEcommerceGlobalmailPacketPriority: "dhl_ecommerce_globalmail_packet_priority",
    DhlEcommerceEasyReturnLight: "dhl_ecommerce_easy_return_light",
    DhlEcommerceParcelPlusExpedited: "dhl_ecommerce_parcel_plus_expedited",
    DhlEcommerceGlobalmailBusinessStandard: "dhl_ecommerce_globalmail_business_standard",
    DhlEcommerceGround: "dhl_ecommerce_ground",
    DhlEcommerceGlobalmailPacketStandard: "dhl_ecommerce_globalmail_packet_standard",
};
/** @internal */
exports.ServiceLevelDHLeCommerceEnum$inboundSchema = z.nativeEnum(exports.ServiceLevelDHLeCommerceEnum);
/** @internal */
exports.ServiceLevelDHLeCommerceEnum$outboundSchema = exports.ServiceLevelDHLeCommerceEnum$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ServiceLevelDHLeCommerceEnum$;
(function (ServiceLevelDHLeCommerceEnum$) {
    /** @deprecated use `ServiceLevelDHLeCommerceEnum$inboundSchema` instead. */
    ServiceLevelDHLeCommerceEnum$.inboundSchema = exports.ServiceLevelDHLeCommerceEnum$inboundSchema;
    /** @deprecated use `ServiceLevelDHLeCommerceEnum$outboundSchema` instead. */
    ServiceLevelDHLeCommerceEnum$.outboundSchema = exports.ServiceLevelDHLeCommerceEnum$outboundSchema;
})(ServiceLevelDHLeCommerceEnum$ || (exports.ServiceLevelDHLeCommerceEnum$ = ServiceLevelDHLeCommerceEnum$ = {}));
