// shippo/models/components/servicelevelsendleenum.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceLevelSendleEnum$ = exports.ServiceLevelSendleEnum$outboundSchema = exports.ServiceLevelSendleEnum$inboundSchema = exports.ServiceLevelSendleEnum = void 0;
const z = __importStar(require("zod"));
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | sendle_parcel | Sendle Parcel|
 * | sendle_standard_dropoff | Sendle Standard Dropoff|
 */
exports.ServiceLevelSendleEnum = {
    SendleParcel: "sendle_parcel",
    SendleStandardDropoff: "sendle_standard_dropoff",
};
/** @internal */
exports.ServiceLevelSendleEnum$inboundSchema = z.nativeEnum(exports.ServiceLevelSendleEnum);
/** @internal */
exports.ServiceLevelSendleEnum$outboundSchema = exports.ServiceLevelSendleEnum$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ServiceLevelSendleEnum$;
(function (ServiceLevelSendleEnum$) {
    /** @deprecated use `ServiceLevelSendleEnum$inboundSchema` instead. */
    ServiceLevelSendleEnum$.inboundSchema = exports.ServiceLevelSendleEnum$inboundSchema;
    /** @deprecated use `ServiceLevelSendleEnum$outboundSchema` instead. */
    ServiceLevelSendleEnum$.outboundSchema = exports.ServiceLevelSendleEnum$outboundSchema;
})(ServiceLevelSendleEnum$ || (exports.ServiceLevelSendleEnum$ = ServiceLevelSendleEnum$ = {}));
