// shippo/models/components/servicelevelaramexaustraliaenum.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceLevelAramexAustraliaEnum$ = exports.ServiceLevelAramexAustraliaEnum$outboundSchema = exports.ServiceLevelAramexAustraliaEnum$inboundSchema = exports.ServiceLevelAramexAustraliaEnum = void 0;
const z = __importStar(require("zod"));
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | fastway_australia_parcel | Parcel|
 * | fastway_australia_satchel | Satchel|
 * | fastway_australia_box_small | Box Small|
 * | fastway_australia_box_medium | Box Medium|
 * | fastway_australia_box_large | Box Large|
 */
exports.ServiceLevelAramexAustraliaEnum = {
    FastwayAustraliaParcel: "fastway_australia_parcel",
    FastwayAustraliaSatchel: "fastway_australia_satchel",
    FastwayAustraliaBoxSmall: "fastway_australia_box_small",
    FastwayAustraliaBoxMedium: "fastway_australia_box_medium",
    FastwayAustraliaBoxLarge: "fastway_australia_box_large",
};
/** @internal */
exports.ServiceLevelAramexAustraliaEnum$inboundSchema = z.nativeEnum(exports.ServiceLevelAramexAustraliaEnum);
/** @internal */
exports.ServiceLevelAramexAustraliaEnum$outboundSchema = exports.ServiceLevelAramexAustraliaEnum$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ServiceLevelAramexAustraliaEnum$;
(function (ServiceLevelAramexAustraliaEnum$) {
    /** @deprecated use `ServiceLevelAramexAustraliaEnum$inboundSchema` instead. */
    ServiceLevelAramexAustraliaEnum$.inboundSchema = exports.ServiceLevelAramexAustraliaEnum$inboundSchema;
    /** @deprecated use `ServiceLevelAramexAustraliaEnum$outboundSchema` instead. */
    ServiceLevelAramexAustraliaEnum$.outboundSchema = exports.ServiceLevelAramexAustraliaEnum$outboundSchema;
})(ServiceLevelAramexAustraliaEnum$ || (exports.ServiceLevelAramexAustraliaEnum$ = ServiceLevelAramexAustraliaEnum$ = {}));
