// shippo/models/components/shipmentextrareturnservicetypeupsenum.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ShipmentExtraReturnServiceTypeUPSEnum$ = exports.ShipmentExtraReturnServiceTypeUPSEnum$outboundSchema = exports.ShipmentExtraReturnServiceTypeUPSEnum$inboundSchema = exports.ShipmentExtraReturnServiceTypeUPSEnum = void 0;
const z = __importStar(require("zod"));
exports.ShipmentExtraReturnServiceTypeUPSEnum = {
    PrintAndMail: "PRINT_AND_MAIL",
    Attempt1: "ATTEMPT_1",
    Attempt3: "ATTEMPT_3",
    ElectronicLabel: "ELECTRONIC_LABEL",
};
/** @internal */
exports.ShipmentExtraReturnServiceTypeUPSEnum$inboundSchema = z.nativeEnum(exports.ShipmentExtraReturnServiceTypeUPSEnum);
/** @internal */
exports.ShipmentExtraReturnServiceTypeUPSEnum$outboundSchema = exports.ShipmentExtraReturnServiceTypeUPSEnum$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ShipmentExtraReturnServiceTypeUPSEnum$;
(function (ShipmentExtraReturnServiceTypeUPSEnum$) {
    /** @deprecated use `ShipmentExtraReturnServiceTypeUPSEnum$inboundSchema` instead. */
    ShipmentExtraReturnServiceTypeUPSEnum$.inboundSchema = exports.ShipmentExtraReturnServiceTypeUPSEnum$inboundSchema;
    /** @deprecated use `ShipmentExtraReturnServiceTypeUPSEnum$outboundSchema` instead. */
    ShipmentExtraReturnServiceTypeUPSEnum$.outboundSchema = exports.ShipmentExtraReturnServiceTypeUPSEnum$outboundSchema;
})(ShipmentExtraReturnServiceTypeUPSEnum$ || (exports.ShipmentExtraReturnServiceTypeUPSEnum$ = ShipmentExtraReturnServiceTypeUPSEnum$ = {}));
