// shippo/models/components/shipmentpaginatedlist.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.shipmentPaginatedListFromJSON = exports.shipmentPaginatedListToJSON = exports.ShipmentPaginatedList$ = exports.ShipmentPaginatedList$outboundSchema = exports.ShipmentPaginatedList$inboundSchema = void 0;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const shipment_js_1 = require("./shipment.js");
/** @internal */
exports.ShipmentPaginatedList$inboundSchema = z.object({
    next: z.string().optional(),
    previous: z.string().optional(),
    results: z.array(shipment_js_1.Shipment$inboundSchema).optional(),
});
/** @internal */
exports.ShipmentPaginatedList$outboundSchema = z.object({
    next: z.string().optional(),
    previous: z.string().optional(),
    results: z.array(shipment_js_1.Shipment$outboundSchema).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ShipmentPaginatedList$;
(function (ShipmentPaginatedList$) {
    /** @deprecated use `ShipmentPaginatedList$inboundSchema` instead. */
    ShipmentPaginatedList$.inboundSchema = exports.ShipmentPaginatedList$inboundSchema;
    /** @deprecated use `ShipmentPaginatedList$outboundSchema` instead. */
    ShipmentPaginatedList$.outboundSchema = exports.ShipmentPaginatedList$outboundSchema;
})(ShipmentPaginatedList$ || (exports.ShipmentPaginatedList$ = ShipmentPaginatedList$ = {}));
function shipmentPaginatedListToJSON(shipmentPaginatedList) {
    return JSON.stringify(exports.ShipmentPaginatedList$outboundSchema.parse(shipmentPaginatedList));
}
exports.shipmentPaginatedListToJSON = shipmentPaginatedListToJSON;
function shipmentPaginatedListFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ShipmentPaginatedList$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ShipmentPaginatedList' from JSON`);
}
exports.shipmentPaginatedListFromJSON = shipmentPaginatedListFromJSON;
