// shippo/models/components/shippoaccountpaginatedlist.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.shippoAccountPaginatedListFromJSON = exports.shippoAccountPaginatedListToJSON = exports.ShippoAccountPaginatedList$ = exports.ShippoAccountPaginatedList$outboundSchema = exports.ShippoAccountPaginatedList$inboundSchema = void 0;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const shippoaccount_js_1 = require("./shippoaccount.js");
/** @internal */
exports.ShippoAccountPaginatedList$inboundSchema = z.object({
    next: z.string().optional(),
    previous: z.string().optional(),
    results: z.array(shippoaccount_js_1.ShippoAccount$inboundSchema).optional(),
});
/** @internal */
exports.ShippoAccountPaginatedList$outboundSchema = z.object({
    next: z.string().optional(),
    previous: z.string().optional(),
    results: z.array(shippoaccount_js_1.ShippoAccount$outboundSchema).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ShippoAccountPaginatedList$;
(function (ShippoAccountPaginatedList$) {
    /** @deprecated use `ShippoAccountPaginatedList$inboundSchema` instead. */
    ShippoAccountPaginatedList$.inboundSchema = exports.ShippoAccountPaginatedList$inboundSchema;
    /** @deprecated use `ShippoAccountPaginatedList$outboundSchema` instead. */
    ShippoAccountPaginatedList$.outboundSchema = exports.ShippoAccountPaginatedList$outboundSchema;
})(ShippoAccountPaginatedList$ || (exports.ShippoAccountPaginatedList$ = ShippoAccountPaginatedList$ = {}));
function shippoAccountPaginatedListToJSON(shippoAccountPaginatedList) {
    return JSON.stringify(exports.ShippoAccountPaginatedList$outboundSchema.parse(shippoAccountPaginatedList));
}
exports.shippoAccountPaginatedListToJSON = shippoAccountPaginatedListToJSON;
function shippoAccountPaginatedListFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ShippoAccountPaginatedList$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ShippoAccountPaginatedList' from JSON`);
}
exports.shippoAccountPaginatedListFromJSON = shippoAccountPaginatedListFromJSON;
