// shippo/models/components/shippoaccount.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.shippoAccountFromJSON = exports.shippoAccountToJSON = exports.ShippoAccount$ = exports.ShippoAccount$outboundSchema = exports.ShippoAccount$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.ShippoAccount$inboundSchema = z.object({
    email: z.string(),
    first_name: z.string(),
    last_name: z.string(),
    company_name: z.string().optional(),
    object_created: z.string().datetime({ offset: true }).transform(v => new Date(v)).optional(),
    object_id: z.string().optional(),
    object_updated: z.string().datetime({ offset: true }).transform(v => new Date(v)).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "first_name": "firstName",
        "last_name": "lastName",
        "company_name": "companyName",
        "object_created": "objectCreated",
        "object_id": "objectId",
        "object_updated": "objectUpdated",
    });
});
/** @internal */
exports.ShippoAccount$outboundSchema = z.object({
    email: z.string(),
    firstName: z.string(),
    lastName: z.string(),
    companyName: z.string().optional(),
    objectCreated: z.date().transform(v => v.toISOString()).optional(),
    objectId: z.string().optional(),
    objectUpdated: z.date().transform(v => v.toISOString()).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        firstName: "first_name",
        lastName: "last_name",
        companyName: "company_name",
        objectCreated: "object_created",
        objectId: "object_id",
        objectUpdated: "object_updated",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ShippoAccount$;
(function (ShippoAccount$) {
    /** @deprecated use `ShippoAccount$inboundSchema` instead. */
    ShippoAccount$.inboundSchema = exports.ShippoAccount$inboundSchema;
    /** @deprecated use `ShippoAccount$outboundSchema` instead. */
    ShippoAccount$.outboundSchema = exports.ShippoAccount$outboundSchema;
})(ShippoAccount$ || (exports.ShippoAccount$ = ShippoAccount$ = {}));
function shippoAccountToJSON(shippoAccount) {
    return JSON.stringify(exports.ShippoAccount$outboundSchema.parse(shippoAccount));
}
exports.shippoAccountToJSON = shippoAccountToJSON;
function shippoAccountFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ShippoAccount$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ShippoAccount' from JSON`);
}
exports.shippoAccountFromJSON = shippoAccountFromJSON;
