// shippo/models/components/shipmentcreaterequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.shipmentCreateRequestFromJSON = exports.shipmentCreateRequestToJSON = exports.ShipmentCreateRequest$ = exports.ShipmentCreateRequest$outboundSchema = exports.ShipmentCreateRequest$inboundSchema = exports.parcelsFromJSON = exports.parcelsToJSON = exports.Parcels$ = exports.Parcels$outboundSchema = exports.Parcels$inboundSchema = exports.shipmentCreateRequestCustomsDeclarationFromJSON = exports.shipmentCreateRequestCustomsDeclarationToJSON = exports.ShipmentCreateRequestCustomsDeclaration$ = exports.ShipmentCreateRequestCustomsDeclaration$outboundSchema = exports.ShipmentCreateRequestCustomsDeclaration$inboundSchema = exports.addressToFromJSON = exports.addressToToJSON = exports.AddressTo$ = exports.AddressTo$outboundSchema = exports.AddressTo$inboundSchema = exports.addressReturnFromJSON = exports.addressReturnToJSON = exports.AddressReturn$ = exports.AddressReturn$outboundSchema = exports.AddressReturn$inboundSchema = exports.addressFromFromJSON = exports.addressFromToJSON = exports.AddressFrom$ = exports.AddressFrom$outboundSchema = exports.AddressFrom$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const addresscreaterequest_js_1 = require("./addresscreaterequest.js");
const customsdeclarationcreaterequest_js_1 = require("./customsdeclarationcreaterequest.js");
const parcelcreatefromtemplaterequest_js_1 = require("./parcelcreatefromtemplaterequest.js");
const parcelcreaterequest_js_1 = require("./parcelcreaterequest.js");
const shipmentextra_js_1 = require("./shipmentextra.js");
/** @internal */
exports.AddressFrom$inboundSchema = z.union([addresscreaterequest_js_1.AddressCreateRequest$inboundSchema, z.string()]);
/** @internal */
exports.AddressFrom$outboundSchema = z.union([addresscreaterequest_js_1.AddressCreateRequest$outboundSchema, z.string()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AddressFrom$;
(function (AddressFrom$) {
    /** @deprecated use `AddressFrom$inboundSchema` instead. */
    AddressFrom$.inboundSchema = exports.AddressFrom$inboundSchema;
    /** @deprecated use `AddressFrom$outboundSchema` instead. */
    AddressFrom$.outboundSchema = exports.AddressFrom$outboundSchema;
})(AddressFrom$ || (exports.AddressFrom$ = AddressFrom$ = {}));
function addressFromToJSON(addressFrom) {
    return JSON.stringify(exports.AddressFrom$outboundSchema.parse(addressFrom));
}
exports.addressFromToJSON = addressFromToJSON;
function addressFromFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.AddressFrom$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'AddressFrom' from JSON`);
}
exports.addressFromFromJSON = addressFromFromJSON;
/** @internal */
exports.AddressReturn$inboundSchema = z.union([addresscreaterequest_js_1.AddressCreateRequest$inboundSchema, z.string()]);
/** @internal */
exports.AddressReturn$outboundSchema = z.union([addresscreaterequest_js_1.AddressCreateRequest$outboundSchema, z.string()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AddressReturn$;
(function (AddressReturn$) {
    /** @deprecated use `AddressReturn$inboundSchema` instead. */
    AddressReturn$.inboundSchema = exports.AddressReturn$inboundSchema;
    /** @deprecated use `AddressReturn$outboundSchema` instead. */
    AddressReturn$.outboundSchema = exports.AddressReturn$outboundSchema;
})(AddressReturn$ || (exports.AddressReturn$ = AddressReturn$ = {}));
function addressReturnToJSON(addressReturn) {
    return JSON.stringify(exports.AddressReturn$outboundSchema.parse(addressReturn));
}
exports.addressReturnToJSON = addressReturnToJSON;
function addressReturnFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.AddressReturn$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'AddressReturn' from JSON`);
}
exports.addressReturnFromJSON = addressReturnFromJSON;
/** @internal */
exports.AddressTo$inboundSchema = z.union([addresscreaterequest_js_1.AddressCreateRequest$inboundSchema, z.string()]);
/** @internal */
exports.AddressTo$outboundSchema = z.union([addresscreaterequest_js_1.AddressCreateRequest$outboundSchema, z.string()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var AddressTo$;
(function (AddressTo$) {
    /** @deprecated use `AddressTo$inboundSchema` instead. */
    AddressTo$.inboundSchema = exports.AddressTo$inboundSchema;
    /** @deprecated use `AddressTo$outboundSchema` instead. */
    AddressTo$.outboundSchema = exports.AddressTo$outboundSchema;
})(AddressTo$ || (exports.AddressTo$ = AddressTo$ = {}));
function addressToToJSON(addressTo) {
    return JSON.stringify(exports.AddressTo$outboundSchema.parse(addressTo));
}
exports.addressToToJSON = addressToToJSON;
function addressToFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.AddressTo$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'AddressTo' from JSON`);
}
exports.addressToFromJSON = addressToFromJSON;
/** @internal */
exports.ShipmentCreateRequestCustomsDeclaration$inboundSchema = z.union([customsdeclarationcreaterequest_js_1.CustomsDeclarationCreateRequest$inboundSchema, z.string()]);
/** @internal */
exports.ShipmentCreateRequestCustomsDeclaration$outboundSchema = z.union([customsdeclarationcreaterequest_js_1.CustomsDeclarationCreateRequest$outboundSchema, z.string()]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ShipmentCreateRequestCustomsDeclaration$;
(function (ShipmentCreateRequestCustomsDeclaration$) {
    /** @deprecated use `ShipmentCreateRequestCustomsDeclaration$inboundSchema` instead. */
    ShipmentCreateRequestCustomsDeclaration$.inboundSchema = exports.ShipmentCreateRequestCustomsDeclaration$inboundSchema;
    /** @deprecated use `ShipmentCreateRequestCustomsDeclaration$outboundSchema` instead. */
    ShipmentCreateRequestCustomsDeclaration$.outboundSchema = exports.ShipmentCreateRequestCustomsDeclaration$outboundSchema;
})(ShipmentCreateRequestCustomsDeclaration$ || (exports.ShipmentCreateRequestCustomsDeclaration$ = ShipmentCreateRequestCustomsDeclaration$ = {}));
function shipmentCreateRequestCustomsDeclarationToJSON(shipmentCreateRequestCustomsDeclaration) {
    return JSON.stringify(exports.ShipmentCreateRequestCustomsDeclaration$outboundSchema.parse(shipmentCreateRequestCustomsDeclaration));
}
exports.shipmentCreateRequestCustomsDeclarationToJSON = shipmentCreateRequestCustomsDeclarationToJSON;
function shipmentCreateRequestCustomsDeclarationFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ShipmentCreateRequestCustomsDeclaration$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ShipmentCreateRequestCustomsDeclaration' from JSON`);
}
exports.shipmentCreateRequestCustomsDeclarationFromJSON = shipmentCreateRequestCustomsDeclarationFromJSON;
/** @internal */
exports.Parcels$inboundSchema = z.union([
    parcelcreatefromtemplaterequest_js_1.ParcelCreateFromTemplateRequest$inboundSchema,
    parcelcreaterequest_js_1.ParcelCreateRequest$inboundSchema,
    z.string(),
]);
/** @internal */
exports.Parcels$outboundSchema = z.union([
    parcelcreatefromtemplaterequest_js_1.ParcelCreateFromTemplateRequest$outboundSchema,
    parcelcreaterequest_js_1.ParcelCreateRequest$outboundSchema,
    z.string(),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Parcels$;
(function (Parcels$) {
    /** @deprecated use `Parcels$inboundSchema` instead. */
    Parcels$.inboundSchema = exports.Parcels$inboundSchema;
    /** @deprecated use `Parcels$outboundSchema` instead. */
    Parcels$.outboundSchema = exports.Parcels$outboundSchema;
})(Parcels$ || (exports.Parcels$ = Parcels$ = {}));
function parcelsToJSON(parcels) {
    return JSON.stringify(exports.Parcels$outboundSchema.parse(parcels));
}
exports.parcelsToJSON = parcelsToJSON;
function parcelsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Parcels$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Parcels' from JSON`);
}
exports.parcelsFromJSON = parcelsFromJSON;
/** @internal */
exports.ShipmentCreateRequest$inboundSchema = z.object({
    extra: shipmentextra_js_1.ShipmentExtra$inboundSchema.optional(),
    metadata: z.string().optional(),
    shipment_date: z.string().optional(),
    address_from: z.union([addresscreaterequest_js_1.AddressCreateRequest$inboundSchema, z.string()]),
    address_return: z.union([addresscreaterequest_js_1.AddressCreateRequest$inboundSchema, z.string()])
        .optional(),
    address_to: z.union([addresscreaterequest_js_1.AddressCreateRequest$inboundSchema, z.string()]),
    customs_declaration: z.union([
        customsdeclarationcreaterequest_js_1.CustomsDeclarationCreateRequest$inboundSchema,
        z.string(),
    ]).optional(),
    async: z.boolean().optional(),
    carrier_accounts: z.array(z.string()).optional(),
    parcels: z.array(z.union([
        parcelcreatefromtemplaterequest_js_1.ParcelCreateFromTemplateRequest$inboundSchema,
        parcelcreaterequest_js_1.ParcelCreateRequest$inboundSchema,
        z.string(),
    ])),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "shipment_date": "shipmentDate",
        "address_from": "addressFrom",
        "address_return": "addressReturn",
        "address_to": "addressTo",
        "customs_declaration": "customsDeclaration",
        "carrier_accounts": "carrierAccounts",
    });
});
/** @internal */
exports.ShipmentCreateRequest$outboundSchema = z.object({
    extra: shipmentextra_js_1.ShipmentExtra$outboundSchema.optional(),
    metadata: z.string().optional(),
    shipmentDate: z.string().optional(),
    addressFrom: z.union([addresscreaterequest_js_1.AddressCreateRequest$outboundSchema, z.string()]),
    addressReturn: z.union([addresscreaterequest_js_1.AddressCreateRequest$outboundSchema, z.string()])
        .optional(),
    addressTo: z.union([addresscreaterequest_js_1.AddressCreateRequest$outboundSchema, z.string()]),
    customsDeclaration: z.union([
        customsdeclarationcreaterequest_js_1.CustomsDeclarationCreateRequest$outboundSchema,
        z.string(),
    ]).optional(),
    async: z.boolean().optional(),
    carrierAccounts: z.array(z.string()).optional(),
    parcels: z.array(z.union([
        parcelcreatefromtemplaterequest_js_1.ParcelCreateFromTemplateRequest$outboundSchema,
        parcelcreaterequest_js_1.ParcelCreateRequest$outboundSchema,
        z.string(),
    ])),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        shipmentDate: "shipment_date",
        addressFrom: "address_from",
        addressReturn: "address_return",
        addressTo: "address_to",
        customsDeclaration: "customs_declaration",
        carrierAccounts: "carrier_accounts",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ShipmentCreateRequest$;
(function (ShipmentCreateRequest$) {
    /** @deprecated use `ShipmentCreateRequest$inboundSchema` instead. */
    ShipmentCreateRequest$.inboundSchema = exports.ShipmentCreateRequest$inboundSchema;
    /** @deprecated use `ShipmentCreateRequest$outboundSchema` instead. */
    ShipmentCreateRequest$.outboundSchema = exports.ShipmentCreateRequest$outboundSchema;
})(ShipmentCreateRequest$ || (exports.ShipmentCreateRequest$ = ShipmentCreateRequest$ = {}));
function shipmentCreateRequestToJSON(shipmentCreateRequest) {
    return JSON.stringify(exports.ShipmentCreateRequest$outboundSchema.parse(shipmentCreateRequest));
}
exports.shipmentCreateRequestToJSON = shipmentCreateRequestToJSON;
function shipmentCreateRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ShipmentCreateRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ShipmentCreateRequest' from JSON`);
}
exports.shipmentCreateRequestFromJSON = shipmentCreateRequestFromJSON;
