// shippo/models/components/shippoaccountupdaterequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.shippoAccountUpdateRequestFromJSON = exports.shippoAccountUpdateRequestToJSON = exports.ShippoAccountUpdateRequest$ = exports.ShippoAccountUpdateRequest$outboundSchema = exports.ShippoAccountUpdateRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.ShippoAccountUpdateRequest$inboundSchema = z.object({
    email: z.string(),
    first_name: z.string(),
    last_name: z.string(),
    company_name: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "first_name": "firstName",
        "last_name": "lastName",
        "company_name": "companyName",
    });
});
/** @internal */
exports.ShippoAccountUpdateRequest$outboundSchema = z.object({
    email: z.string(),
    firstName: z.string(),
    lastName: z.string(),
    companyName: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        firstName: "first_name",
        lastName: "last_name",
        companyName: "company_name",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ShippoAccountUpdateRequest$;
(function (ShippoAccountUpdateRequest$) {
    /** @deprecated use `ShippoAccountUpdateRequest$inboundSchema` instead. */
    ShippoAccountUpdateRequest$.inboundSchema = exports.ShippoAccountUpdateRequest$inboundSchema;
    /** @deprecated use `ShippoAccountUpdateRequest$outboundSchema` instead. */
    ShippoAccountUpdateRequest$.outboundSchema = exports.ShippoAccountUpdateRequest$outboundSchema;
})(ShippoAccountUpdateRequest$ || (exports.ShippoAccountUpdateRequest$ = ShippoAccountUpdateRequest$ = {}));
function shippoAccountUpdateRequestToJSON(shippoAccountUpdateRequest) {
    return JSON.stringify(exports.ShippoAccountUpdateRequest$outboundSchema.parse(shippoAccountUpdateRequest));
}
exports.shippoAccountUpdateRequestToJSON = shippoAccountUpdateRequestToJSON;
function shippoAccountUpdateRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ShippoAccountUpdateRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ShippoAccountUpdateRequest' from JSON`);
}
exports.shippoAccountUpdateRequestFromJSON = shippoAccountUpdateRequestFromJSON;
