// shippo/models/components/trackingstatuslocationbase.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.trackingStatusLocationBaseFromJSON = exports.trackingStatusLocationBaseToJSON = exports.TrackingStatusLocationBase$ = exports.TrackingStatusLocationBase$outboundSchema = exports.TrackingStatusLocationBase$inboundSchema = void 0;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.TrackingStatusLocationBase$inboundSchema = z.object({
    city: z.string().optional(),
    country: z.string().optional(),
    state: z.string().optional(),
    zip: z.string().optional(),
});
/** @internal */
exports.TrackingStatusLocationBase$outboundSchema = z.object({
    city: z.string().optional(),
    country: z.string().optional(),
    state: z.string().optional(),
    zip: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var TrackingStatusLocationBase$;
(function (TrackingStatusLocationBase$) {
    /** @deprecated use `TrackingStatusLocationBase$inboundSchema` instead. */
    TrackingStatusLocationBase$.inboundSchema = exports.TrackingStatusLocationBase$inboundSchema;
    /** @deprecated use `TrackingStatusLocationBase$outboundSchema` instead. */
    TrackingStatusLocationBase$.outboundSchema = exports.TrackingStatusLocationBase$outboundSchema;
})(TrackingStatusLocationBase$ || (exports.TrackingStatusLocationBase$ = TrackingStatusLocationBase$ = {}));
function trackingStatusLocationBaseToJSON(trackingStatusLocationBase) {
    return JSON.stringify(exports.TrackingStatusLocationBase$outboundSchema.parse(trackingStatusLocationBase));
}
exports.trackingStatusLocationBaseToJSON = trackingStatusLocationBaseToJSON;
function trackingStatusLocationBaseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.TrackingStatusLocationBase$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'TrackingStatusLocationBase' from JSON`);
}
exports.trackingStatusLocationBaseFromJSON = trackingStatusLocationBaseFromJSON;
