// shippo/models/components/tracksrequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.tracksRequestFromJSON = exports.tracksRequestToJSON = exports.TracksRequest$ = exports.TracksRequest$outboundSchema = exports.TracksRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.TracksRequest$inboundSchema = z.object({
    carrier: z.string(),
    metadata: z.string().optional(),
    tracking_number: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "tracking_number": "trackingNumber",
    });
});
/** @internal */
exports.TracksRequest$outboundSchema = z.object({
    carrier: z.string(),
    metadata: z.string().optional(),
    trackingNumber: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        trackingNumber: "tracking_number",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var TracksRequest$;
(function (TracksRequest$) {
    /** @deprecated use `TracksRequest$inboundSchema` instead. */
    TracksRequest$.inboundSchema = exports.TracksRequest$inboundSchema;
    /** @deprecated use `TracksRequest$outboundSchema` instead. */
    TracksRequest$.outboundSchema = exports.TracksRequest$outboundSchema;
})(TracksRequest$ || (exports.TracksRequest$ = TracksRequest$ = {}));
function tracksRequestToJSON(tracksRequest) {
    return JSON.stringify(exports.TracksRequest$outboundSchema.parse(tracksRequest));
}
exports.tracksRequestToJSON = tracksRequestToJSON;
function tracksRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.TracksRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'TracksRequest' from JSON`);
}
exports.tracksRequestFromJSON = tracksRequestFromJSON;
