// shippo/models/components/trackingstatussubstatus.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.trackingStatusSubstatusFromJSON = exports.trackingStatusSubstatusToJSON = exports.TrackingStatusSubstatus$ = exports.TrackingStatusSubstatus$outboundSchema = exports.TrackingStatusSubstatus$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.TrackingStatusSubstatus$inboundSchema = z.object({
    code: z.string(),
    text: z.string(),
    action_required: z.boolean(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "action_required": "actionRequired",
    });
});
/** @internal */
exports.TrackingStatusSubstatus$outboundSchema = z.object({
    code: z.string(),
    text: z.string(),
    actionRequired: z.boolean(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        actionRequired: "action_required",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var TrackingStatusSubstatus$;
(function (TrackingStatusSubstatus$) {
    /** @deprecated use `TrackingStatusSubstatus$inboundSchema` instead. */
    TrackingStatusSubstatus$.inboundSchema = exports.TrackingStatusSubstatus$inboundSchema;
    /** @deprecated use `TrackingStatusSubstatus$outboundSchema` instead. */
    TrackingStatusSubstatus$.outboundSchema = exports.TrackingStatusSubstatus$outboundSchema;
})(TrackingStatusSubstatus$ || (exports.TrackingStatusSubstatus$ = TrackingStatusSubstatus$ = {}));
function trackingStatusSubstatusToJSON(trackingStatusSubstatus) {
    return JSON.stringify(exports.TrackingStatusSubstatus$outboundSchema.parse(trackingStatusSubstatus));
}
exports.trackingStatusSubstatusToJSON = trackingStatusSubstatusToJSON;
function trackingStatusSubstatusFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.TrackingStatusSubstatus$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'TrackingStatusSubstatus' from JSON`);
}
exports.trackingStatusSubstatusFromJSON = trackingStatusSubstatusFromJSON;
