// shippo/models/components/transactionpaginatedlist.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.transactionPaginatedListFromJSON = exports.transactionPaginatedListToJSON = exports.TransactionPaginatedList$ = exports.TransactionPaginatedList$outboundSchema = exports.TransactionPaginatedList$inboundSchema = void 0;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const transaction_js_1 = require("./transaction.js");
/** @internal */
exports.TransactionPaginatedList$inboundSchema = z.object({
    next: z.string().optional(),
    previous: z.string().optional(),
    results: z.array(transaction_js_1.Transaction$inboundSchema).optional(),
});
/** @internal */
exports.TransactionPaginatedList$outboundSchema = z.object({
    next: z.string().optional(),
    previous: z.string().optional(),
    results: z.array(transaction_js_1.Transaction$outboundSchema).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var TransactionPaginatedList$;
(function (TransactionPaginatedList$) {
    /** @deprecated use `TransactionPaginatedList$inboundSchema` instead. */
    TransactionPaginatedList$.inboundSchema = exports.TransactionPaginatedList$inboundSchema;
    /** @deprecated use `TransactionPaginatedList$outboundSchema` instead. */
    TransactionPaginatedList$.outboundSchema = exports.TransactionPaginatedList$outboundSchema;
})(TransactionPaginatedList$ || (exports.TransactionPaginatedList$ = TransactionPaginatedList$ = {}));
function transactionPaginatedListToJSON(transactionPaginatedList) {
    return JSON.stringify(exports.TransactionPaginatedList$outboundSchema.parse(transactionPaginatedList));
}
exports.transactionPaginatedListToJSON = transactionPaginatedListToJSON;
function transactionPaginatedListFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.TransactionPaginatedList$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'TransactionPaginatedList' from JSON`);
}
exports.transactionPaginatedListFromJSON = transactionPaginatedListFromJSON;
