// shippo/models/components/transactioncreaterequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.transactionCreateRequestFromJSON = exports.transactionCreateRequestToJSON = exports.TransactionCreateRequest$ = exports.TransactionCreateRequest$outboundSchema = exports.TransactionCreateRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const labelfiletypeenum_js_1 = require("./labelfiletypeenum.js");
/** @internal */
exports.TransactionCreateRequest$inboundSchema = z.object({
    async: z.boolean().default(true),
    label_file_type: labelfiletypeenum_js_1.LabelFileTypeEnum$inboundSchema.optional(),
    metadata: z.string().optional(),
    rate: z.string(),
    order: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "label_file_type": "labelFileType",
    });
});
/** @internal */
exports.TransactionCreateRequest$outboundSchema = z.object({
    async: z.boolean().default(true),
    labelFileType: labelfiletypeenum_js_1.LabelFileTypeEnum$outboundSchema.optional(),
    metadata: z.string().optional(),
    rate: z.string(),
    order: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        labelFileType: "label_file_type",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var TransactionCreateRequest$;
(function (TransactionCreateRequest$) {
    /** @deprecated use `TransactionCreateRequest$inboundSchema` instead. */
    TransactionCreateRequest$.inboundSchema = exports.TransactionCreateRequest$inboundSchema;
    /** @deprecated use `TransactionCreateRequest$outboundSchema` instead. */
    TransactionCreateRequest$.outboundSchema = exports.TransactionCreateRequest$outboundSchema;
})(TransactionCreateRequest$ || (exports.TransactionCreateRequest$ = TransactionCreateRequest$ = {}));
function transactionCreateRequestToJSON(transactionCreateRequest) {
    return JSON.stringify(exports.TransactionCreateRequest$outboundSchema.parse(transactionCreateRequest));
}
exports.transactionCreateRequestToJSON = transactionCreateRequestToJSON;
function transactionCreateRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.TransactionCreateRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'TransactionCreateRequest' from JSON`);
}
exports.transactionCreateRequestFromJSON = transactionCreateRequestFromJSON;
