// shippo/models/components/upsconnectexistingownaccountparameters.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.upsConnectExistingOwnAccountParametersFromJSON = exports.upsConnectExistingOwnAccountParametersToJSON = exports.UPSConnectExistingOwnAccountParameters$ = exports.UPSConnectExistingOwnAccountParameters$outboundSchema = exports.UPSConnectExistingOwnAccountParameters$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.UPSConnectExistingOwnAccountParameters$inboundSchema = z.object({
    account_number: z.string(),
    aia_country_iso2: z.string().optional(),
    billing_address_city: z.string(),
    billing_address_country_iso2: z.string(),
    billing_address_state: z.string(),
    billing_address_street1: z.string(),
    billing_address_street2: z.string().optional(),
    billing_address_zip: z.string(),
    collec_country_iso2: z.string(),
    collec_zip: z.string(),
    company: z.string(),
    currency_code: z.string().optional(),
    email: z.string(),
    full_name: z.string(),
    has_invoice: z.boolean(),
    invoice_controlid: z.string().optional(),
    invoice_date: z.string().optional(),
    invoice_number: z.string().optional(),
    invoice_value: z.string().optional(),
    phone: z.string(),
    title: z.string(),
    ups_agreements: z.boolean(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "account_number": "accountNumber",
        "aia_country_iso2": "aiaCountryIso2",
        "billing_address_city": "billingAddressCity",
        "billing_address_country_iso2": "billingAddressCountryIso2",
        "billing_address_state": "billingAddressState",
        "billing_address_street1": "billingAddressStreet1",
        "billing_address_street2": "billingAddressStreet2",
        "billing_address_zip": "billingAddressZip",
        "collec_country_iso2": "collecCountryIso2",
        "collec_zip": "collecZip",
        "currency_code": "currencyCode",
        "full_name": "fullName",
        "has_invoice": "hasInvoice",
        "invoice_controlid": "invoiceControlid",
        "invoice_date": "invoiceDate",
        "invoice_number": "invoiceNumber",
        "invoice_value": "invoiceValue",
        "ups_agreements": "upsAgreements",
    });
});
/** @internal */
exports.UPSConnectExistingOwnAccountParameters$outboundSchema = z.object({
    accountNumber: z.string(),
    aiaCountryIso2: z.string().optional(),
    billingAddressCity: z.string(),
    billingAddressCountryIso2: z.string(),
    billingAddressState: z.string(),
    billingAddressStreet1: z.string(),
    billingAddressStreet2: z.string().optional(),
    billingAddressZip: z.string(),
    collecCountryIso2: z.string(),
    collecZip: z.string(),
    company: z.string(),
    currencyCode: z.string().optional(),
    email: z.string(),
    fullName: z.string(),
    hasInvoice: z.boolean(),
    invoiceControlid: z.string().optional(),
    invoiceDate: z.string().optional(),
    invoiceNumber: z.string().optional(),
    invoiceValue: z.string().optional(),
    phone: z.string(),
    title: z.string(),
    upsAgreements: z.boolean(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        accountNumber: "account_number",
        aiaCountryIso2: "aia_country_iso2",
        billingAddressCity: "billing_address_city",
        billingAddressCountryIso2: "billing_address_country_iso2",
        billingAddressState: "billing_address_state",
        billingAddressStreet1: "billing_address_street1",
        billingAddressStreet2: "billing_address_street2",
        billingAddressZip: "billing_address_zip",
        collecCountryIso2: "collec_country_iso2",
        collecZip: "collec_zip",
        currencyCode: "currency_code",
        fullName: "full_name",
        hasInvoice: "has_invoice",
        invoiceControlid: "invoice_controlid",
        invoiceDate: "invoice_date",
        invoiceNumber: "invoice_number",
        invoiceValue: "invoice_value",
        upsAgreements: "ups_agreements",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var UPSConnectExistingOwnAccountParameters$;
(function (UPSConnectExistingOwnAccountParameters$) {
    /** @deprecated use `UPSConnectExistingOwnAccountParameters$inboundSchema` instead. */
    UPSConnectExistingOwnAccountParameters$.inboundSchema = exports.UPSConnectExistingOwnAccountParameters$inboundSchema;
    /** @deprecated use `UPSConnectExistingOwnAccountParameters$outboundSchema` instead. */
    UPSConnectExistingOwnAccountParameters$.outboundSchema = exports.UPSConnectExistingOwnAccountParameters$outboundSchema;
})(UPSConnectExistingOwnAccountParameters$ || (exports.UPSConnectExistingOwnAccountParameters$ = UPSConnectExistingOwnAccountParameters$ = {}));
function upsConnectExistingOwnAccountParametersToJSON(upsConnectExistingOwnAccountParameters) {
    return JSON.stringify(exports.UPSConnectExistingOwnAccountParameters$outboundSchema.parse(upsConnectExistingOwnAccountParameters));
}
exports.upsConnectExistingOwnAccountParametersToJSON = upsConnectExistingOwnAccountParametersToJSON;
function upsConnectExistingOwnAccountParametersFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.UPSConnectExistingOwnAccountParameters$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UPSConnectExistingOwnAccountParameters' from JSON`);
}
exports.upsConnectExistingOwnAccountParametersFromJSON = upsConnectExistingOwnAccountParametersFromJSON;
