// shippo/models/components/userparceltemplateupdaterequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.userParcelTemplateUpdateRequestFromJSON = exports.userParcelTemplateUpdateRequestToJSON = exports.UserParcelTemplateUpdateRequest$ = exports.UserParcelTemplateUpdateRequest$outboundSchema = exports.UserParcelTemplateUpdateRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const distanceunitenum_js_1 = require("./distanceunitenum.js");
const weightunitenum_js_1 = require("./weightunitenum.js");
/** @internal */
exports.UserParcelTemplateUpdateRequest$inboundSchema = z.object({
    distance_unit: distanceunitenum_js_1.DistanceUnitEnum$inboundSchema,
    height: z.string(),
    length: z.string(),
    name: z.string(),
    weight: z.string().optional(),
    weight_unit: weightunitenum_js_1.WeightUnitEnum$inboundSchema.optional(),
    width: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "distance_unit": "distanceUnit",
        "weight_unit": "weightUnit",
    });
});
/** @internal */
exports.UserParcelTemplateUpdateRequest$outboundSchema = z.object({
    distanceUnit: distanceunitenum_js_1.DistanceUnitEnum$outboundSchema,
    height: z.string(),
    length: z.string(),
    name: z.string(),
    weight: z.string().optional(),
    weightUnit: weightunitenum_js_1.WeightUnitEnum$outboundSchema.optional(),
    width: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        distanceUnit: "distance_unit",
        weightUnit: "weight_unit",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var UserParcelTemplateUpdateRequest$;
(function (UserParcelTemplateUpdateRequest$) {
    /** @deprecated use `UserParcelTemplateUpdateRequest$inboundSchema` instead. */
    UserParcelTemplateUpdateRequest$.inboundSchema = exports.UserParcelTemplateUpdateRequest$inboundSchema;
    /** @deprecated use `UserParcelTemplateUpdateRequest$outboundSchema` instead. */
    UserParcelTemplateUpdateRequest$.outboundSchema = exports.UserParcelTemplateUpdateRequest$outboundSchema;
})(UserParcelTemplateUpdateRequest$ || (exports.UserParcelTemplateUpdateRequest$ = UserParcelTemplateUpdateRequest$ = {}));
function userParcelTemplateUpdateRequestToJSON(userParcelTemplateUpdateRequest) {
    return JSON.stringify(exports.UserParcelTemplateUpdateRequest$outboundSchema.parse(userParcelTemplateUpdateRequest));
}
exports.userParcelTemplateUpdateRequestToJSON = userParcelTemplateUpdateRequestToJSON;
function userParcelTemplateUpdateRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.UserParcelTemplateUpdateRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UserParcelTemplateUpdateRequest' from JSON`);
}
exports.userParcelTemplateUpdateRequestFromJSON = userParcelTemplateUpdateRequestFromJSON;
