// shippo/models/components/userparceltemplatelist.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.userParcelTemplateListFromJSON = exports.userParcelTemplateListToJSON = exports.UserParcelTemplateList$ = exports.UserParcelTemplateList$outboundSchema = exports.UserParcelTemplateList$inboundSchema = void 0;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const userparceltemplate_js_1 = require("./userparceltemplate.js");
/** @internal */
exports.UserParcelTemplateList$inboundSchema = z.object({
    results: z.array(userparceltemplate_js_1.UserParcelTemplate$inboundSchema).optional(),
});
/** @internal */
exports.UserParcelTemplateList$outboundSchema = z.object({
    results: z.array(userparceltemplate_js_1.UserParcelTemplate$outboundSchema).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var UserParcelTemplateList$;
(function (UserParcelTemplateList$) {
    /** @deprecated use `UserParcelTemplateList$inboundSchema` instead. */
    UserParcelTemplateList$.inboundSchema = exports.UserParcelTemplateList$inboundSchema;
    /** @deprecated use `UserParcelTemplateList$outboundSchema` instead. */
    UserParcelTemplateList$.outboundSchema = exports.UserParcelTemplateList$outboundSchema;
})(UserParcelTemplateList$ || (exports.UserParcelTemplateList$ = UserParcelTemplateList$ = {}));
function userParcelTemplateListToJSON(userParcelTemplateList) {
    return JSON.stringify(exports.UserParcelTemplateList$outboundSchema.parse(userParcelTemplateList));
}
exports.userParcelTemplateListToJSON = userParcelTemplateListToJSON;
function userParcelTemplateListFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.UserParcelTemplateList$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UserParcelTemplateList' from JSON`);
}
exports.userParcelTemplateListFromJSON = userParcelTemplateListFromJSON;
