// shippo/models/components/userparceltemplatewithcarriertemplatecreaterequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.userParcelTemplateWithCarrierTemplateCreateRequestFromJSON = exports.userParcelTemplateWithCarrierTemplateCreateRequestToJSON = exports.UserParcelTemplateWithCarrierTemplateCreateRequest$ = exports.UserParcelTemplateWithCarrierTemplateCreateRequest$outboundSchema = exports.UserParcelTemplateWithCarrierTemplateCreateRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const weightunitenum_js_1 = require("./weightunitenum.js");
/** @internal */
exports.UserParcelTemplateWithCarrierTemplateCreateRequest$inboundSchema = z.object({
    template: z.string().optional(),
    weight: z.string().optional(),
    weight_unit: weightunitenum_js_1.WeightUnitEnum$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "weight_unit": "weightUnit",
    });
});
/** @internal */
exports.UserParcelTemplateWithCarrierTemplateCreateRequest$outboundSchema = z.object({
    template: z.string().optional(),
    weight: z.string().optional(),
    weightUnit: weightunitenum_js_1.WeightUnitEnum$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        weightUnit: "weight_unit",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var UserParcelTemplateWithCarrierTemplateCreateRequest$;
(function (UserParcelTemplateWithCarrierTemplateCreateRequest$) {
    /** @deprecated use `UserParcelTemplateWithCarrierTemplateCreateRequest$inboundSchema` instead. */
    UserParcelTemplateWithCarrierTemplateCreateRequest$.inboundSchema = exports.UserParcelTemplateWithCarrierTemplateCreateRequest$inboundSchema;
    /** @deprecated use `UserParcelTemplateWithCarrierTemplateCreateRequest$outboundSchema` instead. */
    UserParcelTemplateWithCarrierTemplateCreateRequest$.outboundSchema = exports.UserParcelTemplateWithCarrierTemplateCreateRequest$outboundSchema;
})(UserParcelTemplateWithCarrierTemplateCreateRequest$ || (exports.UserParcelTemplateWithCarrierTemplateCreateRequest$ = UserParcelTemplateWithCarrierTemplateCreateRequest$ = {}));
function userParcelTemplateWithCarrierTemplateCreateRequestToJSON(userParcelTemplateWithCarrierTemplateCreateRequest) {
    return JSON.stringify(exports.UserParcelTemplateWithCarrierTemplateCreateRequest$outboundSchema.parse(userParcelTemplateWithCarrierTemplateCreateRequest));
}
exports.userParcelTemplateWithCarrierTemplateCreateRequestToJSON = userParcelTemplateWithCarrierTemplateCreateRequestToJSON;
function userParcelTemplateWithCarrierTemplateCreateRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.UserParcelTemplateWithCarrierTemplateCreateRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UserParcelTemplateWithCarrierTemplateCreateRequest' from JSON`);
}
exports.userParcelTemplateWithCarrierTemplateCreateRequestFromJSON = userParcelTemplateWithCarrierTemplateCreateRequestFromJSON;
