// shippo/models/components/webhookpaginatedlist.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.webhookPaginatedListFromJSON = exports.webhookPaginatedListToJSON = exports.WebhookPaginatedList$ = exports.WebhookPaginatedList$outboundSchema = exports.WebhookPaginatedList$inboundSchema = void 0;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const webhook_js_1 = require("./webhook.js");
/** @internal */
exports.WebhookPaginatedList$inboundSchema = z.object({
    next: z.string().optional(),
    previous: z.string().optional(),
    count: z.number().int().optional(),
    results: z.array(webhook_js_1.Webhook$inboundSchema).optional(),
});
/** @internal */
exports.WebhookPaginatedList$outboundSchema = z.object({
    next: z.string().optional(),
    previous: z.string().optional(),
    count: z.number().int().optional(),
    results: z.array(webhook_js_1.Webhook$outboundSchema).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebhookPaginatedList$;
(function (WebhookPaginatedList$) {
    /** @deprecated use `WebhookPaginatedList$inboundSchema` instead. */
    WebhookPaginatedList$.inboundSchema = exports.WebhookPaginatedList$inboundSchema;
    /** @deprecated use `WebhookPaginatedList$outboundSchema` instead. */
    WebhookPaginatedList$.outboundSchema = exports.WebhookPaginatedList$outboundSchema;
})(WebhookPaginatedList$ || (exports.WebhookPaginatedList$ = WebhookPaginatedList$ = {}));
function webhookPaginatedListToJSON(webhookPaginatedList) {
    return JSON.stringify(exports.WebhookPaginatedList$outboundSchema.parse(webhookPaginatedList));
}
exports.webhookPaginatedListToJSON = webhookPaginatedListToJSON;
function webhookPaginatedListFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.WebhookPaginatedList$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookPaginatedList' from JSON`);
}
exports.webhookPaginatedListFromJSON = webhookPaginatedListFromJSON;
