// shippo/models/components/webhookpayloadtransaction.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.webhookPayloadTransactionFromJSON = exports.webhookPayloadTransactionToJSON = exports.WebhookPayloadTransaction$ = exports.WebhookPayloadTransaction$outboundSchema = exports.WebhookPayloadTransaction$inboundSchema = void 0;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const transaction_js_1 = require("./transaction.js");
const webhookeventtypeenum_js_1 = require("./webhookeventtypeenum.js");
/** @internal */
exports.WebhookPayloadTransaction$inboundSchema = z.object({
    event: webhookeventtypeenum_js_1.WebhookEventTypeEnum$inboundSchema.optional(),
    test: z.boolean().optional(),
    data: transaction_js_1.Transaction$inboundSchema.optional(),
});
/** @internal */
exports.WebhookPayloadTransaction$outboundSchema = z.object({
    event: webhookeventtypeenum_js_1.WebhookEventTypeEnum$outboundSchema.optional(),
    test: z.boolean().optional(),
    data: transaction_js_1.Transaction$outboundSchema.optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebhookPayloadTransaction$;
(function (WebhookPayloadTransaction$) {
    /** @deprecated use `WebhookPayloadTransaction$inboundSchema` instead. */
    WebhookPayloadTransaction$.inboundSchema = exports.WebhookPayloadTransaction$inboundSchema;
    /** @deprecated use `WebhookPayloadTransaction$outboundSchema` instead. */
    WebhookPayloadTransaction$.outboundSchema = exports.WebhookPayloadTransaction$outboundSchema;
})(WebhookPayloadTransaction$ || (exports.WebhookPayloadTransaction$ = WebhookPayloadTransaction$ = {}));
function webhookPayloadTransactionToJSON(webhookPayloadTransaction) {
    return JSON.stringify(exports.WebhookPayloadTransaction$outboundSchema.parse(webhookPayloadTransaction));
}
exports.webhookPayloadTransactionToJSON = webhookPayloadTransactionToJSON;
function webhookPayloadTransactionFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.WebhookPayloadTransaction$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookPayloadTransaction' from JSON`);
}
exports.webhookPayloadTransactionFromJSON = webhookPayloadTransactionFromJSON;
