// shippo/models/components/webhookpayloadbatch.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.webhookPayloadBatchFromJSON = exports.webhookPayloadBatchToJSON = exports.WebhookPayloadBatch$ = exports.WebhookPayloadBatch$outboundSchema = exports.WebhookPayloadBatch$inboundSchema = void 0;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const webhookeventtypeenum_js_1 = require("./webhookeventtypeenum.js");
/** @internal */
exports.WebhookPayloadBatch$inboundSchema = z.object({
    event: webhookeventtypeenum_js_1.WebhookEventTypeEnum$inboundSchema.optional(),
    test: z.boolean().optional(),
    data: z.string().optional(),
});
/** @internal */
exports.WebhookPayloadBatch$outboundSchema = z.object({
    event: webhookeventtypeenum_js_1.WebhookEventTypeEnum$outboundSchema.optional(),
    test: z.boolean().optional(),
    data: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebhookPayloadBatch$;
(function (WebhookPayloadBatch$) {
    /** @deprecated use `WebhookPayloadBatch$inboundSchema` instead. */
    WebhookPayloadBatch$.inboundSchema = exports.WebhookPayloadBatch$inboundSchema;
    /** @deprecated use `WebhookPayloadBatch$outboundSchema` instead. */
    WebhookPayloadBatch$.outboundSchema = exports.WebhookPayloadBatch$outboundSchema;
})(WebhookPayloadBatch$ || (exports.WebhookPayloadBatch$ = WebhookPayloadBatch$ = {}));
function webhookPayloadBatchToJSON(webhookPayloadBatch) {
    return JSON.stringify(exports.WebhookPayloadBatch$outboundSchema.parse(webhookPayloadBatch));
}
exports.webhookPayloadBatchToJSON = webhookPayloadBatchToJSON;
function webhookPayloadBatchFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.WebhookPayloadBatch$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookPayloadBatch' from JSON`);
}
exports.webhookPayloadBatchFromJSON = webhookPayloadBatchFromJSON;
