// shippo/models/components/webhookpayload.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.webhookPayloadFromJSON = exports.webhookPayloadToJSON = exports.WebhookPayload$ = exports.WebhookPayload$outboundSchema = exports.WebhookPayload$inboundSchema = void 0;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const webhookpayloadbatch_js_1 = require("./webhookpayloadbatch.js");
const webhookpayloadtrack_js_1 = require("./webhookpayloadtrack.js");
const webhookpayloadtransaction_js_1 = require("./webhookpayloadtransaction.js");
/** @internal */
exports.WebhookPayload$inboundSchema = z.union([
    webhookpayloadtrack_js_1.WebhookPayloadTrack$inboundSchema.and(z.object({ event: z.literal("track_updated") }).transform((v) => ({
        event: v.event,
    }))),
    webhookpayloadbatch_js_1.WebhookPayloadBatch$inboundSchema.and(z.object({ event: z.literal("batch_created") }).transform((v) => ({
        event: v.event,
    }))),
    webhookpayloadbatch_js_1.WebhookPayloadBatch$inboundSchema.and(z.object({ event: z.literal("batch_purchased") }).transform((v) => ({
        event: v.event,
    }))),
    webhookpayloadtransaction_js_1.WebhookPayloadTransaction$inboundSchema.and(z.object({ event: z.literal("transaction_created") }).transform((v) => ({
        event: v.event,
    }))),
    webhookpayloadtransaction_js_1.WebhookPayloadTransaction$inboundSchema.and(z.object({ event: z.literal("transaction_updated") }).transform((v) => ({
        event: v.event,
    }))),
]);
/** @internal */
exports.WebhookPayload$outboundSchema = z.union([
    webhookpayloadtrack_js_1.WebhookPayloadTrack$outboundSchema.and(z.object({ event: z.literal("track_updated") }).transform((v) => ({
        event: v.event,
    }))),
    webhookpayloadbatch_js_1.WebhookPayloadBatch$outboundSchema.and(z.object({ event: z.literal("batch_created") }).transform((v) => ({
        event: v.event,
    }))),
    webhookpayloadbatch_js_1.WebhookPayloadBatch$outboundSchema.and(z.object({ event: z.literal("batch_purchased") }).transform((v) => ({
        event: v.event,
    }))),
    webhookpayloadtransaction_js_1.WebhookPayloadTransaction$outboundSchema.and(z.object({ event: z.literal("transaction_created") }).transform((v) => ({
        event: v.event,
    }))),
    webhookpayloadtransaction_js_1.WebhookPayloadTransaction$outboundSchema.and(z.object({ event: z.literal("transaction_updated") }).transform((v) => ({
        event: v.event,
    }))),
]);
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebhookPayload$;
(function (WebhookPayload$) {
    /** @deprecated use `WebhookPayload$inboundSchema` instead. */
    WebhookPayload$.inboundSchema = exports.WebhookPayload$inboundSchema;
    /** @deprecated use `WebhookPayload$outboundSchema` instead. */
    WebhookPayload$.outboundSchema = exports.WebhookPayload$outboundSchema;
})(WebhookPayload$ || (exports.WebhookPayload$ = WebhookPayload$ = {}));
function webhookPayloadToJSON(webhookPayload) {
    return JSON.stringify(exports.WebhookPayload$outboundSchema.parse(webhookPayload));
}
exports.webhookPayloadToJSON = webhookPayloadToJSON;
function webhookPayloadFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.WebhookPayload$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookPayload' from JSON`);
}
exports.webhookPayloadFromJSON = webhookPayloadFromJSON;
