// shippo/models/components/webhookeventtypeenum.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookEventTypeEnum$ = exports.WebhookEventTypeEnum$outboundSchema = exports.WebhookEventTypeEnum$inboundSchema = exports.WebhookEventTypeEnum = void 0;
const z = __importStar(require("zod"));
/**
 * Type of event that triggered the webhook.
 */
exports.WebhookEventTypeEnum = {
    TransactionCreated: "transaction_created",
    TransactionUpdated: "transaction_updated",
    TrackUpdated: "track_updated",
    BatchCreated: "batch_created",
    BatchPurchased: "batch_purchased",
    All: "all",
};
/** @internal */
exports.WebhookEventTypeEnum$inboundSchema = z.nativeEnum(exports.WebhookEventTypeEnum);
/** @internal */
exports.WebhookEventTypeEnum$outboundSchema = exports.WebhookEventTypeEnum$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebhookEventTypeEnum$;
(function (WebhookEventTypeEnum$) {
    /** @deprecated use `WebhookEventTypeEnum$inboundSchema` instead. */
    WebhookEventTypeEnum$.inboundSchema = exports.WebhookEventTypeEnum$inboundSchema;
    /** @deprecated use `WebhookEventTypeEnum$outboundSchema` instead. */
    WebhookEventTypeEnum$.outboundSchema = exports.WebhookEventTypeEnum$outboundSchema;
})(WebhookEventTypeEnum$ || (exports.WebhookEventTypeEnum$ = WebhookEventTypeEnum$ = {}));
