// shippo/models/components/webhookpayloadtrack.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.webhookPayloadTrackFromJSON = exports.webhookPayloadTrackToJSON = exports.WebhookPayloadTrack$ = exports.WebhookPayloadTrack$outboundSchema = exports.WebhookPayloadTrack$inboundSchema = void 0;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const track_js_1 = require("./track.js");
const webhookeventtypeenum_js_1 = require("./webhookeventtypeenum.js");
/** @internal */
exports.WebhookPayloadTrack$inboundSchema = z.object({
    event: webhookeventtypeenum_js_1.WebhookEventTypeEnum$inboundSchema.optional(),
    test: z.boolean().optional(),
    data: track_js_1.Track$inboundSchema.optional(),
});
/** @internal */
exports.WebhookPayloadTrack$outboundSchema = z.object({
    event: webhookeventtypeenum_js_1.WebhookEventTypeEnum$outboundSchema.optional(),
    test: z.boolean().optional(),
    data: track_js_1.Track$outboundSchema.optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebhookPayloadTrack$;
(function (WebhookPayloadTrack$) {
    /** @deprecated use `WebhookPayloadTrack$inboundSchema` instead. */
    WebhookPayloadTrack$.inboundSchema = exports.WebhookPayloadTrack$inboundSchema;
    /** @deprecated use `WebhookPayloadTrack$outboundSchema` instead. */
    WebhookPayloadTrack$.outboundSchema = exports.WebhookPayloadTrack$outboundSchema;
})(WebhookPayloadTrack$ || (exports.WebhookPayloadTrack$ = WebhookPayloadTrack$ = {}));
function webhookPayloadTrackToJSON(webhookPayloadTrack) {
    return JSON.stringify(exports.WebhookPayloadTrack$outboundSchema.parse(webhookPayloadTrack));
}
exports.webhookPayloadTrackToJSON = webhookPayloadTrackToJSON;
function webhookPayloadTrackFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.WebhookPayloadTrack$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookPayloadTrack' from JSON`);
}
exports.webhookPayloadTrackFromJSON = webhookPayloadTrackFromJSON;
