// shippo/models/components/webhookupdaterequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.webhookUpdateRequestFromJSON = exports.webhookUpdateRequestToJSON = exports.WebhookUpdateRequest$ = exports.WebhookUpdateRequest$outboundSchema = exports.WebhookUpdateRequest$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const webhookeventtypeenum_js_1 = require("./webhookeventtypeenum.js");
/** @internal */
exports.WebhookUpdateRequest$inboundSchema = z.object({
    event: webhookeventtypeenum_js_1.WebhookEventTypeEnum$inboundSchema,
    url: z.string(),
    active: z.boolean().optional(),
    is_test: z.boolean().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "is_test": "isTest",
    });
});
/** @internal */
exports.WebhookUpdateRequest$outboundSchema = z.object({
    event: webhookeventtypeenum_js_1.WebhookEventTypeEnum$outboundSchema,
    url: z.string(),
    active: z.boolean().optional(),
    isTest: z.boolean().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        isTest: "is_test",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var WebhookUpdateRequest$;
(function (WebhookUpdateRequest$) {
    /** @deprecated use `WebhookUpdateRequest$inboundSchema` instead. */
    WebhookUpdateRequest$.inboundSchema = exports.WebhookUpdateRequest$inboundSchema;
    /** @deprecated use `WebhookUpdateRequest$outboundSchema` instead. */
    WebhookUpdateRequest$.outboundSchema = exports.WebhookUpdateRequest$outboundSchema;
})(WebhookUpdateRequest$ || (exports.WebhookUpdateRequest$ = WebhookUpdateRequest$ = {}));
function webhookUpdateRequestToJSON(webhookUpdateRequest) {
    return JSON.stringify(exports.WebhookUpdateRequest$outboundSchema.parse(webhookUpdateRequest));
}
exports.webhookUpdateRequestToJSON = webhookUpdateRequestToJSON;
function webhookUpdateRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.WebhookUpdateRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'WebhookUpdateRequest' from JSON`);
}
exports.webhookUpdateRequestFromJSON = webhookUpdateRequestFromJSON;
